% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[  ASAPversion,
  ,
  9pt]{elife}
\usepackage{amsmath,amssymb}

\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}

\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={Accurate prediction of transition metal location from a single structure using deep learning},
  pdfauthor={Simon L. DürrAndrea LevyUrsula Röthlisberger},
  pdflang={en-US},
  pdfkeywords={metal, protein engineering, deep learning},
  pdfcreator={LaTeX via pandoc and Manubot}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering


\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{cslreferences}%
  {}%
  {\par}



%% pandoc-fignos: environment for tagged figures
\newenvironment{fignos:tagged-figure}[1][]{
  \let\oldthefigure\thefigure
  \let\oldtheHfigure\theHfigure
  \renewcommand{\thefigure}{#1}
  \renewcommand{\theHfigure}{#1}
}{
  \let\thefigure\oldthefigure
  \let\theHfigure\oldtheHfigure
  \addtocounter{figure}{-1}
}

%% pandoc-tablenos: environment for tagged tables
\newenvironment{tablenos:tagged-table}[1][]{
  \let\oldthetable\thetable
  \let\oldtheHtable\theHtable
  \renewcommand{\thetable}{#1}
  \renewcommand{\theHtable}{#1}
}{
  \let\thetable\oldthetable
  \let\theHtable\oldtheHtable
  \addtocounter{table}{-1}
}











\newcommand{\githubrepository}{\url{}}  %this should be the main github repository for this article

\title{Accurate prediction of transition metal location from a single structure using deep learning}





\author[1]{Simon L. Dürr}
\author[1]{Andrea Levy}
\author[1]{Ursula Röthlisberger}


\affil[1]{Laboratory of Computational Chemistry and Biochemistry, Institute of Chemical Sciences and Engineering, Swiss Federal Institute of Technology (EPFL) CH-1015 Lausanne, Switzerland}



\orcid{Simon L. Dürr}{0000-0002-4304-8106}
\orcid{Andrea Levy}{0000-0003-1255-859X}
\orcid{Ursula Röthlisberger}{0000-0002-1704-8591}


\blurb{This  document is maintained online on GitHub at \githubrepository; to provide feedback, suggestions, or help improve it, please visit the GitHub repository and participate via the issue tracker.}








\begin{document}


\begin{frontmatter}
\maketitle
\end{frontmatter}




This manuscript
was automatically generated
on July 18, 2022. View \href{doi}{non-interactive version on BioRxiv}.

\href{https://orcid.org/0000-0002-4304-8106}{\includegraphics[width=0.11458in,height=0.11458in]{images/orcid.svg}}
Simon L. Dürr 1,
\href{https://orcid.org/0000-0003-1255-859X}{\includegraphics[width=0.11458in,height=0.11458in]{images/orcid.svg}}
Andrea Levy 1,
\href{https://orcid.org/0000-0002-1704-8591}{\includegraphics[width=0.11458in,height=0.11458in]{images/orcid.svg}}
Ursula Röthlisberger 1,✉

✉ --- To whom correspondence should be addressed: ursula.roethlisberger@epfl.ch

Affiliations

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Laboratory of Computational Chemistry and Biochemistry, Institute of Chemical Sciences and Engineering, Swiss Federal Institute of Technology (EPFL) CH-1015 Lausanne, Switzerland
\end{enumerate}

\hypertarget{abstract}{%
\subsection{Abstract}\label{abstract}}

Metal ions are essential cofactors for many proteins. In fact, currently, about half of the structurally characterized proteins contain a metal ion. Metal ions play a crucial role for many applications such as enzyme design or design of protein-protein interactions because they are biologically abundant, tether to the protein using strong interactions, and have favorable catalytic properties e.g.~as Lewis acid. Computational design of metalloproteins is however hampered by the complex electronic structure of many biologically relevant metals such as zinc that can often not be accurately described using a classical force field. In this work, we develop two tools - Metal3D (based on 3D convolutional neural networks) and Metal1D (solely based on geometric criteria) to improve the identification and localization of zinc and other metal ions in experimental and computationally predicted protein structures. Comparison with other currently available tools shows that Metal3D is the most accurate metal ion location predictor to date outperforming geometric predictors including Metal1D by a wide margin using a single structure as input. Metal3D outputs a confidence metric for each predicted site and works on proteins with few homologes in the protein data bank. The predicted metal ion locations for Metal3D are within 0.70 ± 0.64 Å of the experimental locations with half of the sites below 0.5 Å. Metal3D predicts a global metal density that can be used for annotation of structures predicted using e.g.~AlphaFold2 and a per residue metal density that can be used in protein design workflows for the location of suitable metal binding sites and rotamer sampling to create novel metalloproteins. Metal3D is available as easy to use webapp, notebook or commandline interface.

\hypertarget{introduction}{%
\subsection{Introduction}\label{introduction}}

Metalloproteins are ubiquitous in nature and are present in all major enzyme families.\textsuperscript{\protect\hyperlink{ref-gxiv5uTU}{1},\protect\hyperlink{ref-LY9UGI7}{2}}The metals predominantly found in biological systems are the first and second row alkali and earth alkali metals and the first row transition metals such as zinc and copper. Zinc is the most common transition metal (present in \textasciitilde10\% of deposited structures) and can fulfill both a structural (e.g.~in zinc finger proteins) or a catalytic role in up to trinuclear active sites. Zn2+ is an excellent Lewis acid and is most often found in tetrahedral, pentavalent, or octahedral coordination. About 10 \% of all reactions catalyzed by enzymes use zinc as cofactor\textsuperscript{\protect\hyperlink{ref-iqbHG3rv}{3}}.

Metalloproteins are well studied because metal cofactors are essential for the function of many proteins and loss of this function is an important cause of diseases.\textsuperscript{\protect\hyperlink{ref-4sXp2sDw}{4}} Industrial applications for metalloproteins capitalize on the favorable catalytic properties of the metal ion where the protein environment dictates (stereo)-selectivity.\textsuperscript{\protect\hyperlink{ref-oOm5j05D}{5}--\protect\hyperlink{ref-13w4SiWCx}{7}} To crystallize proteins, metal salts are also often added to the crystallization buffer as they can help in the formation of protein crystals overcoming the enthalpic cost of association of protein surfaces. Metal ion binding sites can be used to engineer protein-protein interactions (PPI)\textsuperscript{\protect\hyperlink{ref-ROLANnJl}{10}--\protect\hyperlink{ref-17JMFzNJo}{12}} and the hypothesis has been put forward that one origin of macromolecular complexity is the superficial binding of metal ions in early single domain proteins.\textsuperscript{\protect\hyperlink{ref-17JMFzNJo}{12}}

While simple metal ion binding sites can be rapidly engineered because initial coordination on a protein surface can for example be achieved by creating an i, i+4 di-histidine site on an alpha-helix\textsuperscript{\protect\hyperlink{ref-io987fKE}{13}} or by placing cysteines in spatial proximity,\textsuperscript{\protect\hyperlink{ref-JXKFLH3I}{14}} the engineering of complex metal ion binding sites e.g.~in the protein interior is considerably more difficult\textsuperscript{\protect\hyperlink{ref-LY9UGI7}{2},\protect\hyperlink{ref-Lj2sbjzc}{11}} as such sites are often supported by a network of hydrogen bonds. A complication for computational design of metalloproteins is the unavailability of good (non-bonded) force fields for zinc and other transition metals that accurately reproduce (e.g.~tetrahedral) coordination with the correct coordination distances which renders design using e.g.~Rosetta very difficult.\textsuperscript{\protect\hyperlink{ref-LY9UGI7}{2},\protect\hyperlink{ref-1FR3e6DyA}{15}} In fact, the latest parametrization of the Rosetta energy function (ref2015)\textsuperscript{\protect\hyperlink{ref-1uPiiMnD}{16}} did not refit the parameters for the metal ions which originally are from CHARMM27 with empirically derived Lazaridis-Karplus solvation terms. To adequately treat metal sites in proteins quantum mechanical treatments such as in hybrid quantum mechanics/molecular mechanics (QM/MM) simulations\textsuperscript{\protect\hyperlink{ref-5Vxh8hay}{17},\protect\hyperlink{ref-IXq7vmo3}{18}} is needed whose computational cost is prohibitive for regular protein design tasks. QM/MM simulations can however be used to verify coordination chemistry for select candidate proteins.\textsuperscript{\protect\hyperlink{ref-nlfa683y}{19}} On the other hand, neural network potentials have been developed for zinc however those require the experimental zinc location as input.\textsuperscript{\protect\hyperlink{ref-b5EJeHX3}{20}}

Many tools exist to predict whether a protein contains metals (e.g.~ZincFinder\textsuperscript{\protect\hyperlink{ref-qNdK0F1w}{21}}), which residues in the protein bind a metal (e.g.~IonCom,\textsuperscript{\protect\hyperlink{ref-KNmKWbrP}{22}} MIB\textsuperscript{\protect\hyperlink{ref-1HMhB3vxM}{23}}) and where the metal is bound (AlphaFill,\textsuperscript{\protect\hyperlink{ref-13fLqNwbD}{24}} FindsiteMetal,\textsuperscript{\protect\hyperlink{ref-W4f7UG1p}{25}} BioMetAll,\textsuperscript{\protect\hyperlink{ref-iHxzzTCG}{26}}MIB\textsuperscript{\protect\hyperlink{ref-1HMhB3vxM}{23}} ). The input for these predictors is based on sequence and/or structure information. Sequence-based predictors use pattern recognition to identify the amino acids which might bind a metal.\textsuperscript{\protect\hyperlink{ref-13sNVOrHB}{27}}
Structure-based methods use homology to known structures (MIB, Findsite-metal, AlphaFill) or distance features (BioMetAll) to infer the location of metals. Some tools like Findsite-metal or ZincFinder employ machine learning based approaches such as support vector machines.

Structure based deep learning approaches have been used in the field of protein research for a variety of applications such as protein structure prediction,\textsuperscript{\protect\hyperlink{ref-yZfcMIwh}{28}} prediction of identity of masked residues\textsuperscript{\protect\hyperlink{ref-ls5kyxZ3}{30}--\protect\hyperlink{ref-6szq3cwi}{32}}), functional site prediction,\textsuperscript{\protect\hyperlink{ref-SGggOw7Z}{33},\protect\hyperlink{ref-MKP16DSu}{34}} for ranking of docking poses,\textsuperscript{\protect\hyperlink{ref-q224Kv8w}{35},\protect\hyperlink{ref-yBhqGkBa}{36}} prediction of the location of ligands,\textsuperscript{\protect\hyperlink{ref-yBhqGkBa}{36}--\protect\hyperlink{ref-9UNjBvCL}{40}} and prediction of effects of mutations for stability and disease.\textsuperscript{\protect\hyperlink{ref-4sXp2sDw}{4}}
Current state of the art predictors for metal location are MIB,\textsuperscript{\protect\hyperlink{ref-1HMhB3vxM}{23},\protect\hyperlink{ref-k859wJxx}{42}} which combines structural and sequence information in the ``Fragment Transformation Method'' to search for homologous sites in its database, and BioMetAll,\textsuperscript{\protect\hyperlink{ref-iHxzzTCG}{26}} a geometrical predictor based on backbone preorganization. Both methods have significant drawbacks: MIB excludes metal sites with less than 2 coordination partners from its analysis and is limited by the availability of templates in its database. BioMetAll does not use templates but provides many possible locations for putative binding sites on a regular grid. The individual probes in BioMetAll do not have a confidence metric therefore only allowing to rank sites by the number of probes found, which results in a large uncertainty in the position. Both tools suffer from many false positives. In this work, we present two metal ion location predictors that do not suffer from these drawbacks. The deep learning based Metal3D predictor operates on a voxelized representation of the protein environment and predicts a per residue metal density that can be averaged to get a smooth metal probability density over the whole protein. The distance based predictor Metal1D predicts the location of metals using distances mined from the protein data bank (PDB) directly predicting coordinates of the putative metal binding site. These tools pave the way to perform in silico design of metal ion binding sites without relying on predefined geometrical rules or expensive quantum mechanical calculations.

\hypertarget{results}{%
\subsection{Results}\label{results}}

A dataset of experimental high resolution crystal structures (2085 structures/252324 voxelized environments) containing zinc sites was used for training of the geometric predictor Metal1D and the deep learning predictor Metal3D (Figure \ref{fig:method}). For training, we used the crystal environment including crystal contacts. For predictions, the biological assembly was used.

\begin{figure}
\hypertarget{fig:method}{%
\centering
\includegraphics{images/Metal3D_Metal1D_method.png}
\caption{\textbf{Workflow of Metal3D and Metal1D} \textbf{A} Training of Metal3D and Metal1D is based on experimental Zn2+ sites. Metal1D extracts coordination environments from LINK records, Metal3D is a fully convolutional 3DCNN trained to predict the metal density from voxelized protein environments. \textbf{B} In inference mode Metal3D predicts the location of a metal ion by computing per residue metal densities and then averaging them to obtain a global metal density for the input proteins. The ions can then be placed using the weighted average of voxels above a cutoff. For Metal1D all residues in the protein are scanned for compatibility with the probability map. Metals are placed at the geometric center of residues with high scores according to the probability map. A final ranking of sites is obtained using the probability map.}\label{fig:method}
}
\end{figure}

\hypertarget{metal3d}{%
\subsubsection{Metal3D}\label{metal3d}}

Metal3D takes a protein structure and a set of residues as input, voxelizes the environment around each of the residues and predicts the per residue metal density. The predicted per residue densities (within a 16 x 16 x 16 Å3 volume) can then be averaged to yield a zinc density for the whole protein. At high probability cutoffs the predicted metal densities are spherical (Figure \ref{fig:2cbaprobabilities} E), at low probability cutoffs the predicted densities are non-regular (Figure \ref{fig:2cbaprobabilities} A).

\begin{figure}
\hypertarget{fig:2cbaprobabilities}{%
\centering
\includegraphics{images/2CBA_probabilities.png}
\caption{\textbf{Metal3D probability density} Probability evolution in HCA2 (PDB 2CBA) for different probability cutoffs A) p=0.1 B) p=0.2 C) p=0.3 D) p=0.4 E) p=0.75.}\label{fig:2cbaprobabilities}
}
\end{figure}

We evaluated the quality of the metal densities generated by the model with the discretized Jaccard similarity (Figure \ref{fig:jaccard}) for all environments in the test set. We noticed that at the edges of the residue-centered output densities often spurious density is predicted wherefore we evaluated the similarity of the test set metal density and the predicted metal probability density taking into account a smaller box with zeroed outer edges. Figure \ref{fig:jaccard} shows that the similarity of the boxes does not depend much on the probability cutoff chosen with higher cutoffs yielding slightly higher discretized Jaccard similarity values (0.02 - 0.04 difference between p=0.5 and p=0.9). Reducing the size of the analyzed boxes (i.e trimming of the edges) increases the Jaccard similarity from ≈ 0.64 to 0.88 showing that the metal density in the center of the box is more accurate than the density at the edges.

Metal3D is available as self-contained notebook on GoogleColab and on \href{https://hf.space/simonduerr/metal3d}{Huggingface Spaces}.

\hypertarget{metal1d}{%
\subsubsection{Metal1D}\label{metal1d}}

The statistical analysis for the geometric predictor uses the \texttt{LINK} records present in deposited PDB structures. A probability map for all zinc coordination motifs was extracted from all training structures (Figure \ref{fig:method} A). The mean coordination distance in the training set was found to be 2.2 ± 0.2 Å, and the default search radius for the predictions was therefore set to 5.5 Å (Table \ref{tbl:metal1dreferencepoints}). In total 208 different environments with more than 5 different proteins (at 30 \% sequence identity) were identified.
Metal1D is available as self-contained notebook on Google Colab

\hypertarget{comparison-of-metal1d-metal3d-mib-and-biometall}{%
\subsubsection{Comparison of Metal1D, Metal3D, MIB and BioMetAll}\label{comparison-of-metal1d-metal3d-mib-and-biometall}}

Existing metal ion predictors can be subdivided into two categories: binding site predictors and binding location predictors. The former identify only the residues binding the ion, the latter predict the coordinates of the metal ion itself. Both Metal1D and Metal3D can predict the coordinates of putative binding sites. We therefore assessed their performance by comparing to recent binding location predictors with available code/webserver: BioMetAll\textsuperscript{\protect\hyperlink{ref-iHxzzTCG}{26}} and MIB.\textsuperscript{\protect\hyperlink{ref-1HMhB3vxM}{23}} The main tuning parameter of MIB is the template similarity \texttt{t}, with higher values requiring higher similarity of the templates available for the search in structurally homologous metalloproteins. BioMetAll on the other hand was calibrated on available protein structures and places probes on a regular grid at all sites where the criteria for metal binding are fulfilled. The main adjustable parameter for BioMetAll is the cluster cutoff \texttt{c}, which indicates how many probes in reference to the largest cluster a specific cluster has. We used the recommended cutoff of 0.5 requiring all chosen clusters to have at least 50\% of the probes of the most populous cluster and used the cluster center to compute distances.

We first investigated the potential of all tools to detect the location of a zinc ion binding site in a binary fashion (zinc site or no zinc site). We defined a correctly identified binding site (true positive, TP) as a prediction within 5 Å of an experimental zinc site. In case a tool predicted no metal within the 5 Å radius, we counted this site as false negative (FN). False positive (FP) predictions, i.e sites where a metal was placed spuriously, were clustered in a 5 Å radius and counted once per cluster. All tools were assessed against the held out test biological assemblies for Metal3D and Metal1D. When the performance of MIB (\texttt{t=1.25}) and BioMetAll is compared against Metal3D with probability cutoff p=0.75 we find that Metal3D identifies more sites (85) than MIB (78) or BioMetAll (75) with a much lower number of false positives (Figure \ref{fig:comparison}).
MIB predicts 180 false positive sites, BioMetAll 134 sites whereas Metal3D only predicts 9 false positive sites at the p=0.75 cutoff. Metal1D (\texttt{t=0.5}) offers similar detection capabilities (78 sites detected) with a lower number of false positives (47) compared to MIB and BioMetAll.
As about 1/3 of the zinc binding sites in the PDB are believed to be artifacts\textsuperscript{\protect\hyperlink{ref-Wt0ducot}{43}} we removed 56 sites from the list of zinc sites in the test set (189 total) that had less than 2 unique protein ligands within 2.8 Å of the experimental zinc location. The amount of correct predictions in this reduced set is almost unchanged for all tools (Figure \ref{fig:comparison}) indicating that most tools correctly predict sites if they have 2 or more protein ligands. For Metal3D at p=0.75 and p=0.9 as well as MIB with t=1.9 all sites that are correctly predicted to contain a metal are sites with more than 2 protein ligands. The number of false negatives is reduced for all tools by about 50 sites indicating that most tools do not predict these crystallographic artifacts that might depend on additional coordinating residues from an adjacent molecule in the crystal. Of all tools, Metal3D has the least false positives (1 FP at p=0.9) and the highest number of detected sites (110 at p=0.25). The single false positive at p=0.9 does not contain a zinc ion but is a calcium binding site with three aspartates and one backbone carbonyl ligand (Figure \ref{fig:4jjjFP}).

\begin{figure}
\hypertarget{fig:comparison}{%
\centering
\includegraphics{images/all_2coord_metal_1d_3d_min_biometall_comparison.jpg}
\caption{\textbf{Identification of metal sites} Comparison of Metal1D, Metal3D, BioMetAll and MIB on the test set held out from training of Metal1D and Metal3D. Predicted sites are counted as true positives (TP) if they are within 5 Å of a true metal location and as false negatives (FN) otherwise. False positive (FP) probes are clustered and counted once per cluster. *For MIB we used 2 structures less because the server did not accept these structures.}\label{fig:comparison}
}
\end{figure}

After assessment of how many sites the tools predict, another crucial metric is the spatial precision of the predictions. For the correctly identified sites (TP) we measured the mean absolute distance (MAD) between experimental and predicted position (Figure \ref{fig:selectivity-mad}). The MAD for Metal3D at p=0.9 is 0.70 ± 0.64 Å and 0.73 ± 0.65 Å at p=0.25 indicating that low confidence predictions are still accurately placed inside the protein. The median MAD of predictions for Metal3D at p=0.9 is 0.52 Å indicating that for half of the predictions the model predicts at or better than the grid resolution of 0.5 Å.

BioMetAll is not very precise with an MAD for correctly identified sites of 2.80 ± 1.30 Å. BioMetAll predicts many possible locations per cluster with some of them much closer to the experimental metal binding site than the cluster center. However, it does not provide any ranking of the probes within a cluster and therefore the cluster center was used for the distance calculation. Metal1D (MAD 2.06 ± 1.33 Å) which identifies more sites than BioMetAll is also more precise than BioMetAll. MIB t=1.9 detects sites with high precision (MAD 0.77 ± 1.09 Å) but it relies on the existence of homologous sites to align the found sites.

\begin{figure}
\hypertarget{fig:selectivity-mad}{%
\centering
\includegraphics{images/Figure3_selectivity_mad.png}
\caption{\textbf{Precision of predicted sites and selectivity for Zn2+} \textbf{A} Mean absolute deviation (MAD) of predicted zinc ion locations using Metal1D, Metal3D, BioMetAll and MIB on the test set used to train Metal1D and Metal3D for all correctly identified (TP) sites. n is the number of sites predicted by the tool. *For MIB we used 2 structures less because the server did not accept these structures. For each tool the whisker plot indicates the median (white dot) and the first quartiles (black box). \textbf{B} Recall for the zinc test set and 25 randomly drawn structures for other transition, alkali and earth-alkali metal ions for Metal3D using p=0.5 as cutoff.}\label{fig:selectivity-mad}
}
\end{figure}

\hypertarget{selectivity-for-other-metals}{%
\subsubsection{Selectivity for other metals}\label{selectivity-for-other-metals}}

Both Metal3D and Metal1D were exclusively trained on zinc and we assessed their performance on sodium (Na+, PDB code NA), potassium (K+, PDB code K), calcium (Ca2+, PDB code CA), magnesium (Mg2+, PDB code MG), and various transition metals (Fe2+, Fe3+, Co2+, Cu2+, Cu+, Mn2+ with corresponding PDB codes FE2, FE, CO, CU, CU1, MN, respectively) from 25 randomly drawn structures from the clustered PDB at 30\% identity. Only sites with at least 3 unique protein ligands were used for the analysis to exclude crystallographic artifacts and use only highly defined sites which should exhibit most selectivity towards a specific metal. Figure \ref{fig:selectivity-mad} B shows that recall for Metal3D is high for all transition metals, meaning that the model correctly finds most sites even though it was only trained on zinc. For the alkali and earth alkali metals recall is much lower as the model only finds some sites. The mean probability for found zinc structures (ZN p=0.95 ± 0.10) in the test set is higher than for the other transition metals (Figure \ref{fig:selectivity-probability-metal3d}) and significantly higher than the probability for alkali metals (NA p=0.61 ± 0.10, K p=0.79 ± 0.16) while the probability for the earth alkali metals is slightly higher with MG (p=0.77 ± 0.16) similar to CA (p=0.73 ± 0.16). The MAD for each found metal site is again lowest for zinc (0.56 ± 0.59 Å). The MAD for the found sodium (n=2) and potassium (n=5) sites are as low as for the other transition metals. The only metal with significantly higher MAD (1.45 ± 0.93 Å) is CU1 (Figure \ref{fig:selectivity-distance-metal3d}).

The only two structures where a sodium is detected by Metal3D (2OKQ\textsuperscript{\protect\hyperlink{ref-Qubi5dZD}{44}}, 6KFN\textsuperscript{\protect\hyperlink{ref-g7lHXZp5}{45}}) have at least 2 side chain coordinating ligand atoms and only one backbone (2OKQ) or no backbone ligand atom (6KFN). Canonical sodium binding sites e.g.~such as in PDB 4I0W\textsuperscript{\protect\hyperlink{ref-w9ECLZDp}{46}} with two coordinating backbone carbonyl oxygen atoms and one asparagine side chain have probabilities around 5 \% and are basically indistinguishable from background noise of the model.
For Metal1D overall recall is lower with similar differences in the detection of main group metals versus transition metals (Figures \ref{fig:selectivity-metal1d} and \ref{fig:selectivity-distance-metal1d}).

\hypertarget{applications}{%
\subsubsection{Applications}\label{applications}}

After having evaluated the accuracy of Metal3D on held out test structures we also investigated possible uses in downstream applications such as protein function annotation and protein design.

\hypertarget{alpha-fold}{%
\paragraph{Alpha Fold}\label{alpha-fold}}

AlphaFold2 often predicts side chains in metal ion binding sites in the holo conformation.\textsuperscript{\protect\hyperlink{ref-yZfcMIwh}{28}} Tools like AlphaFill\textsuperscript{\protect\hyperlink{ref-13fLqNwbD}{24}}use structural homology to transplant metals from similar PDB structures to the predicted structure. Metal3D does not require explicit homology based on sequence or structural alignment like AlphaFill so it is potentially suited to annotate the dark proteome that is now accessible from the AlphaFold database with metal binding sites. Metal3D identifies both the catalytic site (1) and the zinc finger (2) for the example (PDB 3RZV\textsuperscript{\protect\hyperlink{ref-wW700ShK}{47}} , Figure \ref{fig:alphafold} A) used in ref\textsuperscript{\protect\hyperlink{ref-13fLqNwbD}{24}} with high probability (p=0.99) even though one of the sites in the AlphaFold model is slightly disordered with one of the binding residues in the solvent facing conformation (D309). The distances between predicted and modeled metal locations for Metal3D are 0.22 Å and 0.37 Å, for AlphaFill they are 0.21 Å and 0.41 Å.

AlphaFill uses a 25\% sequence identity cutoff which can be problematic for certain proteins with no structurally characterized homologues. For human Palmitoyltransferase ZDHHC23 (Uniprot Q8IYP9) a high confidence AlphaFold2 prediction exists but AlphaFill cannot place the zinc ions because the sequence overlap is 24\% to the closest PDB structure (PDB 6BMS\textsuperscript{\protect\hyperlink{ref-RFwEiPTW}{48}}), just below the 25\% cutoff. For the identical site in another human Palmitoyltransferase ZDHHC15 (Uniprot Q96MV8) AlphaFill is able to place the metal because of higher sequence identity to 6BMS (64\%) (Figure \ref{fig:alphafold} B). Metal3D in both cases is able to place the metal with high confidence (MAD 0.75 Å for site 1 and 0.48 Å for site 2, p\textgreater0.99 ) based on the single input structure alone.

\begin{figure}
\hypertarget{fig:alphafold}{%
\centering
\includegraphics{images/3rzv_6bsm_AF2_alphafill_metal3d.png}
\caption{\textbf{Annotation of AlphaFold2 structures} \textbf{A} Predicted metal binding sites (a and b) from Metal3D, respectively AlphaFill compared to the experimentally found zinc positions for Uniprot O95630. Metal3D places the metal with high accuracy even if sidechains are not perfectly predicted by AlphaFold for site 1 \textbf{B} Palmitoyltransferase ZDHHC23 (Uniprot Q8IYP9) and ZDHHC15 (Uniprot Q96MV8). AlphaFill can only place the metal for ZDHHC15 because sequence identity for ZDHHC23 is 24 \%. Probability isosurfaces from Metal3D for both structures at p=0.6, colored in gray.}\label{fig:alphafold}
}
\end{figure}

\hypertarget{metalloprotein-engineering}{%
\paragraph{Metalloprotein engineering}\label{metalloprotein-engineering}}

Human carbonic anhydrase II (HCA2) is a well studied metalloenzyme with a rich amount of mutational data available. For the crystal structure of the wildtype enzyme (PDB 2CBA\textsuperscript{\protect\hyperlink{ref-jyFnBdWm}{49}}), Metal3D recapitulates the location of the active site metal with a RMSD to the true metal location of 0.21 Å with a probability of p=0.99. At lower probability cutoffs (p\textless0.4) the probability map indicates further putative metal ion binding sites with interactions mediated by surface residues (e.g.~H36, D110, p=0.4) (Figure \ref{fig:2cbaprobabilities}).

To investigate the capabilities for protein engineering we used mutational data for first and second shell mutants of the active site residues in HCA2 with corresponding Kd values from a colorimetric assay.\textsuperscript{\protect\hyperlink{ref-13XuOF3Jj}{50}} For most mutants no crystal structures are available so we used the structure builder in the EVOLVE package to choose the most favorable rotamer for each single point mutation based on the EVOLVE-ddg energy function with explicit zinc present (modeled using a dummy atom approach\textsuperscript{\protect\hyperlink{ref-pchdLF0k}{51}}). he analysis was run for every single mutant and the resulting probability maps from Metal3D were analyzed. For the analysis we used the maximum predicted probability as a surrogate to estimate relative changes in Kd. For mutants that decrease zinc binding drastically we observe a drop in the maximum probability predicted by Metal3D (Figure \ref{fig:hca-kd}).The lowest probability mutants are H119N and H119Q with p=0.23 and 0.38. The mutant with the largest loss in zinc affinity H94A has a zinc binding probability of p=0.6. Conservative changes to the primary coordination motif (e.g.~H → C) reduce the predicted probability by 10 - 30 \%. For second shell mutants the influence of the mutations is less drastic with only minor changes in the predicted probabilities.

\begin{figure}
\hypertarget{fig:hca-kd}{%
\centering
\includegraphics{images/kd_vs_p_nolog_newmethod_newmodel_0.5.png}
\caption{\textbf{Protein design application} Experimentally measured Kd values\textsuperscript{\protect\hyperlink{ref-D3VQQdjb}{52}--\protect\hyperlink{ref-TNbPH3Y1}{56}} for 1st and 2nd shell active site mutants of HCA2 and predicted max probability for zinc using Metal3D.}\label{fig:hca-kd}
}
\end{figure}

\hypertarget{discussion}{%
\subsection{Discussion}\label{discussion}}

Metal3D predicts the probability distribution of zinc ions in protein crystal structures based on a neuronal network model trained on natural protein environments. The model performs a segmentation task to determine if a specific point in the input space contains a zinc ion or not. Metal3D predicts zinc ion sites with high accuracy making use of high resolution crystal structures (\textless2.5 Å). The use of high resolution structures is necessary because at resolutions greater than the average zinc ligand coordination distance (2.2 Å) the uncertainty of the zinc location noticeably increases\textsuperscript{\protect\hyperlink{ref-Wt0ducot}{43}} which would likely hamper the accuracy of the site prediction.

In contrast to currently available tools, for Metal3D, it is not necessary to filter the training examples for certain coordination requirements (i.e only sites with at least 2 protein ligands). The model thus sees the whole diversity of zinc ion sites present in the PDB. Such a model is advantageous since metalloprotein design workflows require models to score the full continuum of zinc sites starting from a suboptimal binding site only populated at high metal concentration to a highly organized zinc site in an enzyme with nanomolar metal affinity. The predicted probability can be used as a confidence metric or as an optimization target where mutations are made to increase probability of zinc binding.

\hypertarget{site-quality}{%
\subsubsection{Site quality}\label{site-quality}}

The fraction of artifactual zinc binding sites in the PDB is estimated to be about 1/3\textsuperscript{\protect\hyperlink{ref-Wt0ducot}{43}} similar to our test set used with 70\% (133) well coordinated zinc sites with at least 2 distinct protein ligands. To reduce the amount of artifactual sites in the training set we presented the model with as many complete sites as possible by using crystal symmetry to add adjacent coordinating protein chains. The frequency of artifacts in the training set is therefore much lower than 30 \%. The sites which still remain incomplete or that are wrongly modeled and not excluded through the resolution cutoffs and filtering procedures likely present only a small fraction of the training set and their signal is drowned out by the numerical superiority of the correctly modeled sites. If the model is used on artifactual sites or partially disordered ones it can still predict the metal location with high spatial accuracy but often indicates a lower confidence for the prediction (Figures \ref{fig:2cbaprobabilities} and \ref{fig:alphafold}).

Metal ion locators that rely on homology such as MIB perform worse on partial binding sites because reducing the quality of the available templates by including 1- or 2-coordinate sites would yield many false positives (similar to including less homologous structures for the template search). The deep learning based Metal3D can likely circumvent this because it does not require any engineered features to predict the location of the metal and learns directly from a full representation of the environment surrounding the binding site. This allows looking at low confidence sites in the context of a given environment.

\hypertarget{influence-of-non-protein-ligands}{%
\subsubsection{Influence of non-protein ligands}\label{influence-of-non-protein-ligands}}

Exogenous ligands play an important role for metals in biology as all empty coordination sites of metals are filled with water molecules in case there is no other exogenous ligand with higher affinity present (e.g.~a thiol). Like other predictors, both Metal1D and Metal3D do not consider water molecules or other ligands in the input as the quality of ligand molecules in the PDB varies.\textsuperscript{\protect\hyperlink{ref-9UNjBvCL}{40},\protect\hyperlink{ref-NifwNdQd}{57}} In addition, other potential sources of input such as AlphaFold do not provide explicit waters wherefore models should not rely on water as an input source. It is also not possible to use in silico water predictions because common water placement algorithms to place deep waters\textsuperscript{\protect\hyperlink{ref-9UNjBvCL}{40},\protect\hyperlink{ref-10C4imKQR}{58},\protect\hyperlink{ref-izkvZoLX}{59}} either rely on metal ions being present in the input or ignore them completely. Moreover, in protein design algorithms, water is usually only implicitly modeled (e.g.~in Rosetta).

For Metal3D, the input channel that encodes the total heavy atom density also encodes an implicit water density where all empty space can be interpreted as the solvent. For Metal1D, the contribution of water molecules is considered in an implicit way when the score is assigned to a site by considering coordinations including water compatible with the one observed (e.g.~a HIS3Wat site is equivalent to a His3 site for the scoring).

\hypertarget{choice-of-architecture}{%
\subsubsection{Choice of architecture}\label{choice-of-architecture}}

This work is the first to report a modern deep learning based model destined for identification of metal ligands in proteins. Similar approaches have been used in the more general field of protein-ligand docking where a variety of architectures and representations have been used. 3D CNN based approaches such as LigVoxel\textsuperscript{\protect\hyperlink{ref-mVGaXlum}{38}} and DeepSite\textsuperscript{\protect\hyperlink{ref-Gq7n6vhH}{37}} commonly use a resolution of 1 Å and similar input features as our model to predict the ligand density. However, predicting the density of a multi-atomic ligand is more complex than predicting the density of mononuclear metal ions. We therefore did not deem it necessary to include a conditioning on how many metal ions are present in the box and rather chose to reflect this in the training data where the model needs to learn that only about half of the environments it sees contain one or more metals. This choice is validated by the fact that the output probability densities at sufficiently high probability cutoffs are spherical with their radius approximately matching the van der Waals radius of zinc.

Mesh convolutional neural networks trained on a protein surface representation\textsuperscript{\protect\hyperlink{ref-yBhqGkBa}{36}} also have been used to predict the location and identity of protein ligands but this approach can only label the regions of the surface that bind the metal ion and is conceptually not able to return the exact location of the metal. Some metal ion binding sites are also heavily buried inside proteins as they mediate structural stability rendering them inaccessible to a surface based approach. The most recent approaches such as EquiBind\textsuperscript{\protect\hyperlink{ref-13hjzBbje}{39}} use equivariant neural networks such as En-Transformer\textsuperscript{\protect\hyperlink{ref-Fc42fl1x}{60}} to predict binding keypoints (defined as 1/2 distance between the Cɑ of the binding residue and a ligand atom). Explicit side chains are still too expensive for such models and these models assume a fixed known stoichiometry of the protein and ligand. Metal3D can also deal with proteins that do not bind a metal and does not assume that the amount of ions is known. The lack of explicit side chain information renders equivariant models unsuitable for the design of complex metal ion binding sites supported by an intricate network of hydrogen bonds that need to be positioned with sub-angstrom accuracy. Our model in contrast is less data- and compute-efficient than approaches representing the protein as graph due to the need to voxelize the input and provide different rotations of the input environment in training but the overall processing time for our model is still low taking typically 25 seconds for a 250 residue protein on a multicore GPU workstation.
Sequence based models\textsuperscript{\protect\hyperlink{ref-cbwvbkS6}{61},\protect\hyperlink{ref-Gj0pG9O5}{62}} can only use coevolution signals to infer residues in spatial proximity that can bind a metal. This might be difficult when it comes to ranking similar amino acids such as aspartate and glutamate or even ranking different rotamers where sub-angstrom level precision is needed to identify the mutant with the highest affinity for zinc.

\hypertarget{selectivity}{%
\subsubsection{Selectivity}\label{selectivity}}

In terms of selectivity both of our methods have a clear preference for transition metals over main group metals after having been trained exclusively on zinc binding sites. The only sites that Metal3D identifies for sodium in the test set are the ones that have side chain ligands. Many sodium and potassium sites are using backbone carbonyl coordination exclusively, which is not common for zinc and those sites are therefore not detected. Both of our methods could be rapidly adapted to predict not only location but also the identity of the metal. In the framework of Metal3D even a semantic metal prediction would be possible where the same model predicts different output channels for each metal it was trained on. To achieve perfect selectivity using such a model will be difficult because sometimes non-native metals are used for crystallization experiments. In this work we chose to work exclusively with zinc because it is the most redox stable transition metal and because many training examples are available.

\hypertarget{application-for-protein-design}{%
\subsubsection{Application for protein design}\label{application-for-protein-design}}

Protein design using 3DCNNs trained on residue identity has been successfully demonstrated and we anticipate that our model could be seamlessly integrated into such a workflow\textsuperscript{\protect\hyperlink{ref-6szq3cwi}{32}} to enable fully deep learning based design of metalloproteins. We are currently also investigating the combination of Metal3D combined with a classic energy-based genetic algorithm-based optimization to make design of metalloproteins\textsuperscript{\protect\hyperlink{ref-nlfa683y}{19}} easier without having to explicitly model the metal to compute the stability of the protein. As the model computes a probability density per residue it can be readily integrated into established software like Rosetta relying on rotamer sampling.

The HCA2 application demonstrates the utility of Metal3D for protein engineering (Figure \ref{fig:2cbaprobabilities}). The thermodynamics of metal ion binding to proteins are complicated\textsuperscript{\protect\hyperlink{ref-TlDhIlX}{63}} and there are currently no high-throughput based experimental approaches that could generate a dataset large enough to train a model directly on predicting Kd. The data we use were obtained from a colorimetric assay with very high affinity of zinc in the picomolar range.\textsuperscript{\protect\hyperlink{ref-D3VQQdjb}{52}--\protect\hyperlink{ref-TNbPH3Y1}{56}} More recent studies using ITC\textsuperscript{\protect\hyperlink{ref-TlDhIlX}{63}} instead of the colorimetric assay indicate lower Kd values in the nanomolar range for wild type HCA2. We can therefore only use the colorimetric data to estimate how well the model can recapitulate relative changes in the Kd for different mutations in the first and second shell of a prototypical metalloprotein.

Metal3D allows moving away from using rational approaches such as the i, i + 4 di His motifs used for the assembly and stabilization of metalloproteins to a fully automated approach where potential metal binding configurations can be scored computationally.\textsuperscript{\protect\hyperlink{ref-RJ44nj3R}{64}--\protect\hyperlink{ref-85PeY3aG}{66}}

\hypertarget{metal1d-vs.-metal3d}{%
\subsubsection{Metal1D vs.~Metal3D}\label{metal1d-vs.-metal3d}}

Metal1D is inferior to Metal3D for the prediction of metal ion binding sites because it produces more false positives while at the same time detecting fewer metal sites. Also the positioning of sites is somewhat imprecise. This demonstrates the inherent limitation of using solely distance based features for prediction of metal location. BioMetAll which is the tool most similar to Metal1D also suffers from many false positive predictions. In contrast, Metal1D is more data-efficient than Metal3D and provides predictions faster.

\hypertarget{conclusion}{%
\subsection{Conclusion}\label{conclusion}}

We present two metal ion location predictors: Metal3D based on 3D convolutional neural networks and Metal1D based on distances and amino acid propensity maps. Metal3D is the first tool with sub-angstrom level precision to predict the location of metal ions in proteins that does not rely on searching for structurally homologous proteins in a database. We therefore anticipate different applications such as protein-function annotation of structures predicted using AlphaFold2,\textsuperscript{\protect\hyperlink{ref-10bYywzs6}{67}} integration in protein design software and detection of cryptic metal binding sites that can be used to engineer PPIs. Such cryptic metal ion binding sites in common drug targets could also be used to engineer novel metallodrugs. Many of these applications will allow us to explore the still vastly untapped potential of proteins as large multi-dentate metal ligands with programmable surfaces.

\hypertarget{materials-and-methods}{%
\subsection{Materials and Methods}\label{materials-and-methods}}

\hypertarget{dataset}{%
\subsubsection{Dataset}\label{dataset}}

The input PDB files for training were obtained from the RCSB\textsuperscript{\protect\hyperlink{ref-8cYXwVxI}{68}} protein databank (downloaded 5th March 2021). We use a clustering of the structures at 30\% sequence identity using mmseqs2\textsuperscript{\protect\hyperlink{ref-jj6qOofE}{69}} to largely remove sequence and structural redundancy in the input dataset.
For each cluster, we check whether a zinc is contained in one of the structures, whether the resolution of these structures is better than 2.5 Å, if the experimental method is x-ray crystallography and whether the structure does not contain nucleic acids. If there are multiple structures fulfilling these criteria, the highest resolution structure is used. All structures larger than 3000 residues are discarded. We always use the first biological assembly to sample the training environments. The structures were stripped of all exogenous ligands except for zinc . If there are multiple models with e.g.~alternative residue conformations for a given structure, the first one is used. For each biological assembly we used the symmetry of the asymmetric unit to generate a protein structure that contains all neighboring copies of the protein in the crystal such that metal sites at crystal contacts are fully coordinated.

The train/val/test split was performed based on sequence identity using \texttt{easy-search} in mmseqs2. All proteins that had no (partial) sequence overlap with any other protein in the dataset were put into the test/val set (85 proteins) which we further split into a test set of 59 structures and a validation set of 26 structures. The training set contained 2085 structures. (Supplemental Data 1).

For the analysis, we always used the biological assembly and not the symmetry augmented structure. For the specificity analysis with respect to other transition metals, clusters from the PDB were randomly sampled to extract 25 biological assemblies per metal.

By default all zinc sites in the test and validation set were used for the analysis. Since some of the sites might be affected by the crystallization conditions, we also created a subset of all sites that contained at least 2 amino acid ligands to largely exclude crystallization artifacts. To analyze metal ion selectivity, we selected sites with at least 3 unique protein ligands to only use biologically significant sites with a high degree of metal preorganization as such sites should exhibit more selectivity for specific metals compared to sites with only 2 unique protein-ligands.

\hypertarget{metal-1d}{%
\subsubsection{Metal 1D}\label{metal-1d}}

Metal1D uses a probability map derived from \texttt{LINK} records in protein structures (Figure \ref{fig:method}). The \texttt{LINK} section of a PDB file specifies the connectivity between zinc (or any other metal ion) and the amino acids of the protein, and each \texttt{LINK} record specifies one linkage. This is an extension of the approach by Barber-Zucker \emph{et al.},\textsuperscript{\protect\hyperlink{ref-FB5cckAJ}{70}} in which \texttt{LINK} records were used to investigate the propensity of transition metals to bind different amino acids.

Using the training set we generated a probability map for the propensity of different coordination environments to bind a zinc (e.g CCCC, CCHH etc.). For each zinc ion the coordination is extracted from the\texttt{LINK} records excluding records involving only single amino acids (weak binding sites). Also, \texttt{LINK} records containing water molecules are excluded because of the difficulties in placing water molecules a posteriori in 3D structures when metal ions are present and because data quality of modelled water molecules varies. The probability map contains the counts of coordination environments found. \\
Making a prediction using Metal1D consists of two steps (Figure \ref{fig:method}): Identification of possible metal coordinating residues in the structure via the scoring of each amino acid, and scoring of the likelihood of coordination for putative sites predicted by placing a metal between the identified coordinating residues.

The protein structure is analyzed using the BioPandas python library.\textsuperscript{\protect\hyperlink{ref-CcRTAd7h}{71}} To identify coordinating residues, a per residue score is assigned by performing a geometrical search from a reference point, defined as the coordinate of the most probable metal binding atom, within a search radius considered as roughly twice the typical distance between the metal ion and the binding atom of amino acids in proteins (2.2 ± 0.2 Å as determined from \texttt{LINK} records). The search radius used was 5.5 Å in order to be able to take into account also deviations from the ideal coordination. In the case of amino acids which present more than one putative coordinating atom, such as e.g.~histidine, the mid-point between the donor atoms is used as reference point and the search radius is enlarged accordingly. The atoms used as reference points for each amino acid and the increase in the search radius are reported in Supplemental Table \ref{tbl:metal1dreferencepoints} .
The score is assigned to each amino acid considering all the other reference points of other amino acids within the search radius, and summing the probabilities in the probability map for coordinations compatible with the one observed. In the ideal case, a score of 1 corresponds to an amino acid surrounded by all possible coordinating amino acids observed in the probability map. In practice, scores result between 0 and \textless{} 1. Once all amino acids in the chain are scored, the metal location predictions are made grouping the highest-scored amino acids in clusters (defined as the ones within the chosen threshold with respect to the highest-scored one) based on distance. This is done using \texttt{scipy.spatial.distance\_matrix} and grouping together highest-scored amino acids closer than twice the search radius. For each cluster, a site prediction is made as a weighted average between the coordinates of the reference point of each amino acid, using as weighting factor the amino acid score. For isolated amino acids with a high score (e.g.~a single histidine) the same score is assigned to the closest reference point from another amino acid, to be able to compute the position of the metal as before. Possible artifacts resulting from this fictitious score are resolved in the final step of the prediction.

After the metal has been placed the likelihood of the putative sites can be assessed by performing a geometrical search centered on the predicted metal coordinates (within 60\% of the search radius, i.e.~3.3 Å) and a final score is now assigned to the site. The final score is assigned in the same way as the amino acid scores based on the probability map, and has the advantage of being able to sort the predicted metal sites based on their frequency in the training set. A cutoff parameter is used to exclude sites with a probability lower than a certain threshold with respect to the highest-scored one. This final scoring also mitigates the errors which can be introduced by calculating the coordinates of the site simply as a weighted average excluding or assigning a low probability to the site ending in unfavorable positions in space.

\hypertarget{metal-3d}{%
\subsubsection{Metal 3D}\label{metal-3d}}

\hypertarget{voxelization}{%
\paragraph{Voxelization}\label{voxelization}}

We used the moleculekit python library\textsuperscript{\protect\hyperlink{ref-mVGaXlum}{38},\protect\hyperlink{ref-q2gxO5bP}{72}} tto voxelize the input structures into 3D grids. 8 different input channels are used: aromatic, hydrophobic, positive ionizable, negative ionizable, hbond donor, hbond acceptor, occupancy, and metal ion binding site chain (Supplemental Table \ref{tbl:voxelchannels}). The channels are assigned using AutoDockVina atom names and a boolean mask. For each atom matching one of the categories a pair correlation function centered on the atom is used to assign the voxel value.\textsuperscript{\protect\hyperlink{ref-mVGaXlum}{38}} For the target tensor only the zinc ions were used for the voxelization. The target tensor was discretized setting any voxel above 0.05 to 1 (true location of zinc), all other to 0 (no zinc). We used a box size of 16 Å centered on the Cɑ atom of a residue, rotating each environment randomly for training before voxelization. The voxel grid used a 0.5 Å resolution for the input and target tensors. Any alternative side chain conformations modeled were discarded keeping only the highest occupancy. For the voxelization only heavy atoms were used. For all structures selected for the respective sets we partitioned the residues of the protein into residues within 12 Å of a zinc ion and those further away (based on the distance to the Cɑ atom). A single zinc site will therefore be present many times in the dataset but each time translated and rotated in the box. A balanced set of examples was used sampling equal numbers of residues that are close to a zinc and residues randomly drawn from the non-zinc binding residues. The sampling of residues is based on the biological assembly of the protein, the voxelization is based on the full 3D structure including neighboring asymmetric units in the crystal structure. The environments are precomputed and stored using lxf compression in HDF5 files for concurrent access during training. In total, 252324 environments were voxelized for the training set, 6550 for the test set, 3067 for the validation set. The voxelization was implemented using ray.\textsuperscript{\protect\hyperlink{ref-19MS2ZVR1}{73}}

\hypertarget{model-training}{%
\paragraph{Model training}\label{model-training}}

We used PyTorch 1.10\textsuperscript{\protect\hyperlink{ref-iOWq2JB8}{74}} to train the model. All layers of the network are convolutional layers with filter size 1.5 Å except for the fifth layer where a 8 Å filter is used to capture long range interactions. We use zero padding to keep the size of the boxes constant. Models were trained on a workstation with NVIDIA GTX3090 GPU and 32 CPU cores. Binary Cross Entropy\textsuperscript{\protect\hyperlink{ref-hchMCnC1}{75}} loss is used to train the model. The rectified linear unit (ReLU) non-linearity is used except for the last layer which uses a sigmoid function that yields the probability for zinc per voxel. A dropout layer (p = 0.1) was used between the 5th and 6th layers. The network was trained using AdaDelta employing a stepped learning rate (lr=0.5, ɣ=0.9), a batch size of 150, and 12 epochs to train.

\hypertarget{hyperparameter-tuning}{%
\paragraph{Hyperparameter tuning}\label{hyperparameter-tuning}}

We used the ray{[}tune{]} library\textsuperscript{\protect\hyperlink{ref-19MS2ZVR1}{73}} to perform a hyperparameter search choosing 20 different combinations between the following parameters:

\begin{itemize}
\tightlist
\item
  filtersize: 3,4 (in units of 0.5 Å)
\item
  dropout : 0.1, 0.2, 0.4, 0.5
\item
  learning rate : 0.5, 1.0, 2.0
\item
  gamma: 0.5, 0.7, 0.8, 0.9
\item
  largest dimension 80, 100, 120
\end{itemize}

\hypertarget{grid-averaging}{%
\paragraph{Grid Averaging}\label{grid-averaging}}

The model takes as input a \texttt{(8,32,32,32)} tensor and outputs a \texttt{(1,32,32,32)} tensor containing the probability density for zinc centered on the Cɑ atom of the input residue.
Predictions for a complete protein were obtained by voxelizing select residues of the protein (default all cysteines, histidines, aspartates, glutamates) and averaging the boxes using a global grid (Figure \ref{fig:method} B). 98 \% of the metal sites in the training data have at least one of those residues closeby wherefore this significant decrease in computational cost seems appropriate for most uses. The global grid is obtained by computing the bounding box of all points and using a regular spaced (0.5 Å) grid. For each grid point in the global grid the predicted probability maps within 0.25 Å of the grid point are averaged. The search is sped up using the KD-Tree implementation in scipy\textsuperscript{\protect\hyperlink{ref-8Miti2Gz}{76}}.

\hypertarget{metal-ion-placement}{%
\subparagraph{Metal ion placement}\label{metal-ion-placement}}

The global probability density is used to perform clustering of voxels above a certain probability threshold (default p=0.5, cutoff 7 Å) using agglomerative clustering implemented in scikit-learn.\textsuperscript{\protect\hyperlink{ref-AujvwLp6}{77}} For each cluster the weighted average of the voxels in the cluster is computed using the probabilities for each point as the weight. This results in one metal placed per cluster.

\hypertarget{visualization}{%
\subparagraph{Visualization}\label{visualization}}

We make available a command line program and interactive notebook allowing the user to visualize the results. The averaged probability map is stored as a \texttt{cube} file. The most likely metal coordinates for use in subsequent processing are stored in a \texttt{pdb}file. The command line program uses VMD\textsuperscript{\protect\hyperlink{ref-PEXAwV3k}{78}} to visualize the input protein and the predicted density, for the jupyter notebook 3Dmol.js/py3Dmol\textsuperscript{\protect\hyperlink{ref-BguJBlg6}{79}} is used.

\hypertarget{evaluation}{%
\paragraph{Evaluation}\label{evaluation}}

\hypertarget{comparison}{%
\subparagraph{Comparison}\label{comparison}}

In order to standardize the evaluation between different tools, we always used the same test set used for the training of Metal1D and Metal3D. In order to compute standard metrics such as precision and recall, we chose to assess the performance of all assessed tools (Metal1D, Metal3D, BioMetAll, MIB) in a binary fashion. Any prediction within 5 Å of an experimental metal site is counted as true positive (TP). Multiple predictions by the same tool for the same site are counted as 1 TP. Any experimental site that has no predicted metal within 5 Å is counted as false negative (FN). A false positive (FP) prediction is a prediction that is not within 5 Å of a zinc site and also not within 5 Å of any other false positive prediction. If two or more false positive predictions are within 5 Å, they are counted as a single false positive prediction for the same site. In practice we first evaluate the true positive and false negative predictions and remove those from the set of predicted positions. The remaining predictions are all false positives and are clustered using AgglomerativeClustering with a radius of 5 Å. The number of false positives is determined from the number of clusters. Using the binary metric we assessed how good the models are at discovering sites and how much these predictions can be trusted.

In order to assess the quality of the predictions, we additionally compute for all the true positive predictions the mean of the Euclidean distance between the true and predicted site (mean absolute deviation MAD). For Metal1D, MIB, and BioMetAll, MAD was computed for all predictions above the threshold within 5 Å of a true zinc site where \(\sum\text{predicted sites} \geq \sum\text{ TP}\). This was done as some tools predict the same site for different residue combinations and we wanted to assess the general performance for all predicted sites above a certain cutoff and not just for the best predicted site above the cutoff. For Metal3D the weighted average of all voxels above the cutoff was used.

Precision was calculated as

\[
\text{Precision} =\frac{\#\text{ correct metal sites}}{\#\text{ correct metal sites} + \#\text{ false positive clustered}} = \frac{\text{TP}}{\text{TP}+\text{FP}}
\]

Recall was calculated as

\[
\text{Recall} =\frac{\#\text{ correct metal sites}}{\#\text{ correct metal sites} + \#\text{ not found metal sites}}  = \frac{\text{TP}}{\text{TP}+\text{FN}}
\]

\hypertarget{model-assessment-metal3d}{%
\subparagraph{Model assessment Metal3D}\label{model-assessment-metal3d}}

To evaluate the trained models we monitored loss and how accurately the model predicts the metal density of the test set. We used a discretized version of the Jaccard index setting each voxel either as 0 (no metal) or 1 (zinc present). We tested multiple different decision boundaries (0.5, 0.6, 0.75, 0.9) and also compared a slightly smaller centered box to remove any spurious density at the box edges, where the model has only incomplete information to make predictions.

The Jaccard index is computed as
\[
J=\frac{\#\left|V_{p} \cap V_{exp}\right|}{\#\left|V_{p} \cup V_{exp}\right|},
\]
where \(Vp\) is the array of voxels with predicted probability above the decision boundary and \(V_{exp}\) is the array of voxels with the true metal locations also discretized at the same probability threshold.

\hypertarget{hca2-mutants}{%
\subparagraph{HCA2 mutants}\label{hca2-mutants}}

The data for human carbonic anhydrase 2 (HCA2) mutants was extracted from refs\textsuperscript{\protect\hyperlink{ref-D3VQQdjb}{52}--\protect\hyperlink{ref-TNbPH3Y1}{56}} and the crystal structure 2CBA\textsuperscript{\protect\hyperlink{ref-jyFnBdWm}{49},\protect\hyperlink{ref-WjHESl20}{80}} was used. The zinc was modeled using the zinc cationic dummy model forcefield\textsuperscript{\protect\hyperlink{ref-pchdLF0k}{51}} and we verified that energy minimization produced the correct coordination environment. The Richardson rotamer library\textsuperscript{\protect\hyperlink{ref-Mg40Hdad}{81}} was used with the EVOLVE-ddG energy function to compute the most stable rotamer for a given mutation with the zinc present. The lowest-energy mutant was used for the prediction of the location of metals using Metal3D.

\hypertarget{additional-information}{%
\subsection{Additional information}\label{additional-information}}

\hypertarget{acknowledgement}{%
\subsubsection{Acknowledgement}\label{acknowledgement}}

Supported by Swiss National Science Foundation Grant Number 200020-185092 with computational resources from the Swiss National Computing Centre CSCS.
We thank the developers of \href{https://3dmol.csb.pitt.edu/}{3Dmol.js} and \href{https://manubot.org/}{manubot} that made the interactive version of this manuscript possible.

\hypertarget{conflict-of-interest}{%
\subsubsection{Conflict of interest}\label{conflict-of-interest}}

None declared

\hypertarget{author-contributions}{%
\subsubsection{Author contributions}\label{author-contributions}}

S.L.D and A.L designed research, S.L.D, A.L, U.R conceptualized research, S.L.D and A.L developed methodology and software, S.L.D and A.L wrote first draft, S.L.D, A.L, U.R revised and edited draft, U.R supervised research and acquired funding.

\hypertarget{supplement}{%
\subsection{Supplement}\label{supplement}}

\hypertarget{metal1d-1}{%
\subsubsection{Metal1D}\label{metal1d-1}}

\begin{tablenos:tagged-table}[S1]

\begin{longtable}[]{@{}lccc@{}}
\caption{Atoms used as reference points for each amino acid in Metal1D. In the case of amino acids with more than one possible ligand atom, the search radius is enlarged, the increase is computed from the midpoint between all ligating atoms. Typical values computed for structure data files downloaded from the PDB are reported. \label{tbl:metal1dreferencepoints}}\tabularnewline
\toprule
Amino acid & Residue name & Label(s) & Search radius increase (Å)\tabularnewline
\midrule
\endfirsthead
\toprule
Amino acid & Residue name & Label(s) & Search radius increase (Å)\tabularnewline
\midrule
\endhead
Alanine & ALA & O & 0\tabularnewline
Arginine & ARG & NH1, NH2 & 1.2\tabularnewline
Asparagine & ASN & OD1 & 0\tabularnewline
Aspartic acid & ASP & OD1, OD2 & 1.105\tabularnewline
Cysteine & CYS & SG & 0\tabularnewline
Glutamic acid & GLU & OE1, OE2 & 1.105\tabularnewline
Glutamine & GLN & OE1 & 0\tabularnewline
Glycine & GLY & O & 0\tabularnewline
Histidine & HIS & ND1, ND2 & 1.08\tabularnewline
Isoleucine & ILE & O & 0\tabularnewline
Leucine & LEU & O & 0\tabularnewline
Lysine & LYS & NZ & 0\tabularnewline
Methionine & MET & SD & 0\tabularnewline
Phenylalanine & PHE & O & 0\tabularnewline
Proline & PRO & O & 0\tabularnewline
Serine & SER & OG & 0\tabularnewline
Threonine & THR & OG1 & 0\tabularnewline
Tryptophan & TRP & O & 0\tabularnewline
Tyrosine & TYR & OH & 0\tabularnewline
Valine & VAL & OH & 0\tabularnewline
\bottomrule
\end{longtable}

\end{tablenos:tagged-table}

\begin{tablenos:tagged-table}[S2]

\begin{longtable}[]{@{}ll@{}}
\caption{Atom selections used for voxelization of proteins using moleculekit \label{tbl:voxelchannels}}\tabularnewline
\toprule
\begin{minipage}[b]{0.14\columnwidth}\raggedright
Channel name\strut
\end{minipage} & \begin{minipage}[b]{0.81\columnwidth}\raggedright
Selected atoms\strut
\end{minipage}\tabularnewline
\midrule
\endfirsthead
\toprule
\begin{minipage}[b]{0.14\columnwidth}\raggedright
Channel name\strut
\end{minipage} & \begin{minipage}[b]{0.81\columnwidth}\raggedright
Selected atoms\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.14\columnwidth}\raggedright
aromatic\strut
\end{minipage} & \begin{minipage}[t]{0.81\columnwidth}\raggedright
HIS TRP TYR PHE sidechain without CB\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.14\columnwidth}\raggedright
hydrophobic\strut
\end{minipage} & \begin{minipage}[t]{0.81\columnwidth}\raggedright
element C\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.14\columnwidth}\raggedright
occupancy\strut
\end{minipage} & \begin{minipage}[t]{0.81\columnwidth}\raggedright
all protein heavy atoms\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.14\columnwidth}\raggedright
hbond donor\strut
\end{minipage} & \begin{minipage}[t]{0.81\columnwidth}\raggedright
(ASN GLN TRP MSE SER THR MET CYS and name ND2 NE2 NE1 SG SE OG OG1) and name N\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.14\columnwidth}\raggedright
hbond acceptor\strut
\end{minipage} & \begin{minipage}[t]{0.81\columnwidth}\raggedright
(resname ASP GLU HIS SER THR MSE CYS MET and name ND2 NE2 OE1 OE2 OD1 OD2 OG OG1 SE SG) or name O\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.14\columnwidth}\raggedright
metalbinding\strut
\end{minipage} & \begin{minipage}[t]{0.81\columnwidth}\raggedright
(name ND1 NE2 SG OE1 OE2 OD2) or (protein and name O N)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.14\columnwidth}\raggedright
positive charge\strut
\end{minipage} & \begin{minipage}[t]{0.81\columnwidth}\raggedright
resname LYS ARG HIS and name NZ NH1 NH2 ND1 NE2 NE\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.14\columnwidth}\raggedright
negative charge\strut
\end{minipage} & \begin{minipage}[t]{0.81\columnwidth}\raggedright
resname ASP GLU and name OD1 OD2 OE1 OE2\strut
\end{minipage}\tabularnewline
\bottomrule
\end{longtable}

\end{tablenos:tagged-table}

\begin{fignos:tagged-figure}[S1]

\begin{figure}
\hypertarget{fig:jaccard}{%
\centering
\includegraphics{images/jaccard_0.5.jpg}
\caption{Discretized Jaccard indices using different cutoffs for edge trimming and different probability cutoffs (p(metal)) showing that Metal3D predictions well reproduce the target environments in the test set.}\label{fig:jaccard}
}
\end{figure}

\end{fignos:tagged-figure}

\hypertarget{precisionrecall}{%
\subsubsection{Precision/Recall}\label{precisionrecall}}

\begin{fignos:tagged-figure}[S2]

\begin{figure}
\hypertarget{fig:precisionrecall}{%
\centering
\includegraphics{images/precisionrecall_0.5.jpg}
\caption{Precision recall curve for Metal1D and Metal3D with the probability cutoffs (Metal3D) or thresholds (Metal1D) used in the analysis.}\label{fig:precisionrecall}
}
\end{figure}

\end{fignos:tagged-figure}

\hypertarget{metal-selectivity}{%
\subsubsection{Metal selectivity}\label{metal-selectivity}}

\hypertarget{metal1d-2}{%
\paragraph{Metal1D}\label{metal1d-2}}

\begin{fignos:tagged-figure}[S3]

\begin{figure}
\hypertarget{fig:selectivity-metal1d}{%
\centering
\includegraphics{images/metal1D_metal_selectivity.jpg}
\caption{Recall for zinc testset and a 25 randomly drawn structures for other transition, alkali and earth-alkali metals for Metal1D}\label{fig:selectivity-metal1d}
}
\end{figure}

\end{fignos:tagged-figure}

\begin{fignos:tagged-figure}[S4]

\begin{figure}
\hypertarget{fig:selectivity-distance-metal1d}{%
\centering
\includegraphics{images/model_0.5metal1D_distances_violin.jpg}
\caption{Distance distribution Metal1D}\label{fig:selectivity-distance-metal1d}
}
\end{figure}

\end{fignos:tagged-figure}

\hypertarget{metal3d-1}{%
\paragraph{Metal3D}\label{metal3d-1}}

\begin{fignos:tagged-figure}[S5]

\begin{figure}
\hypertarget{fig:selectivity-distance-metal3d}{%
\centering
\includegraphics{images/model_0.5metal3D_distances_violin_0.5.jpg}
\caption{MAD for Metal3D on all true positive predictions in the test set.}\label{fig:selectivity-distance-metal3d}
}
\end{figure}

\end{fignos:tagged-figure}

\begin{fignos:tagged-figure}[S6]

\begin{figure}
\hypertarget{fig:selectivity-probability-metal3d}{%
\centering
\includegraphics{images/probability_violin.jpg}
\caption{Probability distribution for Metal3D on all true positive predictions in the test set.}\label{fig:selectivity-probability-metal3d}
}
\end{figure}

\end{fignos:tagged-figure}

\hypertarget{comparison-1}{%
\subsubsection{Comparison}\label{comparison-1}}

\begin{fignos:tagged-figure}[S7]

\begin{figure}
\hypertarget{fig:madonlyGoodZnmetal3d}{%
\centering
\includegraphics{images/mad_violin_0.5_2+.jpg}
\caption{MAD only 2 residue coordinated zincs}\label{fig:madonlyGoodZnmetal3d}
}
\end{figure}

\end{fignos:tagged-figure}

\hypertarget{mean-absolute-deviations}{%
\subparagraph{Mean absolute deviations}\label{mean-absolute-deviations}}

The measured MAD for all true positive zinc sites for each tool in the database.

\begin{tablenos:tagged-table}[S3]

\begin{longtable}[]{@{}lrr@{}}
\caption{MAD and median on zinc test set for all zinc sites (n=189). \label{tbl:compmadmediantest}}\tabularnewline
\toprule
tool & MAD {[}Å{]} & median {[}Å{]}\tabularnewline
\midrule
\endfirsthead
\toprule
tool & MAD {[}Å{]} & median {[}Å{]}\tabularnewline
\midrule
\endhead
BioMetAll c=0.5 & 2.72 ± 1.33 & 2.86\tabularnewline
MIB t=1.25 & 1.13 ± 1.24 & 0.60\tabularnewline
MIB t=1.9 & 0.77 ± 1.09 & 0.44\tabularnewline
Metal1D t=0.5 & 2.07 ± 1.33 & 2.06\tabularnewline
Metal1D t=0.75 & 2.19 ± 1.26 & 2.12\tabularnewline
Metal3D p=0.25 & 0.74 ± 0.66 & 0.61\tabularnewline
Metal3D p=0.5 & 0.73 ± 0.66 & 0.54\tabularnewline
Metal3D p=0.75 & 0.71 ± 0.64 & 0.51\tabularnewline
Metal3D p=0.9 & 0.70 ± 0.64 & 0.52\tabularnewline
\bottomrule
\end{longtable}

\end{tablenos:tagged-table}

\begin{tablenos:tagged-table}[S4]

\begin{longtable}[]{@{}lrr@{}}
\caption{MAD and median on zinc test set for zinc sites with at least 2 protein ligands (n=133). \label{tbl:compmadmediantestgoodonly}}\tabularnewline
\toprule
tool & MAD {[}Å{]} & median {[}Å{]}\tabularnewline
\midrule
\endfirsthead
\toprule
tool & MAD {[}Å{]} & median {[}Å{]}\tabularnewline
\midrule
\endhead
BioMetAll c=0.5 & 2.68 ± 1.33 & 2.84\tabularnewline
MIB t=1.25 & 1.09 ± 1.21 & 0.60\tabularnewline
MIB t=1.9 & 0.77 ± 1.09 & 0.44\tabularnewline
Metal1D t=0.5 & 1.97 ± 1.29 & 1.99\tabularnewline
Metal1D t=0.75 & 2.06 ± 1.24 & 2.09\tabularnewline
Metal3D p=0.25 & 0.69 ± 0.58 & 0.56\tabularnewline
Metal3D p=0.5 & 0.69 ± 0.59 & 0.54\tabularnewline
Metal3D p=0.75 & 0.71 ± 0.64 & 0.51\tabularnewline
Metal3D p=0.9 & 0.70 ± 0.64 & 0.52\tabularnewline
\bottomrule
\end{longtable}

\end{tablenos:tagged-table}

\hypertarget{false-positive-metal3d-p0.9}{%
\subparagraph{False positive Metal3D p=0.9}\label{false-positive-metal3d-p0.9}}

\begin{fignos:tagged-figure}[S8]

\begin{figure}
\hypertarget{fig:4jjjFP}{%
\centering
\includegraphics{images/4JJJ_FalsePositive_p=0.9_annotated.png}
\caption{False positive for Metal3D at p=0.9 in PDB 4JJJ. A calcium site is misclassified as zinc site.}\label{fig:4jjjFP}
}
\end{figure}

\end{fignos:tagged-figure}

\hypertarget{references}{%
\subsection{References}\label{references}}

\hypertarget{refs}{}
\begin{cslreferences}
\leavevmode\hypertarget{ref-gxiv5uTU}{}%
(1) Yu, F.; Cangelosi, V. M.; Zastrow, M. L.; Tegoni, M.; Plegaria, J. S.; Tebo, A. G.; Mocny, C. S.; Ruckthong, L.; Qayyum, H.; Pecoraro, V. L. Protein Design: Toward Functional Metalloenzymes. \emph{Chem. Rev.} \textbf{2014}, \emph{114} (7), 3495--3578. \url{https://doi.org/10.1021/cr400458x}.

\leavevmode\hypertarget{ref-LY9UGI7}{}%
(2) Guffy, S. L.; Der, B. S.; Kuhlman, B. Probing the Minimal Determinants of Zinc Binding with Computational Protein Design. \emph{Protein Engineering, Design and Selection} \textbf{2016}, \emph{29} (8), 327--338. \url{https://doi.org/10.1093/protein/gzw026}.

\leavevmode\hypertarget{ref-iqbHG3rv}{}%
(3) Andreini, C.; Bertini, I.; Cavallaro, G.; Holliday, G. L.; Thornton, J. M. Metal Ions in Biological Catalysis: From Enzyme Databases to General Principles. \emph{J Biol Inorg Chem} \textbf{2008}, \emph{13} (8), 1205--1218. \url{https://doi.org/10.1007/s00775-008-0404-5}.

\leavevmode\hypertarget{ref-4sXp2sDw}{}%
(4) Koohi-Moghadam, M.; Wang, H.; Wang, Y.; Yang, X.; Li, H.; Wang, J.; Sun, H. Predicting Disease-Associated Mutation of Metal-Binding Sites in Proteins Using a Deep Learning Approach. \emph{Nat Mach Intell} \textbf{2019}, \emph{1} (12), 561--567. \url{https://doi.org/10.1038/s42256-019-0119-z}.

\leavevmode\hypertarget{ref-oOm5j05D}{}%
(5) Studer, S.; Hansen, D. A.; Pianowski, Z. L.; Mittl, P. R. E.; Debon, A.; Guffy, S. L.; Der, B. S.; Kuhlman, B.; Hilvert, D. Evolution of a Highly Active and Enantiospecific Metalloenzyme from Short Peptides. \emph{Science} \textbf{2018}, \emph{362} (6420), 1285--1288. \url{https://doi.org/10.1126/science.aau3744}.

\leavevmode\hypertarget{ref-LAjO1aDL}{}%
(6) Key, H. M.; Dydio, P.; Clark, D. S.; Hartwig, J. F. Abiological Catalysis by Artificial Haem Proteins Containing Noble Metals in Place of Iron. \emph{Nature} \textbf{2016}, \emph{534} (7608), 534--537. \url{https://doi.org/10.1038/nature17968}.

\leavevmode\hypertarget{ref-13w4SiWCx}{}%
(7) Chalkley, M. J.; Mann, S. I.; DeGrado, W. F. De Novo Metalloprotein Design. \emph{Nat Rev Chem} \textbf{2021}, \emph{6} (1), 31--50. \url{https://doi.org/10.1038/s41570-021-00339-5}.

\leavevmode\hypertarget{ref-khOeAdye}{}%
(8) Der, B. S.; Edwards, D. R.; Kuhlman, B. Catalysis by a de Novo Zinc-Mediated Protein Interface: Implications for Natural Enzyme Evolution and Rational Enzyme Engineering. \emph{Biochemistry} \textbf{2012}, \emph{51} (18), 3933--3940. \url{https://doi.org/10.1021/bi201881p}.

\leavevmode\hypertarget{ref-78bpCEHk}{}%
(9) Fujieda, N.; Schätti, J.; Stuttfeld, E.; Ohkubo, K.; Maier, T.; Fukuzumi, S.; Ward, T. R. Enzyme Repurposing of a Hydrolase as an Emergent Peroxidase Upon Metal Binding. \emph{Chem. Sci.} \textbf{2015}, \emph{6} (7), 4060--4065. \url{https://doi.org/10.1039/c5sc01065a}.

\leavevmode\hypertarget{ref-ROLANnJl}{}%
(10) Brodin, J. D.; Ambroggio, X. I.; Tang, C.; Parent, K. N.; Baker, T. S.; Tezcan, F. A. Metal-Directed, Chemically Tunable Assembly of One-, Two- and Three-Dimensional Crystalline Protein Arrays. \emph{Nature Chem} \textbf{2012}, \emph{4} (5), 375--382. \url{https://doi.org/10.1038/nchem.1290}.

\leavevmode\hypertarget{ref-Lj2sbjzc}{}%
(11) Der, B. S.; Machius, M.; Miley, M. J.; Mills, J. L.; Szyperski, T.; Kuhlman, B. Metal-Mediated Affinity and Orientation Specificity in a Computationally Designed Protein Homodimer. \emph{J. Am. Chem. Soc.} \textbf{2011}, \emph{134} (1), 375--385. \url{https://doi.org/10.1021/ja208015j}.

\leavevmode\hypertarget{ref-17JMFzNJo}{}%
(12) Salgado, E. N.; Radford, R. J.; Tezcan, F. A. Metal-Directed Protein Self-Assembly. \emph{Acc. Chem. Res.} \textbf{2010}, \emph{43} (5), 661--672. \url{https://doi.org/10.1021/ar900273t}.

\leavevmode\hypertarget{ref-io987fKE}{}%
(13) Kakkis, A.; Gagnon, D.; Esselborn, J.; Britt, R. D.; Tezcan, F. A. Metal‐Templated Design of Chemically Switchable Protein Assemblies with High‐Affinity Coordination Sites. \emph{Angew. Chem. Int. Ed.} \textbf{2020}, \emph{59} (49), 21940--21944. \url{https://doi.org/10.1002/anie.202009226}.

\leavevmode\hypertarget{ref-JXKFLH3I}{}%
(14) Zastrow, M. L.; Peacock, A. F. A.; Stuckey, J. A.; Pecoraro, V. L. Hydrolytic Catalysis and Structural Stabilization in a Designed Metalloprotein. \emph{Nature Chem} \textbf{2011}, \emph{4} (2), 118--123. \url{https://doi.org/10.1038/nchem.1201}.

\leavevmode\hypertarget{ref-1FR3e6DyA}{}%
(15) Song, L. F.; Sengupta, A.; Merz, K. M., Jr. Thermodynamics of Transition Metal Ion Binding to Proteins. \emph{J. Am. Chem. Soc.} \textbf{2020}, \emph{142} (13), 6365--6374. \url{https://doi.org/10.1021/jacs.0c01329}.

\leavevmode\hypertarget{ref-1uPiiMnD}{}%
(16) Alford, R. F.; Leaver-Fay, A.; Jeliazkov, J. R.; O'Meara, M. J.; DiMaio, F. P.; Park, H.; Shapovalov, M. V.; Renfrew, P. D.; Mulligan, V. K.; Kappel, K.; Labonte, J. W.; Pacella, M. S.; Bonneau, R.; Bradley, P.; Dunbrack, R. L., Jr.; Das, R.; Baker, D.; Kuhlman, B.; Kortemme, T.; Gray, J. J. The Rosetta All-Atom Energy Function for Macromolecular Modeling and Design. \emph{J. Chem. Theory Comput.} \textbf{2017}, \emph{13} (6), 3031--3048. \url{https://doi.org/10.1021/acs.jctc.7b00125}.

\leavevmode\hypertarget{ref-5Vxh8hay}{}%
(17) Brunk, E.; Rothlisberger, U. Mixed Quantum Mechanical/Molecular Mechanical Molecular Dynamics Simulations of Biological Systems in Ground and Electronically Excited States. \emph{Chem. Rev.} \textbf{2015}, \emph{115} (12), 6217--6263. \url{https://doi.org/10.1021/cr500628b}.

\leavevmode\hypertarget{ref-IXq7vmo3}{}%
(18) Yang, Z.; Twidale, R. M.; Gervasoni, S.; Suardíaz, R.; Colenso, C. K.; Lang, E. J. M.; Spencer, J.; Mulholland, A. J. Multiscale Workflow for Modeling Ligand Complexes of Zinc Metalloproteins. \emph{J. Chem. Inf. Model.} \textbf{2021}, \emph{61} (11), 5658--5672. \url{https://doi.org/10.1021/acs.jcim.1c01109}.

\leavevmode\hypertarget{ref-nlfa683y}{}%
(19) Bozkurt, E.; Perez, M. A. S.; Hovius, R.; Browning, N. J.; Rothlisberger, U. Genetic Algorithm Based Design and Experimental Characterization of a Highly Thermostable Metalloprotein. \emph{J. Am. Chem. Soc.} \textbf{2018}, \emph{140} (13), 4517--4521. \url{https://doi.org/10.1021/jacs.7b10660}.

\leavevmode\hypertarget{ref-b5EJeHX3}{}%
(20). \url{https://doi.org/10.3389/fchem.2021.692200}.

\leavevmode\hypertarget{ref-qNdK0F1w}{}%
(21) Passerini, A.; Andreini, C.; Menchetti, S.; Rosato, A.; Frasconi, P. Predicting Zinc Binding at the Proteome Level. \emph{BMC Bioinformatics} \textbf{2007}, \emph{8} (1). \url{https://doi.org/10.1186/1471-2105-8-39}.

\leavevmode\hypertarget{ref-KNmKWbrP}{}%
(22) Hu, X.; Dong, Q.; Yang, J.; Zhang, Y. Recognizing Metal and Acid Radical Ion-Binding Sites by Integratingab Initiomodeling with Template-Based Transferals. \emph{Bioinformatics} \textbf{2016}, \emph{32} (21), 3260--3269. \url{https://doi.org/10.1093/bioinformatics/btw396}.

\leavevmode\hypertarget{ref-1HMhB3vxM}{}%
(23) Lin, Y.-F.; Cheng, C.-W.; Shih, C.-S.; Hwang, J.-K.; Yu, C.-S.; Lu, C.-H. MIB: Metal Ion-Binding Site Prediction and Docking Server. \emph{J. Chem. Inf. Model.} \textbf{2016}, \emph{56} (12), 2287--2291. \url{https://doi.org/10.1021/acs.jcim.6b00407}.

\leavevmode\hypertarget{ref-13fLqNwbD}{}%
(24) Hekkelman, M. L.; de Vries, I.; Joosten, R. P.; Perrakis, A. AlphaFill: Enriching the AlphaFold Models with Ligands and Co-Factors, 2021. \url{https://doi.org/10.1101/2021.11.26.470110}.

\leavevmode\hypertarget{ref-W4f7UG1p}{}%
(25) Brylinski, M.; Skolnick, J. FINDSITE-Metal: Integrating Evolutionary Information and Machine Learning for Structure-Based Metal-Binding Site Prediction at the Proteome Level. \emph{Proteins} \textbf{2010}, \emph{79} (3), 735--751. \url{https://doi.org/10.1002/prot.22913}.

\leavevmode\hypertarget{ref-iHxzzTCG}{}%
(26) Sánchez-Aparicio, J.-E.; Tiessler-Sala, L.; Velasco-Carneros, L.; Roldán-Martín, L.; Sciortino, G.; Maréchal, J.-D. BioMetAll: Identifying Metal-Binding Sites in Proteins from Backbone Preorganization. \emph{J. Chem. Inf. Model.} \textbf{2020}, \emph{61} (1), 311--323. \url{https://doi.org/10.1021/acs.jcim.0c00827}.

\leavevmode\hypertarget{ref-13sNVOrHB}{}%
(27) Haberal, İ.; Oğul, H. Prediction of Protein Metal Binding Sites Using Deep Neural Networks. \emph{Mol. Inf.} \textbf{2019}, \emph{38} (7), 1800169. \url{https://doi.org/10.1002/minf.201800169}.

\leavevmode\hypertarget{ref-yZfcMIwh}{}%
(28) Jumper, J.; Evans, R.; Pritzel, A.; Green, T.; Figurnov, M.; Ronneberger, O.; Tunyasuvunakool, K.; Bates, R.; Žídek, A.; Potapenko, A.; Bridgland, A.; Meyer, C.; Kohl, S. A. A.; Ballard, A. J.; Cowie, A.; Romera-Paredes, B.; Nikolov, S.; Jain, R.; Adler, J.; Back, T.; Petersen, S.; Reiman, D.; Clancy, E.; Zielinski, M.; Steinegger, M.; Pacholska, M.; Berghammer, T.; Bodenstein, S.; Silver, D.; Vinyals, O.; Senior, A. W.; Kavukcuoglu, K.; Kohli, P.; Hassabis, D. Highly Accurate Protein Structure Prediction with AlphaFold. \emph{Nature} \textbf{2021}, \emph{596} (7873), 583--589. \url{https://doi.org/10.1038/s41586-021-03819-2}.

\leavevmode\hypertarget{ref-rnu2o8C9}{}%
(29) Baek, M.; DiMaio, F.; Anishchenko, I.; Dauparas, J.; Ovchinnikov, S.; Lee, G. R.; Wang, J.; Cong, Q.; Kinch, L. N.; Schaeffer, R. D.; Millán, C.; Park, H.; Adams, C.; Glassman, C. R.; DeGiovanni, A.; Pereira, J. H.; Rodrigues, A. V.; van Dijk, A. A.; Ebrecht, A. C.; Opperman, D. J.; Sagmeister, T.; Buhlheller, C.; Pavkov-Keller, T.; Rathinaswamy, M. K.; Dalwadi, U.; Yip, C. K.; Burke, J. E.; Garcia, K. C.; Grishin, N. V.; Adams, P. D.; Read, R. J.; Baker, D. Accurate Prediction of Protein Structures and Interactions Using a Three-Track Neural Network. \emph{Science} \textbf{2021}, \emph{373} (6557), 871--876. \url{https://doi.org/10.1126/science.abj8754}.

\leavevmode\hypertarget{ref-ls5kyxZ3}{}%
(30) Torng, W.; Altman, R. B. 3D Deep Convolutional Neural Networks for Amino Acid Environment Similarity Analysis. \emph{BMC Bioinformatics} \textbf{2017}, \emph{18} (1). \url{https://doi.org/10.1186/s12859-017-1702-0}.

\leavevmode\hypertarget{ref-F7P74Etp}{}%
(31) Shroff, R.; Cole, A. W.; Diaz, D. J.; Morrow, B. R.; Donnell, I.; Annapareddy, A.; Gollihar, J.; Ellington, A. D.; Thyer, R. Discovery of Novel Gain-of-Function Mutations Guided by Structure-Based Deep Learning. \emph{ACS Synth. Biol.} \textbf{2020}, \emph{9} (11), 2927--2935. \url{https://doi.org/10.1021/acssynbio.0c00345}.

\leavevmode\hypertarget{ref-6szq3cwi}{}%
(32) Anand, N.; Eguchi, R.; Mathews, I. I.; Perez, C. P.; Derry, A.; Altman, R. B.; Huang, P.-S. Protein Sequence Design with a Learned Potential. \emph{Nat Commun} \textbf{2022}, \emph{13} (1). \url{https://doi.org/10.1038/s41467-022-28313-9}.

\leavevmode\hypertarget{ref-SGggOw7Z}{}%
(33) Torng, W.; Altman, R. B. High Precision Protein Functional Site Detection Using 3D Convolutional Neural Networks. \emph{Bioinformatics} \textbf{2018}, \emph{35} (9), 1503--1512. \url{https://doi.org/10.1093/bioinformatics/bty813}.

\leavevmode\hypertarget{ref-MKP16DSu}{}%
(34) Feehan, R.; Franklin, M. W.; Slusky, J. S. G. Machine Learning Differentiates Enzymatic and Non-Enzymatic Metals in Proteins. \emph{Nat Commun} \textbf{2021}, \emph{12} (1). \url{https://doi.org/10.1038/s41467-021-24070-3}.

\leavevmode\hypertarget{ref-q224Kv8w}{}%
(35) Renaud, N.; Geng, C.; Georgievska, S.; Ambrosetti, F.; Ridder, L.; Marzella, D. F.; Réau, M. F.; Bonvin, A. M. J. J.; Xue, L. C. DeepRank: A Deep Learning Framework for Data Mining 3D Protein-Protein Interfaces. \emph{Nat Commun} \textbf{2021}, \emph{12} (1). \url{https://doi.org/10.1038/s41467-021-27396-0}.

\leavevmode\hypertarget{ref-yBhqGkBa}{}%
(36) Gainza, P.; Sverrisson, F.; Monti, F.; Rodolà, E.; Boscaini, D.; Bronstein, M. M.; Correia, B. E. Deciphering Interaction Fingerprints from Protein Molecular Surfaces Using Geometric Deep Learning. \emph{Nat Methods} \textbf{2019}, \emph{17} (2), 184--192. \url{https://doi.org/10.1038/s41592-019-0666-6}.

\leavevmode\hypertarget{ref-Gq7n6vhH}{}%
(37) Jiménez, J.; Doerr, S.; Martínez-Rosell, G.; Rose, A. S.; De Fabritiis, G. DeepSite: Protein-Binding Site Predictor Using 3D-Convolutional Neural Networks. \emph{Bioinformatics} \textbf{2017}, \emph{33} (19), 3036--3042. \url{https://doi.org/10.1093/bioinformatics/btx350}.

\leavevmode\hypertarget{ref-mVGaXlum}{}%
(38) Skalic, M.; Varela-Rial, A.; Jiménez, J.; Martínez-Rosell, G.; De Fabritiis, G. LigVoxel: Inpainting Binding Pockets Using 3D-Convolutional Neural Networks. \emph{Bioinformatics} \textbf{2018}, \emph{35} (2), 243--250. \url{https://doi.org/10.1093/bioinformatics/bty583}.

\leavevmode\hypertarget{ref-13hjzBbje}{}%
(39) Stärk, H.; Ganea, O.-E.; Pattanaik, L.; Barzilay, R.; Jaakkola, T. EquiBind: Geometric Deep Learning for Drug Binding Structure Prediction. arXiv 2022. \url{https://doi.org/10.48550/arxiv.2202.05146}.

\leavevmode\hypertarget{ref-9UNjBvCL}{}%
(40) Park, S.; Seok, C. GalaxyWater-CNN: Prediction of Water Positions on the Protein Structure by a 3D-Convolutional Neural Network. \emph{J. Chem. Inf. Model.} \textbf{2022}, \emph{62} (13), 3157--3168. \url{https://doi.org/10.1021/acs.jcim.2c00306}.

\leavevmode\hypertarget{ref-9K88m3YT}{}%
(41) Li, B.; Yang, Y. T.; Capra, J. A.; Gerstein, M. B. Predicting Changes in Protein Thermodynamic Stability Upon Point Mutation with Deep 3D Convolutional Neural Networks. \emph{PLoS Comput Biol} \textbf{2020}, \emph{16} (11), e1008291. \url{https://doi.org/10.1371/journal.pcbi.1008291}.

\leavevmode\hypertarget{ref-k859wJxx}{}%
(42) Lu, C.-H.; Lin, Y.-F.; Lin, J.-J.; Yu, C.-S. Prediction of Metal Ion--Binding Sites in Proteins Using the Fragment Transformation Method. \emph{PLoS ONE} \textbf{2012}, \emph{7} (6), e39252. \url{https://doi.org/10.1371/journal.pone.0039252}.

\leavevmode\hypertarget{ref-Wt0ducot}{}%
(43) Laitaoja, M.; Valjakka, J.; Jänis, J. Zinc Coordination Spheres in Protein Structures. \emph{Inorg. Chem.} \textbf{2013}, \emph{52} (19), 10983--10991. \url{https://doi.org/10.1021/ic401072d}.

\leavevmode\hypertarget{ref-Qubi5dZD}{}%
(44) Minasov, G.; Vorontsov, I. I.; Shuvalova, L.; Brunzelle, J. S.; Kiryukhina, O.; Collart, F. R.; Joachimiak, A.; Anderson, W. F.;. Crystal Structure of Unknown Conserved ybaA Protein from Shigella Flexneri, 2007. \url{https://doi.org/10.2210/pdb2okq/pdb}.

\leavevmode\hypertarget{ref-g7lHXZp5}{}%
(45) Itoh, T.; Nakagawa, E.; Yoda, M.; Nakaichi, A.; Hibi, T.; Kimoto, H. Crystal Structure of Alginate Lyase from Paenibacillus Sp. Str. FPU-7, 2019. \url{https://doi.org/10.2210/pdb6kfn/pdb}.

\leavevmode\hypertarget{ref-w9ECLZDp}{}%
(46) Adams, C. M.; Eckenroth, B. E.; Doublie, S. Structure of the Clostridium Perfringens CspB Protease, 2013. \url{https://doi.org/10.2210/pdb4i0w/pdb}.

\leavevmode\hypertarget{ref-wW700ShK}{}%
(47) Davies, C. W.; Das, C. The Crystal Structure of a E280A Mutant of the Catalytic Domain of AMSH, 2011. \url{https://doi.org/10.2210/pdb3rzv/pdb}.

\leavevmode\hypertarget{ref-RFwEiPTW}{}%
(48) Rana, M. S.; Kumar, P.; Lee, C.-J.; Verardi, R.; Rajashankar, K. R.; Banerjee, A. Fatty Acyl Recognition and Transfer by an Integral Membrane \emph{S} -Acyltransferase. \emph{Science} \textbf{2018}, \emph{359} (6372). \url{https://doi.org/10.1126/science.aao6326}.

\leavevmode\hypertarget{ref-jyFnBdWm}{}%
(49) Hakansson, K.; Carlsson, M.; Svensson, L. A.; Liljas, A. STRUCTURE OF NATIVE AND APO CARBONIC ANHYDRASE II AND SOME OF ITS ANION-LIGAND COMPLEXES, 1993. \url{https://doi.org/10.2210/pdb2cba/pdb}.

\leavevmode\hypertarget{ref-13XuOF3Jj}{}%
(50) Hunt, J. B.; Neece, S. H.; Ginsburg, A. The Use of 4-(2-Pyridylazo)resorcinol in Studies of Zinc Release from Escherichia Coli Aspartate Transcarbamoylase. \emph{Anal Biochem} \textbf{1985}, \emph{146} (1), 150--157. \url{https://doi.org/10.1016/0003-2697(85)90409-9}.

\leavevmode\hypertarget{ref-pchdLF0k}{}%
(51) Pang, Y. P.; Xu, K.; Yazal, J. E.; Prendergas, F. G. Successful Molecular Dynamics Simulation of the Zinc-Bound Farnesyltransferase Using the Cationic Dummy Atom Approach. \emph{Protein Sci} \textbf{2000}, \emph{9} (10), 1857--1865.

\leavevmode\hypertarget{ref-D3VQQdjb}{}%
(52) Kiefer, L. L.; Fierke, C. A. Functional Characterization of Human Carbonic Anhydrase II Variants with Altered Zinc Binding Sites. \emph{Biochemistry} \textbf{1994}, \emph{33} (51), 15233--15240. \url{https://doi.org/10.1021/bi00255a003}.

\leavevmode\hypertarget{ref-w3tzhp92}{}%
(53) Kiefer, L. L.; Ippolito, J. A.; Fierke, C. A.; Christianson, D. W. Redesigning the Zinc Binding Site of Human Carbonic Anhydrase II: Structure of a His2Asp-Zn2+ Metal Coordination Polyhedron. \emph{J. Am. Chem. Soc.} \textbf{1993}, \emph{115} (26), 12581--12582. \url{https://doi.org/10.1021/ja00079a046}.

\leavevmode\hypertarget{ref-iu33cNMg}{}%
(54) Ippolito, J. A.; Christianson, D. W. Structure of an Engineered His3 Cys Zinc Binding Site in Human Carbonic Anhydrase II. \emph{Biochemistry} \textbf{1993}, \emph{32} (38), 9901--9905. \url{https://doi.org/10.1021/bi00089a005}.

\leavevmode\hypertarget{ref-148Si7kTL}{}%
(55) Ippolito, J. A.; Baird, T. T., Jr; McGee, S. A.; Christianson, D. W.; Fierke, C. A. Structure-Assisted Redesign of a Protein-Zinc-Binding Site with Femtomolar Affinity. \emph{Proc. Natl. Acad. Sci. U.S.A.} \textbf{1995}, \emph{92} (11), 5017--5021. \url{https://doi.org/10.1073/pnas.92.11.5017}.

\leavevmode\hypertarget{ref-TNbPH3Y1}{}%
(56) Huang, C.-c.; Lesburg, C. A.; Kiefer, L. L.; Fierke, C. A.; Christianson, D. W. Reversal of the Hydrogen Bond to Zinc Ligand Histidine-119 Dramatically Diminishes Catalysis and Enhances Metal Equilibration Kinetics in Carbonic Anhydrase II. \emph{Biochemistry} \textbf{1996}, \emph{35} (11), 3439--3446. \url{https://doi.org/10.1021/bi9526692}.

\leavevmode\hypertarget{ref-NifwNdQd}{}%
(57) Savage, H.; Wlodawer, A. Determination of Water Structure Around Biomolecules Using X-Ray and Neutron Diffraction Methods. \emph{Methods Enzymol} \textbf{1986}, \emph{127}, 162--183. \url{https://doi.org/10.1016/0076-6879(86)27014-7}.

\leavevmode\hypertarget{ref-10C4imKQR}{}%
(58) Morozenko, A.; Stuchebrukhov, A. A. Dowser++, a New Method of Hydrating Protein Structures. \emph{Proteins} \textbf{2016}, \emph{84} (10), 1347--1357. \url{https://doi.org/10.1002/prot.25081}.

\leavevmode\hypertarget{ref-izkvZoLX}{}%
(59) Sridhar, A.; Ross, G. A.; Biggin, P. C. Waterdock 2.0: Water Placement Prediction for Holo-Structures with a Pymol Plugin. \emph{PLoS ONE} \textbf{2017}, \emph{12} (2), e0172743. \url{https://doi.org/10.1371/journal.pone.0172743}.

\leavevmode\hypertarget{ref-Fc42fl1x}{}%
(60) Satorras, V. G.; Hoogeboom, E.; Welling, M. E(n) Equivariant Graph Neural Networks. arXiv 2021. \url{https://doi.org/10.48550/arxiv.2102.09844}.

\leavevmode\hypertarget{ref-cbwvbkS6}{}%
(61) Gligorijević, V.; Berenberg, D.; Ra, S.; Watkins, A.; Kelow, S.; Cho, K.; Bonneau, R. Function-Guided Protein Design by Deep Manifold Sampling, 2021. \url{https://doi.org/10.1101/2021.12.22.473759}.

\leavevmode\hypertarget{ref-Gj0pG9O5}{}%
(62) Greener, J. G.; Moffat, L.; Jones, D. T. Design of Metalloproteins and Novel Protein Folds Using Variational Autoencoders. \emph{Sci Rep} \textbf{2018}, \emph{8} (1). \url{https://doi.org/10.1038/s41598-018-34533-1}.

\leavevmode\hypertarget{ref-TlDhIlX}{}%
(63) Song, H.; Wilson, D. L.; Farquhar, E. R.; Lewis, E. A.; Emerson, J. P. Revisiting Zinc Coordination in Human Carbonic Anhydrase II. \emph{Inorg. Chem.} \textbf{2012}, \emph{51} (20), 11098--11105. \url{https://doi.org/10.1021/ic301645j}.

\leavevmode\hypertarget{ref-RJ44nj3R}{}%
(64) Handel, T. M.; Williams, S. A.; DeGrado, W. F. Metal Ion-Dependent Modulation of the Dynamics of a Designed Protein. \emph{Science} \textbf{1993}, \emph{261} (5123), 879--885. \url{https://doi.org/10.1126/science.8346440}.

\leavevmode\hypertarget{ref-15drt0nXZ}{}%
(65) Arnold, F. H.; Haymore, B. L. Engineered Metal-Binding Proteins: Purification to Protein Folding. \emph{Science} \textbf{1991}, \emph{252} (5014), 1796--1797. \url{https://doi.org/10.1126/science.1648261}.

\leavevmode\hypertarget{ref-85PeY3aG}{}%
(66) Krantz, B. A.; Sosnick, T. R. \emph{Nat. Struct Biol.} \textbf{2001}, \emph{8} (12), 1042--1047. \url{https://doi.org/10.1038/nsb723}.

\leavevmode\hypertarget{ref-10bYywzs6}{}%
(67) Tunyasuvunakool, K.; Adler, J.; Wu, Z.; Green, T.; Zielinski, M.; Žídek, A.; Bridgland, A.; Cowie, A.; Meyer, C.; Laydon, A.; Velankar, S.; Kleywegt, G. J.; Bateman, A.; Evans, R.; Pritzel, A.; Figurnov, M.; Ronneberger, O.; Bates, R.; Kohl, S. A. A.; Potapenko, A.; Ballard, A. J.; Romera-Paredes, B.; Nikolov, S.; Jain, R.; Clancy, E.; Reiman, D.; Petersen, S.; Senior, A. W.; Kavukcuoglu, K.; Birney, E.; Kohli, P.; Jumper, J.; Hassabis, D. Highly Accurate Protein Structure Prediction for the Human Proteome. \emph{Nature} \textbf{2021}, \emph{596} (7873), 590--596. \url{https://doi.org/10.1038/s41586-021-03828-1}.

\leavevmode\hypertarget{ref-8cYXwVxI}{}%
(68) Berman, H. M. The Protein Data Bank. \emph{Nucleic Acids Research} \textbf{2000}, \emph{28} (1), 235--242. \url{https://doi.org/10.1093/nar/28.1.235}.

\leavevmode\hypertarget{ref-jj6qOofE}{}%
(69) Steinegger, M.; Söding, J. MMseqs2 Enables Sensitive Protein Sequence Searching for the Analysis of Massive Data Sets. \emph{Nat Biotechnol} \textbf{2017}, \emph{35} (11), 1026--1028. \url{https://doi.org/10.1038/nbt.3988}.

\leavevmode\hypertarget{ref-FB5cckAJ}{}%
(70) Barber-Zucker, S.; Shaanan, B.; Zarivach, R. Transition Metal Binding Selectivity in Proteins and Its Correlation with the Phylogenomic Classification of the Cation Diffusion Facilitator Protein Family. \emph{Sci Rep} \textbf{2017}, \emph{7} (1). \url{https://doi.org/10.1038/s41598-017-16777-5}.

\leavevmode\hypertarget{ref-CcRTAd7h}{}%
(71) Raschka, S. BioPandas: Working with Molecular Structures in Pandas DataFrames. \emph{JOSS} \textbf{2017}, \emph{2} (14), 279. \url{https://doi.org/10.21105/joss.00279}.

\leavevmode\hypertarget{ref-q2gxO5bP}{}%
(72) Doerr, S.; Harvey, M. J.; Noé, F.; De Fabritiis, G. HTMD: High-Throughput Molecular Dynamics for Molecular Discovery. \emph{J. Chem. Theory Comput.} \textbf{2016}, \emph{12} (4), 1845--1852. \url{https://doi.org/10.1021/acs.jctc.6b00049}.

\leavevmode\hypertarget{ref-19MS2ZVR1}{}%
(73) Moritz, P.; Nishihara, R.; Wang, S.; Tumanov, A.; Liaw, R.; Liang, E.; Elibol, M.; Yang, Z.; Paul, W.; Jordan, M. I.; Stoica, I. Ray: A Distributed Framework for Emerging AI Applications. arXiv 2017. \url{https://doi.org/10.48550/arxiv.1712.05889}.

\leavevmode\hypertarget{ref-iOWq2JB8}{}%
(74) Paszke, A.; Gross, S.; Massa, F.; Lerer, A.; Bradbury, J.; Chanan, G.; Killeen, T.; Lin, Z.; Gimelshein, N.; Antiga, L.; Desmaison, A.; Köpf, A.; Yang, E.; DeVito, Z.; Raison, M.; Tejani, A.; Chilamkurthy, S.; Steiner, B.; Fang, L.; Bai, J.; Chintala, S. PyTorch: An Imperative Style, High-Performance Deep Learning Library. arXiv 2019. \url{https://doi.org/10.48550/arxiv.1912.01703}.

\leavevmode\hypertarget{ref-hchMCnC1}{}%
(75) de Boer, P.-T.; Kroese, D. P.; Mannor, S.; Rubinstein, R. Y. A Tutorial on the Cross-Entropy Method. \emph{Ann Oper Res} \textbf{2005}, \emph{134} (1), 19--67. \url{https://doi.org/10.1007/s10479-005-5724-z}.

\leavevmode\hypertarget{ref-8Miti2Gz}{}%
(76) Virtanen, P.; Gommers, R.; Oliphant, T. E.; Haberland, M.; Reddy, T.; Cournapeau, D.; Burovski, E.; Peterson, P.; Weckesser, W.; Bright, J.; van der Walt, S. J.; Brett, M.; Wilson, J.; Millman, K. J.; Mayorov, N.; Nelson, A. R. J.; Jones, E.; Kern, R.; Larson, E.; Carey, C. J.; Polat, İ.; Feng, Y.; Moore, E. W.; VanderPlas, J.; Laxalde, D.; Perktold, J.; Cimrman, R.; Henriksen, I.; Quintero, E. A.; Harris, C. R.; Archibald, A. M.; Ribeiro, A. H.; Pedregosa, F.; van Mulbregt, P.; Vijaykumar, A.; Bardelli, A. P.; Rothberg, A.; Hilboll, A.; Kloeckner, A.; Scopatz, A.; Lee, A.; Rokem, A.; Woods, C. N.; Fulton, C.; Masson, C.; Häggström, C.; Fitzgerald, C.; Nicholson, D. A.; Hagen, D. R.; Pasechnik, D. V.; Olivetti, E.; Martin, E.; Wieser, E.; Silva, F.; Lenders, F.; Wilhelm, F.; Young, G.; Price, G. A.; Ingold, G.-L.; Allen, G. E.; Lee, G. R.; Audren, H.; Probst, I.; Dietrich, J. P.; Silterra, J.; Webber, J. T.; Slavič, J.; Nothman, J.; Buchner, J.; Kulick, J.; Schönberger, J. L.; de Miranda Cardoso, J. V.; Reimer, J.; Harrington, J.; Rodríguez, J. L. C.; Nunez-Iglesias, J.; Kuczynski, J.; Tritz, K.; Thoma, M.; Newville, M.; Kümmerer, M.; Bolingbroke, M.; Tartre, M.; Pak, M.; Smith, N. J.; Nowaczyk, N.; Shebanov, N.; Pavlyk, O.; Brodtkorb, P. A.; Lee, P.; McGibbon, R. T.; Feldbauer, R.; Lewis, S.; Tygier, S.; Sievert, S.; Vigna, S.; Peterson, S.; More, S.; Pudlik, T.; Oshima, T.; Pingel, T. J.; Robitaille, T. P.; Spura, T.; Jones, T. R.; Cera, T.; Leslie, T.; Zito, T.; Krauss, T.; Upadhyay, U.; Halchenko, Y. O.; Vázquez-Baeza, Y.;. SciPy 1.0: Fundamental Algorithms for Scientific Computing in Python. \emph{Nat Methods} \textbf{2020}, \emph{17} (3), 261--272. \url{https://doi.org/10.1038/s41592-019-0686-2}.

\leavevmode\hypertarget{ref-AujvwLp6}{}%
(77) Pedregosa, F.; Varoquaux, G.; Gramfort, A.; Michel, V.; Thirion, B.; Grisel, O.; Blondel, M.; Müller, A.; Nothman, J.; Louppe, G.; Prettenhofer, P.; Weiss, R.; Dubourg, V.; Vanderplas, J.; Passos, A.; Cournapeau, D.; Brucher, M.; Perrot, M.; Duchesnay, É. Scikit-Learn: Machine Learning in Python. \emph{arXiv} \textbf{2012}. \url{https://doi.org/10.48550/arxiv.1201.0490}.

\leavevmode\hypertarget{ref-PEXAwV3k}{}%
(78) Humphrey, W.; Dalke, A.; Schulten, K. VMD: Visual Molecular Dynamics. \emph{J Mol Graph} \textbf{1996}, \emph{14} (1), 33--38, 27--28. \url{https://doi.org/10.1016/0263-7855(96)00018-5}.

\leavevmode\hypertarget{ref-BguJBlg6}{}%
(79) Rego, N.; Koes, D. 3Dmol.js: Molecular Visualization with WebGL. \emph{Bioinformatics} \textbf{2014}, \emph{31} (8), 1322--1324. \url{https://doi.org/10.1093/bioinformatics/btu829}.

\leavevmode\hypertarget{ref-WjHESl20}{}%
(80) Håkansson, K.; Carlsson, M.; Svensson, L.; Liljas, A. Structure of Native and Apo Carbonic Anhydrase II and Structure of Some of Its Anion-Ligand Complexes. \emph{Journal of Molecular Biology} \textbf{1992}, \emph{227} (4), 1192--1204. \url{https://doi.org/10.1016/0022-2836(92)90531-n}.

\leavevmode\hypertarget{ref-Mg40Hdad}{}%
(81) Lovell, S. C.; Word, J. M.; Richardson, J. S.; Richardson, D. C. The Penultimate Rotamer Library. \emph{Proteins} \textbf{2000}, \emph{40} (3), 389--408.
\end{cslreferences}



\end{document}
