<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">

<head>
    <meta charset="utf-8" />
    <meta name="generator" content="pandoc" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
    <meta name="author" content="Simon L. Dürr" />
    <meta name="author" content="Andrea Levy" />
    <meta name="author" content="Ursula Rothlisberger" />

    <meta name="dcterms.date" content="2022-08-22" />
    <meta name="keywords" content="metal, protein engineering, deep learning" />
    <title>Accurate prediction of transition metal ion location via deep learning</title>
    <style>
        code {
            white-space: pre-wrap;
        }

        span.smallcaps {
            font-variant: small-caps;
        }

        span.underline {
            text-decoration: underline;
        }

        div.column {
            display: inline-block;
            vertical-align: top;
            width: 50%;
        }

        div.hanging-indent {
            margin-left: 1.5em;
            text-indent: -1.5em;
        }

        ul.task-list {
            list-style: none;
        }
    </style>
    <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
    <link href="https://cdnjs.cloudflare.com/ajax/libs/tabulator/4.7.1/css/tabulator_simple.min.css" rel="stylesheet">
    <script type="text/javascript" src="https://unpkg.com/tabulator-tables@4.7.1/dist/js/tabulator.min.js"></script>
    <script type="text/javascript" src="https://oss.sheetjs.com/sheetjs/xlsx.full.min.js"></script>

    <script src="https://cdn.plot.ly/plotly-latest.min.js"></script>


    <style>
        @media print {
            .noprint {
                display: none;
            }
        }

        @media screen {
            .noscreen {
                display: none;
            }
        }
    </style>
</head>

<body>
    <header id="title-block-header">
        <h1 class="title">Accurate prediction of transition metal ion location via deep learning</h1>

    </header>
    <!--  -->
    <div style="text-align:center">
        <p><small><em>
                    This manuscript
                    was automatically generated

                    on August 22, 2022. <br>View <a href="doi">non-interactive version on BioRxiv</a>.
                </em></small></p>
    </div>
    <!--  -->
    <!-- 
+ **Simon L. Dürr**  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0002-4304-8106](https://orcid.org/0000-0002-4304-8106)
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [simonduerr](https://twitter.com/simonduerr)<br>
  <small>
     Laboratory of Computational Chemistry and Biochemistry, Institute of Chemical Sciences and Engineering, Swiss Federal Institute of Technology (EPFL) CH-1015 Lausanne, Switzerland
  </small>

+ **Andrea Levy**  <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0003-1255-859X](https://orcid.org/0000-0003-1255-859X)<br>
  <small>
     Laboratory of Computational Chemistry and Biochemistry, Institute of Chemical Sciences and Engineering, Swiss Federal Institute of Technology (EPFL) CH-1015 Lausanne, Switzerland
  </small>

+ **Ursula Rothlisberger** ✉ <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0002-1704-8591](https://orcid.org/0000-0002-1704-8591)<br>
  <small>
     Laboratory of Computational Chemistry and Biochemistry, Institute of Chemical Sciences and Engineering, Swiss Federal Institute of Technology (EPFL) CH-1015 Lausanne, Switzerland
  </small>
 -->
    <p><a href="https://orcid.org/0000-0002-4304-8106"><img src="images/orcid.svg" width="11" height="11"
                alt="ORCID icon" /></a>
        Simon L. Dürr <sup>1</sup>,
        <a href="https://orcid.org/0000-0003-1255-859X"><img src="images/orcid.svg" width="11" height="11"
                alt="ORCID icon" /></a>
        Andrea Levy <sup>1</sup>,
        <a href="https://orcid.org/0000-0002-1704-8591"><img src="images/orcid.svg" width="11" height="11"
                alt="ORCID icon" /></a>
        Ursula Rothlisberger <sup>1,✉</sup>
    </p>
    <!--<sup>☯</sup> --- These authors contributed equally. <br> -->
    <p><sup>✉</sup> — To whom correspondence should be addressed: ursula.roethlisberger@epfl.ch
        <small>
    </p>
    <h6 id="affiliations">Affiliations</h6>
    <ol type="1">
        <li>Laboratory of Computational Chemistry and Biochemistry, Institute of Chemical Sciences and Engineering,
            Swiss Federal Institute of Technology (EPFL) CH-1015 Lausanne, Switzerland</li>
    </ol>
    <h2 id="abstract">Abstract</h2>
    <p>Metal ions are essential cofactors for many proteins. In fact, currently, about half of the structurally
        characterized proteins contain a metal ion. Metal ions play a crucial role for many applications such as enzyme
        design or design of protein-protein interactions because they are biologically abundant, tether to the protein
        using strong interactions, and have favorable catalytic properties e.g. as Lewis acid. Computational design of
        metalloproteins is however hampered by the complex electronic structure of many biologically relevant metals
        such as zinc that can often not be accurately described using a classical force field. In this work, we develop
        two tools - Metal3D (based on 3D convolutional neural networks) and Metal1D (solely based on geometric criteria)
        to improve the identification and localization of zinc and other metal ions in experimental and computationally
        predicted protein structures. Comparison with other currently available tools shows that Metal3D is the most
        accurate metal ion location predictor to date outperforming geometric predictors including Metal1D by a wide
        margin using a single structure as input. Metal3D outputs a confidence metric for each predicted site and works
        on proteins with few homologes in the protein data bank. The predicted metal ion locations for Metal3D are
        within 0.70 ± 0.64 Å of the experimental locations with half of the sites below 0.5 Å. Metal3D predicts a global
        metal density that can be used for annotation of structures predicted using e.g. AlphaFold2 and a per residue
        metal density that can be used in protein design workflows for the location of suitable metal binding sites and
        rotamer sampling to create novel metalloproteins. Metal3D is available as easy to use webapp, notebook or
        commandline interface.</p>
    <h2 id="introduction">Introduction</h2>
    <p>Metalloproteins are ubiquitous in nature and are present in all major enzyme families.<span class="citation"
            data-cites="gxiv5uTU LY9UGI7"><sup><a href="#ref-gxiv5uTU" role="doc-biblioref">1</a>,<a href="#ref-LY9UGI7"
                    role="doc-biblioref">2</a></sup></span>The metals predominantly found in biological systems are the
        first and second row alkali and earth alkali metals and the first row transition metals such as zinc and copper.
        Zinc is the most common transition metal (present in ~10% of deposited structures) and can fulfill both a
        structural (e.g. in zinc finger proteins) or a catalytic role in up to trinuclear active sites. Zn<sup>2+</sup>
        is an excellent Lewis acid and is most often found in tetrahedral, pentavalent, or octahedral coordination.
        About 10 % of all reactions catalyzed by enzymes use zinc as cofactor<span class="citation"
            data-cites="iqbHG3rv"><sup><a href="#ref-iqbHG3rv" role="doc-biblioref">3</a></sup></span>.</p>
    <p>Metalloproteins are well studied because metal cofactors are essential for the function of many proteins and loss
        of this function is an important cause of diseases.<span class="citation" data-cites="4sXp2sDw"><sup><a
                    href="#ref-4sXp2sDw" role="doc-biblioref">4</a></sup></span> Industrial applications for
        metalloproteins capitalize on the favorable catalytic properties of the metal ion where the protein environment
        dictates (stereo)-selectivity.<span class="citation" data-cites="oOm5j05D LAjO1aDL 13w4SiWCx"><sup><a
                    href="#ref-oOm5j05D" role="doc-biblioref">5</a>–<a href="#ref-13w4SiWCx"
                    role="doc-biblioref">7</a></sup></span> To crystallize proteins, metal salts are also often added to
        the crystallization buffer as they can help in the formation of protein crystals overcoming the enthalpic cost
        of association of protein surfaces. Metal ion binding sites can be used to engineer protein-protein interactions
        (PPI)<span class="citation" data-cites="ROLANnJl Lj2sbjzc 17JMFzNJo"><sup><a href="#ref-ROLANnJl"
                    role="doc-biblioref">10</a>–<a href="#ref-17JMFzNJo" role="doc-biblioref">12</a></sup></span> and
        the hypothesis has been put forward that one origin of macromolecular complexity is the superficial binding of
        metal ions in early single domain proteins.<span class="citation" data-cites="17JMFzNJo"><sup><a
                    href="#ref-17JMFzNJo" role="doc-biblioref">12</a></sup></span></p>
    <p>While simple metal ion binding sites can be rapidly engineered because initial coordination on a protein surface
        can for example be achieved by creating an i, i+4 di-histidine site on an alpha-helix<span class="citation"
            data-cites="io987fKE"><sup><a href="#ref-io987fKE" role="doc-biblioref">13</a></sup></span> or by placing
        cysteines in spatial proximity,<span class="citation" data-cites="JXKFLH3I"><sup><a href="#ref-JXKFLH3I"
                    role="doc-biblioref">14</a></sup></span> the engineering of complex metal ion binding sites e.g. in
        the protein interior is considerably more difficult<span class="citation" data-cites="LY9UGI7 Lj2sbjzc"><sup><a
                    href="#ref-LY9UGI7" role="doc-biblioref">2</a>,<a href="#ref-Lj2sbjzc"
                    role="doc-biblioref">11</a></sup></span> as such sites are often supported by a network of hydrogen
        bonds. A complication for computational design of metalloproteins is the unavailability of good (non-bonded)
        force fields for zinc and other transition metals that accurately reproduce (e.g. tetrahedral) coordination with
        the correct coordination distances which renders design using e.g. Rosetta very difficult.<span class="citation"
            data-cites="LY9UGI7 1FR3e6DyA"><sup><a href="#ref-LY9UGI7" role="doc-biblioref">2</a>,<a
                    href="#ref-1FR3e6DyA" role="doc-biblioref">15</a></sup></span> In fact, the latest parametrization
        of the Rosetta energy function (ref2015)<span class="citation" data-cites="1uPiiMnD"><sup><a
                    href="#ref-1uPiiMnD" role="doc-biblioref">16</a></sup></span> did not refit the parameters for the
        metal ions which originally are from CHARMM27 with empirically derived Lazaridis-Karplus solvation terms. To
        adequately treat metal sites in proteins quantum mechanical treatments such as in hybrid quantum
        mechanics/molecular mechanics (QM/MM) simulations<span class="citation" data-cites="5Vxh8hay IXq7vmo3"><sup><a
                    href="#ref-5Vxh8hay" role="doc-biblioref">17</a>,<a href="#ref-IXq7vmo3"
                    role="doc-biblioref">18</a></sup></span> is needed whose computational cost is prohibitive for
        regular protein design tasks. QM/MM simulations can however be used to verify coordination chemistry for select
        candidate proteins.<span class="citation" data-cites="nlfa683y"><sup><a href="#ref-nlfa683y"
                    role="doc-biblioref">19</a></sup></span> On the other hand, neural network potentials have been
        developed for zinc however those require the experimental zinc location as input.<span class="citation"
            data-cites="b5EJeHX3"><sup><a href="#ref-b5EJeHX3" role="doc-biblioref">20</a></sup></span></p>
    <p>Many tools exist to predict whether a protein contains metals (e.g. ZincFinder<span class="citation"
            data-cites="qNdK0F1w"><sup><a href="#ref-qNdK0F1w" role="doc-biblioref">21</a></sup></span>), which residues
        in the protein bind a metal (e.g. IonCom,<span class="citation" data-cites="KNmKWbrP"><sup><a
                    href="#ref-KNmKWbrP" role="doc-biblioref">22</a></sup></span> MIB<span class="citation"
            data-cites="1HMhB3vxM"><sup><a href="#ref-1HMhB3vxM" role="doc-biblioref">23</a></sup></span>) and where the
        metal is bound (AlphaFill,<span class="citation" data-cites="13fLqNwbD"><sup><a href="#ref-13fLqNwbD"
                    role="doc-biblioref">24</a></sup></span> FindsiteMetal,<span class="citation"
            data-cites="W4f7UG1p"><sup><a href="#ref-W4f7UG1p" role="doc-biblioref">25</a></sup></span> BioMetAll,<span
            class="citation" data-cites="iHxzzTCG"><sup><a href="#ref-iHxzzTCG"
                    role="doc-biblioref">26</a></sup></span>MIB<span class="citation" data-cites="1HMhB3vxM"><sup><a
                    href="#ref-1HMhB3vxM" role="doc-biblioref">23</a></sup></span> ). The input for these predictors is
        based on sequence and/or structure information. Sequence-based predictors use pattern recognition to identify
        the amino acids which might bind a metal.<span class="citation" data-cites="13sNVOrHB"><sup><a
                    href="#ref-13sNVOrHB" role="doc-biblioref">27</a></sup></span>
        Structure-based methods use homology to known structures (MIB, Findsite-metal, AlphaFill) or distance features
        (BioMetAll) to infer the location of metals. Some tools like Findsite-metal or ZincFinder employ machine
        learning based approaches such as support vector machines.</p>
    <p>Structure based deep learning approaches have been used in the field of protein research for a variety of
        applications such as protein structure prediction,<span class="citation" data-cites="yZfcMIwh"><sup><a
                    href="#ref-yZfcMIwh" role="doc-biblioref">28</a></sup></span> prediction of identity of masked
        residues,<span class="citation" data-cites="ls5kyxZ3 F7P74Etp 6szq3cwi"><sup><a href="#ref-ls5kyxZ3"
                    role="doc-biblioref">30</a>–<a href="#ref-6szq3cwi" role="doc-biblioref">32</a></sup></span>
        functional site prediction,<span class="citation" data-cites="SGggOw7Z MKP16DSu"><sup><a href="#ref-SGggOw7Z"
                    role="doc-biblioref">33</a>,<a href="#ref-MKP16DSu" role="doc-biblioref">34</a></sup></span> for
        ranking of docking poses,<span class="citation" data-cites="q224Kv8w yBhqGkBa"><sup><a href="#ref-q224Kv8w"
                    role="doc-biblioref">35</a>,<a href="#ref-yBhqGkBa" role="doc-biblioref">36</a></sup></span>
        prediction of the location of ligands,<span class="citation"
            data-cites="Gq7n6vhH mVGaXlum 13hjzBbje yBhqGkBa 9UNjBvCL"><sup><a href="#ref-yBhqGkBa"
                    role="doc-biblioref">36</a>–<a href="#ref-9UNjBvCL" role="doc-biblioref">40</a></sup></span> and
        prediction of effects of mutations for stability and disease.<span class="citation"
            data-cites="4sXp2sDw"><sup><a href="#ref-4sXp2sDw" role="doc-biblioref">4</a></sup></span>
        Current state of the art predictors for metal location are MIB,<span class="citation"
            data-cites="k859wJxx 1HMhB3vxM"><sup><a href="#ref-1HMhB3vxM" role="doc-biblioref">23</a>,<a
                    href="#ref-k859wJxx" role="doc-biblioref">42</a></sup></span> which combines structural and sequence
        information in the “Fragment Transformation Method” to search for homologous sites in its database, and
        BioMetAll,<span class="citation" data-cites="iHxzzTCG"><sup><a href="#ref-iHxzzTCG"
                    role="doc-biblioref">26</a></sup></span> a geometrical predictor based on backbone preorganization.
        Both methods have significant drawbacks: MIB excludes metal sites with less than 2 coordination partners from
        its analysis and is limited by the availability of templates in its database. BioMetAll does not use templates
        but provides many possible locations for putative binding sites on a regular grid. The individual probes in
        BioMetAll do not have a confidence metric therefore only allowing to rank sites by the number of probes found,
        which results in a large uncertainty in the position. Both tools suffer from many false positives. In this work,
        we present two metal ion location predictors that do not suffer from these drawbacks. The deep learning based
        Metal3D predictor operates on a voxelized representation of the protein environment and predicts a per residue
        metal density that can be averaged to get a smooth metal probability density over the whole protein. The
        distance based predictor Metal1D predicts the location of metals using distances mined from the protein data
        bank (PDB) directly predicting coordinates of the putative metal binding site. These tools pave the way to
        perform in silico design of metal ion binding sites without relying on predefined geometrical rules or expensive
        quantum mechanical calculations.</p>
    <h2 id="results">Results</h2>
    <p>A dataset of experimental high resolution crystal structures (2085 structures/252324 voxelized environments)
        containing zinc sites was used for training of the geometric predictor Metal1D and the deep learning predictor
        Metal3D (Figure <a href="#fig:method">1</a>). For training, we used the crystal environment including crystal
        contacts. For predictions, the biological assembly was used.</p>
    <div id="fig:method" class="fignos">
        <figure>
            <img src="images/Metal3D_Metal1D_method.png" alt="" />
            <figcaption><span>Figure 1:</span> <strong>Workflow of Metal3D and Metal1D</strong> <strong>A</strong>
                Training of Metal3D and Metal1D is based on experimental Zn<sup>2+</sup> sites. Metal1D extracts
                coordination environments from LINK records, Metal3D is a fully convolutional 3DCNN trained to predict
                the metal density from voxelized protein environments. <strong>B</strong> In inference mode Metal3D
                predicts the location of a metal ion by computing per residue metal densities and then averaging them to
                obtain a global metal density for the input proteins. The ions can then be placed using the weighted
                average of voxels above a cutoff. For Metal1D all residues in the protein are scanned for compatibility
                with the probability map. Metals are placed at the geometric center of residues with high scores
                according to the probability map. A final ranking of sites is obtained using the probability map.
            </figcaption>
        </figure>
    </div>
    <h3 id="metal3d">Metal3D</h3>
    <p>Metal3D takes a protein structure and a set of residues as input, voxelizes the environment around each of the
        residues and predicts the per residue metal density. The predicted per residue densities (within a 16 x 16 x 16
        Å<sup>3</sup> volume) can then be averaged to yield a zinc density for the whole protein. At high probability
        cutoffs the predicted metal densities are spherical (Figure <a href="#fig:2cbaprobabilities">2</a> E), at low
        probability cutoffs the predicted densities are non-regular (Figure <a href="#fig:2cbaprobabilities">2</a> A).
    </p>
    <div id="fig:2cbaprobabilities" class="fignos">
        <figure>
            <img src="images/2CBA_probabilities.png" alt="" />
            <figcaption><span>Figure 2:</span> <strong>Metal3D probability density</strong> Probability evolution in
                HCA2 (PDB 2CBA) for different probability cutoffs A) p=0.1 B) p=0.2 C) p=0.3 D) p=0.4 E) p=0.75.
            </figcaption>
        </figure>
    </div>
    <p>We evaluated the quality of the metal densities generated by the model with the discretized Jaccard similarity
        (Figure <a href="#fig:jaccard">S1</a>) for all environments in the test set. We noticed that at the edges of the
        residue-centered output densities often spurious density is predicted wherefore we evaluated the similarity of
        the test set metal density and the predicted metal probability density taking into account a smaller box with
        zeroed outer edges. Figure <a href="#fig:jaccard">S1</a> shows that the similarity of the boxes does not depend
        much on the probability cutoff chosen with higher cutoffs yielding slightly higher discretized Jaccard
        similarity values (0.02 - 0.04 difference between p=0.5 and p=0.9). Reducing the size of the analyzed boxes (i.e
        trimming of the edges) increases the Jaccard similarity from ≈ 0.64 to 0.88 showing that the metal density in
        the center of the box is more accurate than the density at the edges.</p>
    <p>Metal3D is available as self-contained notebook on <a
            href="https://colab.research.google.com/github/lcbc-epfl/metal-site-prediction/blob/main/Metal3D/ColabMetal.ipynb">GoogleColab</a>
        and on <a href="https://hf.space/simonduerr/metal3d">Huggingface Spaces</a>.</p>
    <script type="module" src="https://gradio.s3-us-west-2.amazonaws.com/3.0.18/gradio.js"></script>
    <p>
        <gradio-app space="simonduerr/metal3d"></gradio-app>
    </p>
    <h3 id="metal1d">Metal1D</h3>
    <p>The statistical analysis for the geometric predictor uses the <code>LINK</code> records present in deposited PDB
        structures. A probability map for all zinc coordination motifs was extracted from all training structures
        (Figure <a href="#fig:method">1</a> A). The mean coordination distance in the training set was found to be 2.2 ±
        0.2 Å, and the default search radius for the predictions was therefore set to 5.5 Å (Table <a
            href="#tbl:metal1dreferencepoints">S1</a>). In total 208 different environments with more than 5 different
        proteins (at 30 % sequence identity) were identified.
        Metal1D is available as self-contained notebook on <a
            href="https://colab.research.google.com/github/lcbc-epfl/metal-site-prediction/blob/main/Metal1D/ColabMetal1D.ipynb">Google
            Colab</a>.</p>
    <h3 id="comparison-of-metal1d-metal3d-mib-and-biometall">Comparison of Metal1D, Metal3D, MIB and BioMetAll</h3>
    <p>Existing metal ion predictors can be subdivided into two categories: binding site predictors and binding location
        predictors. The former identify only the residues binding the ion, the latter predict the coordinates of the
        metal ion itself. Both Metal1D and Metal3D can predict the coordinates of putative binding sites. We therefore
        assessed their performance by comparing to recent binding location predictors with available code/webserver:
        BioMetAll<span class="citation" data-cites="iHxzzTCG"><sup><a href="#ref-iHxzzTCG"
                    role="doc-biblioref">26</a></sup></span> and MIB.<span class="citation"
            data-cites="1HMhB3vxM"><sup><a href="#ref-1HMhB3vxM" role="doc-biblioref">23</a></sup></span> The main
        tuning parameter of MIB is the template similarity <code>t</code>, with higher values requiring higher
        similarity of the templates available for the search in structurally homologous metalloproteins. BioMetAll on
        the other hand was calibrated on available protein structures and places probes on a regular grid at all sites
        where the criteria for metal binding are fulfilled. The main adjustable parameter for BioMetAll is the cluster
        cutoff <code>c</code>, which indicates how many probes in reference to the largest cluster a specific cluster
        has. We used the recommended cutoff of 0.5 requiring all chosen clusters to have at least 50% of the probes of
        the most populous cluster and used the cluster center to compute distances.</p>
    <p>We first investigated the potential of all tools to detect the location of a zinc ion binding site in a binary
        fashion (zinc site or no zinc site). We defined a correctly identified binding site (true positive, TP) as a
        prediction within 5 Å of an experimental zinc site. In case a tool predicted no metal within the 5 Å radius, we
        counted this site as false negative (FN). False positive (FP) predictions, i.e sites where a metal was placed
        spuriously, were clustered in a 5 Å radius and counted once per cluster. All tools were assessed against the
        held out test biological assemblies for Metal3D and Metal1D. When the performance of MIB (<code>t=1.25</code>)
        and BioMetAll is compared against Metal3D with probability cutoff p=0.75 we find that Metal3D identifies more
        sites (85) than MIB (78) or BioMetAll (75) with a much lower number of false positives (Figure <a
            href="#fig:comparison">3</a>).
        MIB predicts 180 false positive sites, BioMetAll 134 sites whereas Metal3D only predicts 9 false positive sites
        at the p=0.75 cutoff. Metal1D (<code>t=0.5</code>) offers similar detection capabilities (78 sites detected)
        with a lower number of false positives (47) compared to MIB and BioMetAll.
        We removed 56 sites from the list of zinc sites in the test set (189 total) that had less than 2 unique protein
        ligands within 2.8 Å of the experimental zinc location. The amount of correct predictions in this reduced set is
        almost unchanged for all tools (Figure <a href="#fig:comparison">3</a>) indicating that most tools correctly
        predict sites if they have 2 or more protein ligands. For Metal3D at p=0.75 and p=0.9 as well as MIB with t=1.9
        all sites that are correctly predicted to contain a metal are sites with more than 2 protein ligands. The number
        of false negatives is reduced for all tools by about 50 sites indicating that most tools do not predict these
        crystallographic artifacts that might depend on additional coordinating residues from an adjacent molecule in
        the crystal. Of all tools, Metal3D has the least false positives (1 FP at p=0.9) and the highest number of
        detected sites (110 at p=0.25). The single false positive at p=0.9 does not contain a zinc ion but is a calcium
        binding site with three aspartates and one backbone carbonyl ligand (Figure <a href="#fig:4jjjFP">S8</a>).</p>
    <div id="fig:comparison" class="fignos">
        <figure>
            <img src="images/all_2coord_metal_1d_3d_min_biometall_comparison.jpg" alt="" />
            <figcaption><span>Figure 3:</span> <strong>Identification of metal sites</strong> Comparison of Metal1D,
                Metal3D, BioMetAll and MIB on the test set held out from training of Metal1D and Metal3D. Predicted
                sites are counted as true positives (TP) if they are within 5 Å of a true metal location and as false
                negatives (FN) otherwise. False positive (FP) probes are clustered and counted once per cluster. *For
                MIB we used 2 structures less because the server did not accept these structures.</figcaption>
        </figure>
    </div>
    <p>After assessment of how many sites the tools predict, another crucial metric is the spatial precision of the
        predictions. For the correctly identified sites (TP) we measured the mean absolute distance (MAD) between
        experimental and predicted position (Figure <a href="#fig:selectivity-mad">4</a>). The MAD for Metal3D at p=0.9
        is 0.70 ± 0.64 Å and 0.73 ± 0.65 Å at p=0.25 indicating that low confidence predictions are still accurately
        placed inside the protein. The median MAD of predictions for Metal3D at p=0.9 is 0.52 Å indicating that for half
        of the predictions the model predicts at or better than the grid resolution of 0.5 Å.</p>

    <p>BioMetAll is not very precise with a MAD for correctly identified sites of 2.80 ± 1.30 Å. BioMetAll predicts many
        possible locations per cluster with some of them much closer to the experimental metal binding site than the
        cluster center. However, it does not provide any ranking of the probes within a cluster and therefore the
        cluster center was used for the distance calculation. Metal1D (MAD 2.06 ± 1.33 Å) which identifies more sites
        than BioMetAll is also more precise than BioMetAll. MIB t=1.9 detects sites with high precision (MAD 0.77 ± 1.09
        Å) but it relies on the existence of homologous sites to align the found sites.</p>

    <div id="fig:selectivity-mad" class="fignos">
        <figure>
            <img src="images/Figure3_selectivity_mad.png" alt="" />
            <figcaption><span>Figure 4:</span> <strong>Precision of predicted sites and selectivity for
                    Zn<sup>2+</sup></strong> <strong>A</strong> Mean absolute deviation (MAD) of predicted zinc ion
                locations using Metal1D, Metal3D, BioMetAll and MIB on the test set used to train Metal1D and Metal3D
                for all correctly identified (TP) sites. n is the number of sites predicted by the tool. *For MIB we
                used 2 structures less because the server did not accept these structures. For each tool the whisker
                plot indicates the median (white dot) and the first quartiles (black box). <strong>B</strong> Recall for
                the zinc test set and 25 randomly drawn structures for other transition, alkali and earth-alkali metal
                ions for Metal3D using p=0.5 as cutoff.</figcaption>
        </figure>
    </div>
    <h3 id="selectivity-for-other-metals">Selectivity for other metals</h3>
    <p>Both Metal3D and Metal1D were exclusively trained on zinc and we assessed their performance on sodium (Na+, PDB
        code NA), potassium (K+, PDB code K), calcium (Ca2+, PDB code CA), magnesium (Mg2+, PDB code MG), and various
        transition metals (Fe2+, Fe3+, Co2+, Cu2+, Cu+, Mn2+ with corresponding PDB codes FE2, FE, CO, CU, CU1, MN,
        respectively) from 25 randomly drawn structures from the clustered PDB at 30% identity. Only sites with at least
        3 unique protein ligands were used for the analysis to exclude crystallographic artifacts and use only highly
        defined sites which should exhibit most selectivity towards a specific metal. Figure <a
            href="#fig:selectivity-mad">4</a> B shows that recall for Metal3D is high for all transition metals, meaning
        that the model correctly finds most sites even though it was only trained on zinc. For the alkali and earth
        alkali metals recall is much lower as the model only finds some sites. The mean probability for found zinc
        structures (ZN p=0.95 ± 0.10) in the test set is higher than for the other transition metals (Figure <a
            href="#fig:selectivity-probability-metal3d">S6</a>) and significantly higher than the probability for alkali
        metals (NA p=0.61 ± 0.10, K p=0.79 ± 0.16) while the probability for the earth alkali metals is slightly higher
        with MG (p=0.77 ± 0.16) similar to CA (p=0.73 ± 0.16). The MAD for each found metal site is again lowest for
        zinc (0.56 ± 0.59 Å). The MAD for the found sodium (n=2) and potassium (n=5) sites are as low as for the other
        transition metals. The only metal with significantly higher MAD (1.45 ± 0.93 Å) is CU1 (Figure <a
            href="#fig:selectivity-distance-metal3d">S5</a>).</p>
    <p>The only two structures where a sodium is detected by Metal3D (2OKQ<span class="citation"
            data-cites="Qubi5dZD"><sup><a href="#ref-Qubi5dZD" role="doc-biblioref">43</a></sup></span>, 6KFN<span
            class="citation" data-cites="g7lHXZp5"><sup><a href="#ref-g7lHXZp5"
                    role="doc-biblioref">44</a></sup></span>) have at least 2 side chain coordinating ligand atoms and
        only one backbone (2OKQ) or no backbone ligand atom (6KFN). Canonical sodium binding sites e.g. such as in PDB
        4I0W<span class="citation" data-cites="w9ECLZDp"><sup><a href="#ref-w9ECLZDp"
                    role="doc-biblioref">45</a></sup></span> with two coordinating backbone carbonyl oxygen atoms and
        one asparagine side chain have probabilities around 5 % and are basically indistinguishable from background
        noise of the model.
        For Metal1D overall recall is lower with similar differences in the detection of main group metals versus
        transition metals (Figures <a href="#fig:selectivity-metal1d">S3</a> and <a
            href="#fig:selectivity-distance-metal1d">S4</a>).</p>
    <h3 id="applications">Applications</h3>
    <p>After having evaluated the accuracy of Metal3D on held out test structures we also investigated possible uses in
        downstream applications such as protein function annotation and protein design.</p>
    <h4 id="alpha-fold">Alpha Fold</h4>
    <p>AlphaFold2 often predicts side chains in metal ion binding sites in the holo conformation.<span class="citation"
            data-cites="yZfcMIwh"><sup><a href="#ref-yZfcMIwh" role="doc-biblioref">28</a></sup></span> Tools like
        AlphaFill<span class="citation" data-cites="13fLqNwbD"><sup><a href="#ref-13fLqNwbD"
                    role="doc-biblioref">24</a></sup></span>use structural homology to transplant metals from similar
        PDB structures to the predicted structure. Metal3D does not require explicit homology based on sequence or
        structural alignment like AlphaFill so it is potentially suited to annotate the dark proteome that is now
        accessible from the AlphaFold database with metal binding sites. Metal3D identifies both the catalytic site (1)
        and the zinc finger (2) for the example (PDB 3RZV<span class="citation" data-cites="wW700ShK"><sup><a
                    href="#ref-wW700ShK" role="doc-biblioref">46</a></sup></span> , Figure <a
            href="#fig:alphafold">5</a> A) used in ref<span class="citation" data-cites="13fLqNwbD"><sup><a
                    href="#ref-13fLqNwbD" role="doc-biblioref">24</a></sup></span> with high probability (p=0.99) even
        though one of the sites in the AlphaFold model is slightly disordered with one of the binding residues in the
        solvent facing conformation (D309). The distances between predicted and modeled metal locations for Metal3D are
        0.22 Å and 0.37 Å, for AlphaFill they are 0.21 Å and 0.41 Å.</p>
    <p>AlphaFill uses a 25% sequence identity cutoff which can be problematic for certain proteins with no structurally
        characterized homologues. For human palmitoyltransferase ZDHHC23 (Uniprot Q8IYP9) a high confidence AlphaFold2
        prediction exists but AlphaFill cannot place the zinc ions because the sequence identity is 24% to the closest
        PDB structure (PDB 6BMS<span class="citation" data-cites="RFwEiPTW"><sup><a href="#ref-RFwEiPTW"
                    role="doc-biblioref">47</a></sup></span>), i.e below the 25% cutoff. For the identical site in
        another human palmitoyltransferase ZDHHC15 (Uniprot Q96MV8) AlphaFill is able to place the metal because of
        higher sequence identity to 6BMS (64%) (Figure <a href="#fig:alphafold">5</a> B). For ZDHHC23 Metal3D is able to
        place the metal with high confidence (MAD 0.75 Å for site 1 and 0.48 Å for site 2, p&gt;0.99 ) based on the
        single input structure alone.</p>
    <div id="fig:alphafold" class="fignos">
        <figure>
            <img src="images/3rzv_6bsm_AF2_alphafill_metal3d.png" alt="" />
            <figcaption><span>Figure 5:</span> <strong>Annotation of AlphaFold2 structures</strong> <strong>A</strong>
                Predicted metal binding sites (a and b) from Metal3D, respectively AlphaFill compared to the
                experimentally found zinc positions for Uniprot O95630. Metal3D places the metal with high accuracy even
                if sidechains are not perfectly predicted by AlphaFold for site 1 <strong>B</strong>
                palmitoyltransferase ZDHHC23 (Uniprot Q8IYP9) and ZDHHC15 (Uniprot Q96MV8). AlphaFill can only place the
                metal for ZDHHC15 because sequence identity for ZDHHC23 is only 24 %. Probability isosurfaces from
                Metal3D for both structures at p=0.6, colored in gray.</figcaption>
        </figure>
    </div>
    <h4 id="metalloprotein-engineering">Metalloprotein engineering</h4>
    <p>Human carbonic anhydrase II (HCA2) is a well studied metalloenzyme with a rich amount of mutational data
        available. For the crystal structure of the wildtype enzyme (PDB 2CBA<span class="citation"
            data-cites="jyFnBdWm"><sup><a href="#ref-jyFnBdWm" role="doc-biblioref">48</a></sup></span>), Metal3D
        recapitulates the location of the active site metal with a RMSD to the true metal location of 0.21 Å with a
        probability of p=0.99. At lower probability cutoffs (p&lt;0.4) the probability map indicates further putative
        metal ion binding sites with interactions mediated by surface residues (e.g. H36, D110, p=0.22) (Figure <a
            href="#fig:2cbaprobabilities">2</a>).</p>
    <iframe src="interactive/index.html" title="2CBA probabilities" width="100%" height="900px" frameborder="0">
    </iframe>
    <p>To investigate the capabilities for protein engineering we used mutational data for first and second shell
        mutants of the active site residues in HCA2 with corresponding K<sub>d</sub> values from a colorimetric
        assay.<span class="citation" data-cites="13XuOF3Jj"><sup><a href="#ref-13XuOF3Jj"
                    role="doc-biblioref">49</a></sup></span> For most mutants no crystal structures are available so we
        used the structure builder in the EVOLVE package
        <!--REF--> to choose the most favorable rotamer for each single point mutation based on the EVOLVE-ddg energy
        function
        <!--REF--> with explicit zinc present (modeled using a dummy atom approach<span class="citation"
            data-cites="pchdLF0k"><sup><a href="#ref-pchdLF0k" role="doc-biblioref">50</a></sup></span>). The analysis
        was run for every single mutant and the resulting probability maps from Metal3D were analyzed. For the analysis
        we used the maximum predicted probability as a surrogate to estimate relative changes in K<sub>d</sub>. For
        mutants that decrease zinc binding drastically we observe a drop in the maximum probability predicted by Metal3D
        (Figure <a href="#fig:hca-kd">6</a>).The lowest probability mutants are H119N and H119Q with p=0.23 and 0.38.
        The mutant with the largest loss in zinc affinity H94A has a zinc binding probability of p=0.6. Conservative
        changes to the primary coordination motif (e.g. H → C) reduce the predicted probability by 10 - 30 %. For second
        shell mutants the influence of the mutations is less drastic with only minor changes in the predicted
        probabilities.
    </p>
    <div id="fig:hca-kd" class="fignos">
        <figure>
            <img src="images/kd_vs_p_nolog_newmethod_newmodel_0.5.png" alt="" />
            <figcaption><span>Figure 6:</span> <strong>Protein design application</strong> Experimentally measured
                K<sub>d</sub> values<span class="citation"
                    data-cites="D3VQQdjb w3tzhp92 iu33cNMg 148Si7kTL TNbPH3Y1"><sup><a href="#ref-D3VQQdjb"
                            role="doc-biblioref">51</a>–<a href="#ref-TNbPH3Y1" role="doc-biblioref">55</a></sup></span>
                for 1<sup>st</sup> and 2<sup>nd</sup> shell active site mutants of HCA2 and predicted max probability
                for zinc using Metal3D.</figcaption>
        </figure>
    </div>
    <h2 id="discussion">Discussion</h2>
    <p>Metal3D predicts the probability distribution of zinc ions in protein crystal structures based on a neuronal
        network model trained on natural protein environments. The model performs a segmentation task to determine if a
        specific point in the input space contains a zinc ion or not. Metal3D predicts zinc ion sites with high accuracy
        making use of high resolution crystal structures (&lt;2.5 Å). The use of high resolution structures is necessary
        because at resolutions greater than the average zinc ligand coordination distance (2.2 Å) the uncertainty of the
        zinc location noticeably increases<span class="citation" data-cites="Wt0ducot"><sup><a href="#ref-Wt0ducot"
                    role="doc-biblioref">56</a></sup></span> which would likely hamper the accuracy of the site
        prediction.</p>
    <p>In contrast to currently available tools, for Metal3D, it is not necessary to filter the training examples for
        certain coordination requirements (i.e only sites with at least 2 protein ligands). The model thus sees the
        whole diversity of zinc ion sites present in the PDB. Such a model is advantageous since metalloprotein design
        workflows require models to score the full continuum of zinc sites starting from a suboptimal binding site only
        populated at high metal concentration to a highly organized zinc site in an enzyme with nanomolar metal
        affinity. The predicted probability can be used as a confidence metric or as an optimization target where
        mutations are made to increase probability of zinc binding.</p>
    <h3 id="site-quality">Site quality</h3>
    <p>The fraction of artifactual zinc binding sites in the PDB is estimated to be about 1/3<span class="citation"
            data-cites="Wt0ducot"><sup><a href="#ref-Wt0ducot" role="doc-biblioref">56</a></sup></span> similar to our
        test set used with 70% (133) well coordinated zinc sites with at least 2 distinct protein ligands. To reduce the
        amount of artifactual sites in the training set we presented the model with as many complete sites as possible
        by using crystal symmetry to add adjacent coordinating protein chains. The frequency of artifacts in the
        training set is therefore much lower than 30 %. The sites which still remain incomplete or that are wrongly
        modeled and not excluded through the resolution cutoffs and filtering procedures likely present only a small
        fraction of the training set and their signal is drowned out by the numerical superiority of the correctly
        modeled sites. If the model is used on artifactual sites or partially disordered ones it can still predict the
        metal location with high spatial accuracy but often indicates a lower confidence for the prediction (Figures <a
            href="#fig:2cbaprobabilities">2</a> and <a href="#fig:alphafold">5</a>).</p>
    <p>Metal ion locators that rely on homology such as MIB perform worse on partial binding sites because reducing the
        quality of the available templates by including 1- or 2-coordinate sites would yield many false positives
        (similar to including less homologous structures for the template search). The deep learning based Metal3D can
        likely circumvent this because it does not require any engineered features to predict the location of the metal
        and learns directly from a full representation of the environment surrounding the binding site. This allows
        looking at low confidence sites in the context of a given environment.</p>
    <h3 id="influence-of-non-protein-ligands">Influence of non-protein ligands</h3>
    <p>Exogenous ligands play an important role for metals in biology as all empty coordination sites of metals are
        filled with water molecules in case there is no other exogenous ligand with higher affinity present (e.g. a
        thiol). Like other predictors, both Metal1D and Metal3D do not consider water molecules or other ligands in the
        input as the quality of ligand molecules in the PDB varies.<span class="citation"
            data-cites="NifwNdQd 9UNjBvCL"><sup><a href="#ref-9UNjBvCL" role="doc-biblioref">40</a>,<a
                    href="#ref-NifwNdQd" role="doc-biblioref">57</a></sup></span> In addition, other potential sources
        of input such as AlphaFold do not provide explicit waters wherefore models should not rely on water as an input
        source. It is also not possible to use in silico water predictions because common water placement algorithms to
        place deep waters<span class="citation" data-cites="10C4imKQR izkvZoLX 9UNjBvCL"><sup><a href="#ref-9UNjBvCL"
                    role="doc-biblioref">40</a>,<a href="#ref-10C4imKQR" role="doc-biblioref">58</a>,<a
                    href="#ref-izkvZoLX" role="doc-biblioref">59</a></sup></span> either rely on metal ions being
        present in the input or ignore them completely. Moreover, in protein design algorithms, water is usually only
        implicitly modeled (e.g. in Rosetta).</p>
    <p>For Metal3D, the input channel that encodes the total heavy atom density also encodes an implicit water density
        where all empty space can be interpreted as the solvent. For Metal1D, the contribution of water molecules is
        considered in an implicit way when the score is assigned to a site by considering coordinations including water
        compatible with the one observed (e.g. a HIS<sub>3</sub>Wat site is equivalent to a His<sub>3</sub> site for the
        scoring).</p>
    <h3 id="choice-of-architecture">Choice of architecture</h3>
    <p>This work is the first to report a modern deep learning based model destined for identification of metal ligands
        in proteins. Similar approaches have been used in the more general field of protein-ligand docking where a
        variety of architectures and representations have been used. 3D CNN based approaches such as LigVoxel<span
            class="citation" data-cites="mVGaXlum"><sup><a href="#ref-mVGaXlum" role="doc-biblioref">38</a></sup></span>
        and DeepSite<span class="citation" data-cites="Gq7n6vhH"><sup><a href="#ref-Gq7n6vhH"
                    role="doc-biblioref">37</a></sup></span> commonly use a resolution of 1 Å and similar input features
        as our model to predict the ligand density. However, predicting the density of a multi-atomic ligand is more
        complex than predicting the density of mononuclear metal ions. We therefore did not deem it necessary to include
        a conditioning on how many metal ions are present in the box and rather chose to reflect this in the training
        data where the model needs to learn that only about half of the environments it sees contain one or more metals.
        This choice is validated by the fact that the output probability densities at sufficiently high probability
        cutoffs are spherical with their radius approximately matching the van der Waals radius of zinc.</p>
    <p>Mesh convolutional neural networks trained on a protein surface representation<span class="citation"
            data-cites="yBhqGkBa"><sup><a href="#ref-yBhqGkBa" role="doc-biblioref">36</a></sup></span> also have been
        used to predict the location and identity of protein ligands but this approach can only label the regions of the
        surface that bind the metal ion and is conceptually not able to return the exact location of the metal. Some
        metal ion binding sites are also heavily buried inside proteins as they mediate structural stability rendering
        them inaccessible to a surface based approach. The most recent approaches such as EquiBind<span class="citation"
            data-cites="13hjzBbje"><sup><a href="#ref-13hjzBbje" role="doc-biblioref">39</a></sup></span> use
        equivariant neural networks such as En-Transformer<span class="citation" data-cites="Fc42fl1x"><sup><a
                    href="#ref-Fc42fl1x" role="doc-biblioref">60</a></sup></span> to predict binding keypoints (defined
        as 1/2 distance between the Cɑ of the binding residue and a ligand atom). Explicit side chains are still too
        expensive for such models and these models assume a fixed known stoichiometry of the protein and ligand. Metal3D
        can also deal with proteins that do not bind a metal and does not assume that the amount of ions is known. The
        lack of explicit side chain information renders equivariant models unsuitable for the design of complex metal
        ion binding sites supported by an intricate network of hydrogen bonds that need to be positioned with
        sub-angstrom accuracy. Our model in contrast is less data- and compute-efficient than approaches representing
        the protein as graph due to the need to voxelize the input and provide different rotations of the input
        environment in training but the overall processing time for our model is still low taking typically 25 seconds
        for a 250 residue protein on a multicore GPU workstation (20 CPUs, GTX2070).
        Sequence based models<span class="citation" data-cites="cbwvbkS6 Gj0pG9O5"><sup><a href="#ref-cbwvbkS6"
                    role="doc-biblioref">61</a>,<a href="#ref-Gj0pG9O5" role="doc-biblioref">62</a></sup></span> can
        only use coevolution signals to infer residues in spatial proximity that can bind a metal. This might be
        difficult when it comes to ranking similar amino acids such as aspartate and glutamate or even ranking different
        rotamers where sub-angstrom level precision is needed to identify the mutant with the highest affinity for zinc.
    </p>
    <h3 id="selectivity">Selectivity</h3>
    <p>In terms of selectivity both of our methods have a clear preference for transition metals over main group metals
        after having been trained exclusively on zinc binding sites. The only sites that Metal3D identifies for sodium
        in the test set are the ones that have side chain ligands. Many sodium and potassium sites are using backbone
        carbonyl coordination exclusively, which is not common for zinc and those sites are therefore not detected. Both
        of our methods could be rapidly adapted to predict not only location but also the identity of the metal. In the
        framework of Metal3D even a semantic metal prediction would be possible where the same model predicts different
        output channels for each metal it was trained on. To achieve perfect selectivity using such a model will be
        difficult because sometimes non-native metals are used for crystallization experiments. In this work we chose to
        work exclusively with zinc because it is the most redox stable transition metal and because many training
        examples are available.</p>
    <h3 id="application-for-protein-design">Application for protein design</h3>
    <p>Protein design using 3DCNNs trained on residue identity has been successfully demonstrated and we anticipate that
        our model could be seamlessly integrated into such a workflow<span class="citation"
            data-cites="6szq3cwi"><sup><a href="#ref-6szq3cwi" role="doc-biblioref">32</a></sup></span> to enable fully
        deep learning based design of metalloproteins. We are currently also investigating the combination of Metal3D
        combined with a classic energy-based genetic algorithm-based optimization to make design of metalloproteins<span
            class="citation" data-cites="nlfa683y"><sup><a href="#ref-nlfa683y" role="doc-biblioref">19</a></sup></span>
        easier without having to explicitly model the metal to compute the stability of the protein. As the model
        computes a probability density per residue it can be readily integrated into established software like Rosetta
        relying on rotamer sampling.</p>
    <p>The HCA2 application demonstrates the utility of Metal3D for protein engineering (Figure <a
            href="#fig:2cbaprobabilities">2</a>). The thermodynamics of metal ion binding to proteins are
        complicated<span class="citation" data-cites="TlDhIlX"><sup><a href="#ref-TlDhIlX"
                    role="doc-biblioref">63</a></sup></span> and there are currently no high-throughput based
        experimental approaches that could generate a dataset large enough to train a model directly on predicting
        K<sub>d</sub>. The data we use were obtained from a colorimetric assay with very high affinity of zinc in the
        picomolar range.<span class="citation" data-cites="D3VQQdjb w3tzhp92 iu33cNMg 148Si7kTL TNbPH3Y1"><sup><a
                    href="#ref-D3VQQdjb" role="doc-biblioref">51</a>–<a href="#ref-TNbPH3Y1"
                    role="doc-biblioref">55</a></sup></span> More recent studies using ITC<span class="citation"
            data-cites="TlDhIlX"><sup><a href="#ref-TlDhIlX" role="doc-biblioref">63</a></sup></span> instead of the
        colorimetric assay indicate lower K<sub>d</sub> values in the nanomolar range for wild type HCA2. We can
        therefore only use the colorimetric data to estimate how well the model can recapitulate relative changes in the
        K<sub>d</sub> for different mutations in the first and second shell of a prototypical metalloprotein.</p>
    <p>Metal3D allows moving away from using rational approaches such as the i, i + 4 di His motifs used for the
        assembly and stabilization of metalloproteins to a fully automated approach where potential metal binding
        configurations can be scored computationally.<span class="citation"
            data-cites="RJ44nj3R 15drt0nXZ 85PeY3aG"><sup><a href="#ref-RJ44nj3R" role="doc-biblioref">64</a>–<a
                    href="#ref-85PeY3aG" role="doc-biblioref">66</a></sup></span></p>
    <h3 id="metal1d-vs.-metal3d">Metal1D vs. Metal3D</h3>
    <p>Metal1D is inferior to Metal3D for the prediction of metal ion binding sites because it produces more false
        positives while at the same time detecting fewer metal sites. Also the positioning of sites is somewhat
        imprecise. This demonstrates the inherent limitation of using solely distance based features for prediction of
        metal location. BioMetAll which is the tool most similar to Metal1D also suffers from many false positive
        predictions. In contrast, Metal1D is more data-efficient than Metal3D and provides predictions faster.</p>
    <p>
        <!-- KD values discussion (new ITC data vs. old data by Kiefer, Fierke etc): 
 
 Old method: Enzyme-
bound zinc (E—Zn) was quantitated using the colorimetric
4-(2-pyridylazo)resorcinol (PAR) method of Hunt et al.
(1984) and measuring the absorbance at 500 nm. 4-(2-Pyridylazo)resorcinol (PAR) is a dibasic acid that forms the protonated complexes with most metal ions. It serves as a metallochromic indicator and is suitable as a chromogenic agent for the quantitative determination of over 50 elements.

removing unbound zinc by chromatography on a
PD-10 column, and measuring the protein concentration and
bound zinc concentration in the eluant using the PAR assay
(Hunt et al., 1984).
The concentration of free zinc in the
dialysis buffer was calculated from the Tris—zinc stability
constants (Dawson et al., 1986). The dissociation constant
was calculated using KaleidaGraph program with eq:
[E-ZN]/[E]tot = C/(1+Kd/[Zn]free)

Newer method: [@doi:10.1021/ic301645j]
 -->
    </p>
    <h2 id="conclusion">Conclusion</h2>
    <p>We present two metal ion location predictors: Metal3D based on 3D convolutional neural networks and Metal1D based
        on distances and amino acid propensity maps. Metal3D is the first tool with sub-angstrom level precision to
        predict the location of metal ions in proteins that does not rely on searching for structurally homologous
        proteins in a database. We therefore anticipate different applications such as protein-function annotation of
        structures predicted using AlphaFold2,<span class="citation" data-cites="10bYywzs6"><sup><a
                    href="#ref-10bYywzs6" role="doc-biblioref">67</a></sup></span> integration in protein design
        software and detection of cryptic metal binding sites that can be used to engineer PPIs. Such cryptic metal ion
        binding sites in common drug targets could also be used to engineer novel metallodrugs. Many of these
        applications will allow us to explore the still vastly untapped potential of proteins as large multi-dentate
        metal ligands with programmable surfaces.</p>
    <!-- Zinc is generally available in the intracellular volume and extracellularly with the total concentration of zinc in cells is estimated to be about about 200 μM [@doi:10.1021/cr800556u]. The real available concentration of free zinc is picomolar[@doi:10.1021/cr800556u] wherefore it is crucial that the design tool produces well designed sites with high spatial precision of the metal ion.  -->
    <h2 id="materials-and-methods">Materials and Methods</h2>
    <h3 id="dataset">Dataset</h3>
    <p>The input PDB files for training were obtained from the RCSB<span class="citation" data-cites="8cYXwVxI"><sup><a
                    href="#ref-8cYXwVxI" role="doc-biblioref">68</a></sup></span> protein databank (downloaded 5th March
        2021). We use a clustering of the structures at 30% sequence identity using mmseqs2<span class="citation"
            data-cites="jj6qOofE"><sup><a href="#ref-jj6qOofE" role="doc-biblioref">69</a></sup></span> to largely
        remove sequence and structural redundancy in the input dataset.
        For each cluster, we check whether a zinc is contained in one of the structures, whether the resolution of these
        structures is better than 2.5 Å, if the experimental method is x-ray crystallography and whether the structure
        does not contain nucleic acids. If there are multiple structures fulfilling these criteria, the highest
        resolution structure is used. All structures larger than 3000 residues are discarded. We always use the first
        biological assembly to sample the training environments. The structures were stripped of all exogenous ligands
        except for zinc . If there are multiple models with e.g. alternative residue conformations for a given
        structure, the first one is used. For each biological assembly we used the symmetry of the asymmetric unit to
        generate a protein structure that contains all neighboring copies of the protein in the crystal such that metal
        sites at crystal contacts are fully coordinated.</p>
    <p>The train/val/test split was performed based on sequence identity using <code>easy-search</code> in mmseqs2. All
        proteins that had no (partial) sequence overlap with any other protein in the dataset were put into the test/val
        set (85 proteins) which we further split into a test set of 59 structures and a validation set of 26 structures.
        The training set contained 2085 structures. (Supplemental Data 1).</p>
    <p>For the analysis, we always used the biological assembly and not the symmetry augmented structure. For the
        specificity analysis with respect to other transition metals, clusters from the PDB were randomly sampled to
        extract 25 biological assemblies per metal.</p>
    <p>By default all zinc sites in the test and validation set were used for the analysis. Since some of the sites
        might be affected by the crystallization conditions, we also created a subset of all sites that contained at
        least 2 amino acid ligands to largely exclude crystallization artifacts. To analyze metal ion selectivity, we
        selected sites with at least 3 unique protein ligands to only use biologically significant sites with a high
        degree of metal preorganization as such sites should exhibit more selectivity for specific metals compared to
        sites with only 2 unique protein-ligands.</p>
    <h3 id="metal-1d">Metal 1D</h3>
    <p>Metal1D uses a probability map derived from <code>LINK</code> records in protein structures (Figure <a
            href="#fig:method">1</a>). The <code>LINK</code> section of a PDB file specifies the connectivity between
        zinc (or any other ligand) and the amino acids of the protein, and each <code>LINK</code> record specifies one
        linkage. This is an extension of the approach by Barber-Zucker <em>et al.</em>,<span class="citation"
            data-cites="FB5cckAJ"><sup><a href="#ref-FB5cckAJ" role="doc-biblioref">70</a></sup></span> in which
        <code>LINK</code> records were used to investigate the propensity of transition metals to bind different amino
        acids.</p>
    <p>Using the training set we generated a probability map for the propensity of different coordination environments
        to bind a zinc (e.g CCCC, CCHH etc.). For each zinc ion the coordination is extracted from the<code>LINK</code>
        records excluding records involving only single amino acids (weak binding sites). Also, <code>LINK</code>
        records containing water molecules are excluded because of the difficulties in placing water molecules a
        posteriori in 3D structures when metal ions are present and because data quality of modelled water molecules
        varies. The probability map contains the counts of coordination environments found.
        Making a prediction using Metal1D consists of two steps (Figure <a href="#fig:method">1</a>): Identification of
        possible metal coordinating residues in the structure via the scoring of each amino acid, and scoring of the
        likelihood of coordination for putative sites predicted by placing a metal between the identified coordinating
        residues.</p>
    <p>The protein structure is analyzed using the BioPandas python library.<span class="citation"
            data-cites="CcRTAd7h"><sup><a href="#ref-CcRTAd7h" role="doc-biblioref">71</a></sup></span> To identify
        coordinating residues, a per residue score is assigned by performing a geometrical search from a reference
        point, defined as the coordinate of the most probable metal binding atom, within a search radius considered as
        roughly twice the typical distance between the metal ion and the binding atom of amino acids in proteins (2.2 ±
        0.2 Å as determined from <code>LINK</code> records). The search radius used was 5.5 Å in order to be able to
        take into account also deviations from the ideal coordination. In the case of amino acids which present more
        than one putative coordinating atom, such as e.g. histidine, the mid-point between the donor atoms is used as
        reference point and the search radius is enlarged accordingly. The atoms used as reference points for each amino
        acid and the increase in the search radius are reported in Supplemental Table <a
            href="#tbl:metal1dreferencepoints">S1</a> .
        The score is assigned to each amino acid considering all the other reference points of other amino acids within
        the search radius, and summing the probabilities in the probability map for coordinations compatible with the
        one observed. In the ideal case, a score of 1 corresponds to an amino acid surrounded by all possible
        coordinating amino acids observed in the probability map. In practice, scores result between 0 and &lt; 1. Once
        all amino acids in the chain are scored, the metal location predictions are made grouping the highest-scored
        amino acids in clusters (defined as the ones within the chosen threshold with respect to the highest-scored one)
        based on distance. This is done using <code>scipy.spatial.distance_matrix</code> and grouping together
        highest-scored amino acids closer than twice the search radius. For each cluster, a site prediction is made as a
        weighted average between the coordinates of the reference point of each amino acid, using as weighting factor
        the amino acid score. For isolated amino acids with a high score (e.g. a single histidine) the same score is
        assigned to the closest reference point from another amino acid, to be able to compute the position of the metal
        as before. Possible artifacts resulting from this fictitious score are resolved in the final step of the
        prediction.</p>
    <p>After the metal has been placed the likelihood of the putative sites can be assessed by performing a geometrical
        search centered on the predicted metal coordinates (within 60% of the search radius, i.e 3.3 Å) and a final
        score is now assigned to the site. The final score is assigned in the same way as the amino acid scores based on
        the probability map, and has the advantage of being able to sort the predicted metal sites based on their
        frequency in the training set. A cutoff parameter is used to exclude sites with a probability lower than a
        certain threshold with respect to the highest-scored one. This final scoring also mitigates the errors which can
        be introduced by calculating the coordinates of the site simply as a weighted average excluding or assigning a
        low probability to the site ending in unfavorable positions in space.</p>
    <h3 id="metal-3d">Metal 3D</h3>
    <h4 id="voxelization">Voxelization</h4>
    <p>We used the moleculekit python library<span class="citation" data-cites="q2gxO5bP mVGaXlum"><sup><a
                    href="#ref-mVGaXlum" role="doc-biblioref">38</a>,<a href="#ref-q2gxO5bP"
                    role="doc-biblioref">72</a></sup></span> to voxelize the input structures into 3D grids. 8 different
        input channels are used: aromatic, hydrophobic, positive ionizable, negative ionizable, hbond donor, hbond
        acceptor, occupancy, and metal ion binding site chain (Supplemental Table <a href="#tbl:voxelchannels">S2</a>).
        The channels are assigned using AutoDockVina atom names and a boolean mask. For each atom matching one of the
        categories a pair correlation function centered on the atom is used to assign the voxel value.<span
            class="citation" data-cites="mVGaXlum"><sup><a href="#ref-mVGaXlum" role="doc-biblioref">38</a></sup></span>
        For the target tensor only the zinc ions were used for the voxelization. The target tensor was discretized
        setting any voxel above 0.05 to 1 (true location of zinc), all other to 0 (no zinc). We used a box size of 16 Å
        centered on the Cɑ atom of a residue, rotating each environment randomly for training before voxelization. The
        voxel grid used a 0.5 Å resolution for the input and target tensors. Any alternative side chain conformations
        modeled were discarded keeping only the highest occupancy. For the voxelization only heavy atoms were used. For
        all structures selected for the respective sets we partitioned the residues of the protein into residues within
        12 Å of a zinc ion and those further away (based on the distance to the Cɑ atom). A single zinc site will
        therefore be present many times in the dataset but each time translated and rotated in the box. A balanced set
        of examples was used sampling equal numbers of residues that are close to a zinc and residues randomly drawn
        from the non-zinc binding residues. The sampling of residues is based on the biological assembly of the protein,
        the voxelization is based on the full 3D structure including neighboring asymmetric units in the crystal
        structure. The environments are precomputed and stored using lxf compression in HDF5 files for concurrent access
        during training. In total, 252324 environments were voxelized for the training set, 6550 for the test set, 3067
        for the validation set. The voxelization was implemented using ray.<span class="citation"
            data-cites="19MS2ZVR1"><sup><a href="#ref-19MS2ZVR1" role="doc-biblioref">73</a></sup></span></p>
    <h4 id="model-training">Model training</h4>
    <p>We used PyTorch 1.10<span class="citation" data-cites="iOWq2JB8"><sup><a href="#ref-iOWq2JB8"
                    role="doc-biblioref">74</a></sup></span> to train the model. All layers of the network are
        convolutional layers with filter size 1.5 Å except for the fifth layer where a 8 Å filter is used to capture
        long range interactions. We use zero padding to keep the size of the boxes constant. Models were trained on a
        workstation with NVIDIA GTX3090 GPU and 32 CPU cores. Binary Cross Entropy<span class="citation"
            data-cites="hchMCnC1"><sup><a href="#ref-hchMCnC1" role="doc-biblioref">75</a></sup></span> loss is used to
        train the model. The rectified linear unit (ReLU) non-linearity is used except for the last layer which uses a
        sigmoid function that yields the probability for zinc per voxel. A dropout layer (p = 0.1) was used between the
        5th and 6th layers. The network was trained using AdaDelta employing a stepped learning rate (lr=0.5, ɣ=0.9), a
        batch size of 150, and 12 epochs to train.</p>
    <h4 id="hyperparameter-tuning">Hyperparameter tuning</h4>
    <p>We used the ray[tune] library<span class="citation" data-cites="19MS2ZVR1"><sup><a href="#ref-19MS2ZVR1"
                    role="doc-biblioref">73</a></sup></span> to perform a hyperparameter search choosing 20 different
        combinations between the following parameters with the best combination of parameters in bold:</p>
    <ul>
        <li>filtersize: <strong>3</strong>,4 (in units of 0.5 Å)</li>
        <li>dropout : <strong>0.1</strong>, 0.2, 0.4, 0.5</li>
        <li>learning rate : <strong>0.5</strong>, 1.0, 2.0</li>
        <li>gamma: 0.5, 0.7, 0.8, <strong>0.9</strong></li>
        <li>largest dimension 80, 100, <strong>120</strong></li>
    </ul>
    <h4 id="grid-averaging">Grid Averaging</h4>
    <p>The model takes as input a <code>(8,32,32,32)</code> tensor and outputs a <code>(1,32,32,32)</code> tensor
        containing the probability density for zinc centered on the Cɑ atom of the input residue.
        Predictions for a complete protein were obtained by voxelizing select residues of the protein (default all
        cysteines, histidines, aspartates, glutamates) and averaging the boxes using a global grid (Figure <a
            href="#fig:method">1</a> B). 98 % of the metal sites in the training data have at least one of those
        residues closeby wherefore this significant decrease in computational cost seems appropriate for most uses. The
        global grid is obtained by computing the bounding box of all points and using a regular spaced (0.5 Å) grid. For
        each grid point in the global grid the predicted probability maps within 0.25 Å of the grid point are averaged.
        The search is sped up using the KD-Tree implementation in scipy.<span class="citation"
            data-cites="8Miti2Gz"><sup><a href="#ref-8Miti2Gz" role="doc-biblioref">76</a></sup></span></p>
    <h5 id="metal-ion-placement">Metal ion placement</h5>

    <p>The global probability density is used to perform clustering of voxels above a certain probability threshold
        (default p=0.15, cutoff 7 Å) using AgglomerativeClustering implemented in scikit-learn.<span class="citation"
            data-cites="AujvwLp6"><sup><a href="#ref-AujvwLp6" role="doc-biblioref">77</a></sup></span> For each cluster
        the weighted average of the voxels in the cluster is computed using the probabilities for each point as the
        weight. This results in one metal placed per cluster.</p>

    <h5 id="visualization">Visualization</h5>
    <p>We make available a command line program and interactive notebook allowing the user to visualize the results. The
        averaged probability map is stored as a <code>cube</code> file. The most likely metal coordinates for use in
        subsequent processing are stored in a <code>pdb</code> file. The command line program uses VMD<span
            class="citation" data-cites="PEXAwV3k"><sup><a href="#ref-PEXAwV3k" role="doc-biblioref">78</a></sup></span>
        to visualize the input protein and the predicted density, for the jupyter notebook 3Dmol.js/py3Dmol<span
            class="citation" data-cites="BguJBlg6"><sup><a href="#ref-BguJBlg6" role="doc-biblioref">79</a></sup></span>
        is used.</p>
    <h4 id="evaluation">Evaluation</h4>
    <h5 id="comparison">Comparison</h5>
    <p>In order to standardize the evaluation between different tools, we always used the same test set used for the
        training of Metal1D and Metal3D. In order to compute standard metrics such as precision and recall, we chose to
        assess the performance of all assessed tools (Metal1D, Metal3D, BioMetAll, MIB) in a binary fashion. Any
        prediction within 5 Å of an experimental metal site is counted as true positive (TP). Multiple predictions by
        the same tool for the same site are counted as 1 TP. Any experimental site that has no predicted metal within 5
        Å is counted as false negative (FN). A false positive (FP) prediction is a prediction that is not within 5 Å of
        a zinc site and also not within 5 Å of any other false positive prediction. If two or more false positive
        predictions are within 5 Å, they are counted as a single false positive prediction for the same site. In
        practice we first evaluate the true positive and false negative predictions and remove those from the set of
        predicted positions. The remaining predictions are all false positives and are clustered using
        AgglomerativeClustering with a radius of 5 Å. The number of false positives is determined from the number of
        clusters. Using the binary metric we assessed how good the models are at discovering sites and how much these
        predictions can be trusted.</p>
    <p>In order to assess the quality of the predictions, we additionally compute for all the true positive predictions
        the mean of the Euclidean distance between the true and predicted site (mean absolute deviation MAD). For
        Metal1D, MIB, and BioMetAll, MAD was computed for all predictions above the threshold within 5 Å of a true zinc
        site where <span class="math inline">\(\sum\text{predicted sites} \geq \sum\text{ TP}\)</span>. This was done as
        some tools predict the same site for different residue combinations and we wanted to assess the general
        performance for all predicted sites above a certain cutoff and not just for the best predicted site above the
        cutoff. For Metal3D the weighted average of all voxels above the cutoff was used.</p>
    <p>Precision was calculated as</p>
    <p><span class="math display">\[
            \text{Precision} =\frac{\#\text{ correct metal sites}}{\#\text{ correct metal sites} + \#\text{ false
            positive clustered}} = \frac{\text{TP}}{\text{TP}+\text{FP}}
            \]</span></p>
    <p>Recall was calculated as</p>
    <p><span class="math display">\[
            \text{Recall} =\frac{\#\text{ correct metal sites}}{\#\text{ correct metal sites} + \#\text{ not found metal
            sites}} = \frac{\text{TP}}{\text{TP}+\text{FN}}
            \]</span></p>
    <h5 id="model-assessment-metal3d">Model assessment Metal3D</h5>
    <p>To evaluate the trained models we monitored loss and how accurately the model predicts the metal density of the
        test set. We used a discretized version of the Jaccard index setting each voxel either as 0 (no metal) or 1
        (zinc present). We tested multiple different decision boundaries (0.5, 0.6, 0.75, 0.9) and also compared a
        slightly smaller centered box to remove any spurious density at the box edges, where the model has only
        incomplete information to make predictions.</p>
    <p>The Jaccard index is computed as
        <span class="math display">\[
            J=\frac{\#\left|V_{p} \cap V_{exp}\right|}{\#\left|V_{p} \cup V_{exp}\right|},
            \]</span>
        where <span class="math inline">\(Vp\)</span> is the array of voxels with predicted probability above the
        decision boundary and <span class="math inline">\(V_{exp}\)</span> is the array of voxels with the true metal
        locations also discretized at the same probability threshold.
    </p>
    <h5 id="hca2-mutants">HCA2 mutants</h5>
    <p>The data for human carbonic anhydrase 2 (HCA2) mutants was extracted from refs<span class="citation"
            data-cites="D3VQQdjb w3tzhp92 iu33cNMg 148Si7kTL TNbPH3Y1"><sup><a href="#ref-D3VQQdjb"
                    role="doc-biblioref">51</a>–<a href="#ref-TNbPH3Y1" role="doc-biblioref">55</a></sup></span> and the
        crystal structure 2CBA<span class="citation" data-cites="WjHESl20 jyFnBdWm"><sup><a href="#ref-jyFnBdWm"
                    role="doc-biblioref">48</a>,<a href="#ref-WjHESl20" role="doc-biblioref">80</a></sup></span> was
        used. The zinc was modeled using the zinc cationic dummy model forcefield<span class="citation"
            data-cites="pchdLF0k"><sup><a href="#ref-pchdLF0k" role="doc-biblioref">50</a></sup></span> and we verified
        that energy minimization produced the correct coordination environment. The Richardson rotamer library<span
            class="citation" data-cites="Mg40Hdad"><sup><a href="#ref-Mg40Hdad" role="doc-biblioref">81</a></sup></span>
        was used with the EVOLVE-ddG energy function to compute the most stable rotamer for a given mutation with the
        zinc present. The lowest-energy mutant was used for the prediction of the location of metals using Metal3D.</p>
    <h2 id="additional-information">Additional information</h2>
    <h3 id="acknowledgement">Acknowledgement</h3>
    <p>Supported by Swiss National Science Foundation Grant Number 200020-185092 with computational resources from the
        Swiss National Computing Centre CSCS.
        We thank the developers of <a href="https://3dmol.csb.pitt.edu/">3Dmol.js</a> and <a
            href="https://manubot.org/">manubot</a> that made the interactive version of this manuscript possible.</p>
    <h3 id="data-availibility">Data availibility</h3>
    <p>Code and training data are available on <a
            href="https://github.com/lcbc-epfl/metal-site-prediction">lcbc-epfl/metal-site-prediction</a> and on Zenodo.
    </p>
    <h3 id="conflict-of-interest">Conflict of interest</h3>
    <p>None declared</p>
    <h3 id="author-contributions">Author contributions</h3>
    <p>S.L.D and A.L designed research, S.L.D, A.L, U.R conceptualized research, S.L.D and A.L developed methodology and
        software, S.L.D and A.L wrote first draft, S.L.D, A.L, U.R revised and edited draft, U.R supervised research and
        acquired funding.</p>
    <h2 id="supplement">Supplement</h2>
    <h3 id="metal1d-1">Metal1D</h3>
    <div id="tbl:metal1dreferencepoints" class="tablenos">
        <table>
            <caption><span>Table S1:</span> Atoms used as reference points for each amino acid in Metal1D. In the case
                of amino acids with more than one possible ligand atom, the search radius is enlarged, the increase is
                computed from the midpoint between all ligating atoms. Typical values computed for structure data files
                downloaded from the PDB are reported. </caption>
            <thead>
                <tr class="header">
                    <th style="text-align: left;">Amino acid</th>
                    <th style="text-align: center;">Residue name</th>
                    <th style="text-align: center;">Label(s)</th>
                    <th style="text-align: center;">Search radius increase (Å)</th>
                </tr>
            </thead>
            <tbody>
                <tr class="odd">
                    <td style="text-align: left;">Alanine</td>
                    <td style="text-align: center;">ALA</td>
                    <td style="text-align: center;">O</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Arginine</td>
                    <td style="text-align: center;">ARG</td>
                    <td style="text-align: center;">NH1, NH2</td>
                    <td style="text-align: center;">1.2</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Asparagine</td>
                    <td style="text-align: center;">ASN</td>
                    <td style="text-align: center;">OD1</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Aspartic acid</td>
                    <td style="text-align: center;">ASP</td>
                    <td style="text-align: center;">OD1, OD2</td>
                    <td style="text-align: center;">1.105</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Cysteine</td>
                    <td style="text-align: center;">CYS</td>
                    <td style="text-align: center;">SG</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Glutamic acid</td>
                    <td style="text-align: center;">GLU</td>
                    <td style="text-align: center;">OE1, OE2</td>
                    <td style="text-align: center;">1.105</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Glutamine</td>
                    <td style="text-align: center;">GLN</td>
                    <td style="text-align: center;">OE1</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Glycine</td>
                    <td style="text-align: center;">GLY</td>
                    <td style="text-align: center;">O</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Histidine</td>
                    <td style="text-align: center;">HIS</td>
                    <td style="text-align: center;">ND1, ND2</td>
                    <td style="text-align: center;">1.08</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Isoleucine</td>
                    <td style="text-align: center;">ILE</td>
                    <td style="text-align: center;">O</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Leucine</td>
                    <td style="text-align: center;">LEU</td>
                    <td style="text-align: center;">O</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Lysine</td>
                    <td style="text-align: center;">LYS</td>
                    <td style="text-align: center;">NZ</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Methionine</td>
                    <td style="text-align: center;">MET</td>
                    <td style="text-align: center;">SD</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Phenylalanine</td>
                    <td style="text-align: center;">PHE</td>
                    <td style="text-align: center;">O</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Proline</td>
                    <td style="text-align: center;">PRO</td>
                    <td style="text-align: center;">O</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Serine</td>
                    <td style="text-align: center;">SER</td>
                    <td style="text-align: center;">OG</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Threonine</td>
                    <td style="text-align: center;">THR</td>
                    <td style="text-align: center;">OG1</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Tryptophan</td>
                    <td style="text-align: center;">TRP</td>
                    <td style="text-align: center;">O</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Tyrosine</td>
                    <td style="text-align: center;">TYR</td>
                    <td style="text-align: center;">OH</td>
                    <td style="text-align: center;">0</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Valine</td>
                    <td style="text-align: center;">VAL</td>
                    <td style="text-align: center;">OH</td>
                    <td style="text-align: center;">0</td>
                </tr>
            </tbody>
        </table>
    </div>
    <div id="tbl:voxelchannels" class="tablenos">
        <table>
            <caption><span>Table S2:</span> Atom selections used for voxelization of proteins using moleculekit
            </caption>
            <colgroup>
                <col style="width: 14%" />
                <col style="width: 85%" />
            </colgroup>
            <thead>
                <tr class="header">
                    <th style="text-align: left;">Channel name</th>
                    <th style="text-align: left;">Selected atoms</th>
                </tr>
            </thead>
            <tbody>
                <tr class="odd">
                    <td style="text-align: left;">aromatic</td>
                    <td style="text-align: left;">HIS TRP TYR PHE sidechain without CB</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">hydrophobic</td>
                    <td style="text-align: left;">element C</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">occupancy</td>
                    <td style="text-align: left;">all protein heavy atoms</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">hbond donor</td>
                    <td style="text-align: left;">(ASN GLN TRP MSE SER THR MET CYS and name ND2 NE2 NE1 SG SE OG OG1)
                        and name N</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">hbond acceptor</td>
                    <td style="text-align: left;">(resname ASP GLU HIS SER THR MSE CYS MET and name ND2 NE2 OE1 OE2 OD1
                        OD2 OG OG1 SE SG) or name O</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">metalbinding</td>
                    <td style="text-align: left;">(name ND1 NE2 SG OE1 OE2 OD2) or (protein and name O N)</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">positive charge</td>
                    <td style="text-align: left;">resname LYS ARG HIS and name NZ NH1 NH2 ND1 NE2 NE</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">negative charge</td>
                    <td style="text-align: left;">resname ASP GLU and name OD1 OD2 OE1 OE2</td>
                </tr>
            </tbody>
        </table>
    </div>
    <div id="fig:jaccard" class="fignos">
        <figure>
            <img src="images/jaccard_0.5.jpg" data-tag="S1" alt="" />
            <figcaption><span>Figure S1:</span> Discretized Jaccard indices using different cutoffs for edge trimming
                and different probability cutoffs (p(metal)) showing that Metal3D predictions well reproduce the target
                environments in the test set.</figcaption>
        </figure>
    </div>
    <h3 id="precisionrecall">Precision/Recall</h3>
    <div id="fig:precisionrecall" class="fignos">
        <figure>
            <img src="images/precisionrecall_0.5.jpg" data-tag="S2" alt="" />
            <figcaption><span>Figure S2:</span> Precision recall curve for Metal1D and Metal3D with the probability
                cutoffs (Metal3D) or thresholds (Metal1D) used in the analysis.</figcaption>
        </figure>
    </div>
    <h3 id="metal-selectivity">Metal selectivity</h3>
    <h4 id="metal1d-2">Metal1D</h4>
    <div id="fig:selectivity-metal1d" class="fignos">
        <figure>
            <img src="images/metal1D_metal_selectivity.jpg" data-tag="S3" alt="" />
            <figcaption><span>Figure S3:</span> Recall for zinc testset and a 25 randomly drawn structures for other
                transition, alkali and earth-alkali metals for Metal1D</figcaption>
        </figure>
    </div>
    <div id="fig:selectivity-distance-metal1d" class="fignos">
        <figure>
            <img src="images/model_0.5metal1D_distances_violin.jpg" data-tag="S4" alt="" />
            <figcaption><span>Figure S4:</span> Distance distribution Metal1D</figcaption>
        </figure>
    </div>
    <h4 id="metal3d-1">Metal3D</h4>
    <div id="fig:selectivity-distance-metal3d" class="fignos">
        <figure>
            <img src="images/model_0.5metal3D_distances_violin_0.5.jpg" data-tag="S5" alt="" />
            <figcaption><span>Figure S5:</span> MAD for Metal3D for all sites with 3+ unique protein ligands in the test
                set and for the selected structures for the other metals. For each ion the whisker plot indicates the
                median probability (white dot) and the first quartiles (black box).</figcaption>
        </figure>
    </div>
    <div id="fig:selectivity-probability-metal3d" class="fignos">
        <figure>
            <img src="images/probability_violin.jpg" data-tag="S6" alt="" />
            <figcaption><span>Figure S6:</span> Probability distribution for Metal3D on sites with 3+ unique protein
                ligands in the test set and for the selected structures for the other metals. For each ion the whisker
                plot indicates the median probability (white dot) and the first quartiles (black box).</figcaption>
        </figure>
    </div>
    <h3 id="comparison-1">Comparison</h3>
    <div id="fig:madonlyGoodZnmetal3d" class="fignos">
        <figure>
            <img src="images/mad_violin_0.5_2+.jpg" data-tag="S7" alt="" />
            <figcaption><span>Figure S7:</span> MAD only 2 residue coordinated zincs. For each tool the whisker plot
                indicates the median (white dot) and the first quartiles (black box).</figcaption>
        </figure>
    </div>
    <h5 id="false-positive-metal3d-p0.9">False positive Metal3D p=0.9</h5>
    <div id="fig:4jjjFP" class="fignos">
        <figure>
            <img src="images/4JJJ_FalsePositive_p=0.9_annotated.png" data-tag="S8" alt="" />
            <figcaption><span>Figure S8:</span> False positive for Metal3D at p=0.9 in PDB 4JJJ. A calcium site is
                misclassified as zinc site.</figcaption>
        </figure>
    </div>
    <h5 id="mean-absolute-deviations">Mean absolute deviations</h5>
    <p>The measured MAD for all true positive zinc sites for each tool in the database.</p>
    <div id="tbl:compmadmediantest" class="tablenos">
        <table>
            <caption><span>Table S3:</span> MAD and median on zinc test set for all zinc sites (n=189). </caption>
            <thead>
                <tr class="header">
                    <th style="text-align: left;">tool</th>
                    <th style="text-align: right;">MAD [Å]</th>
                    <th style="text-align: right;">median [Å]</th>
                </tr>
            </thead>
            <tbody>
                <tr class="odd">
                    <td style="text-align: left;">BioMetAll c=0.5</td>
                    <td style="text-align: right;">2.72 ± 1.33</td>
                    <td style="text-align: right;">2.86</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">MIB t=1.25</td>
                    <td style="text-align: right;">1.13 ± 1.24</td>
                    <td style="text-align: right;">0.60</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">MIB t=1.9</td>
                    <td style="text-align: right;">0.77 ± 1.09</td>
                    <td style="text-align: right;"><strong>0.44</strong></td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Metal1D t=0.5</td>
                    <td style="text-align: right;">2.07 ± 1.33</td>
                    <td style="text-align: right;">2.06</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Metal1D t=0.75</td>
                    <td style="text-align: right;">2.19 ± 1.26</td>
                    <td style="text-align: right;">2.12</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Metal3D p=0.25</td>
                    <td style="text-align: right;">0.74 ± 0.66</td>
                    <td style="text-align: right;">0.61</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Metal3D p=0.5</td>
                    <td style="text-align: right;">0.73 ± 0.66</td>
                    <td style="text-align: right;">0.54</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Metal3D p=0.75</td>
                    <td style="text-align: right;">0.71 ± 0.64</td>
                    <td style="text-align: right;">0.51</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Metal3D p=0.9</td>
                    <td style="text-align: right;"><strong>0.70 ± 0.64</strong></td>
                    <td style="text-align: right;">0.52</td>
                </tr>
            </tbody>
        </table>
    </div>
    <div id="tbl:compmadmediantestgoodonly" class="tablenos">
        <table>
            <caption><span>Table S4:</span> MAD and median on zinc test set for zinc sites with at least 2 protein
                ligands (n=133). </caption>
            <thead>
                <tr class="header">
                    <th style="text-align: left;">tool</th>
                    <th style="text-align: right;">MAD [Å]</th>
                    <th style="text-align: right;">median [Å]</th>
                </tr>
            </thead>
            <tbody>
                <tr class="odd">
                    <td style="text-align: left;">BioMetAll c=0.5</td>
                    <td style="text-align: right;">2.68 ± 1.33</td>
                    <td style="text-align: right;">2.84</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">MIB t=1.25</td>
                    <td style="text-align: right;">1.09 ± 1.21</td>
                    <td style="text-align: right;">0.60</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">MIB t=1.9</td>
                    <td style="text-align: right;">0.77 ± 1.09</td>
                    <td style="text-align: right;"><strong>0.44</strong></td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Metal1D t=0.5</td>
                    <td style="text-align: right;">1.97 ± 1.29</td>
                    <td style="text-align: right;">1.99</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Metal1D t=0.75</td>
                    <td style="text-align: right;">2.06 ± 1.24</td>
                    <td style="text-align: right;">2.09</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Metal3D p=0.25</td>
                    <td style="text-align: right;">0.69 ± 0.58</td>
                    <td style="text-align: right;">0.56</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Metal3D p=0.5</td>
                    <td style="text-align: right;">0.69 ± 0.59</td>
                    <td style="text-align: right;">0.54</td>
                </tr>
                <tr class="even">
                    <td style="text-align: left;">Metal3D p=0.75</td>
                    <td style="text-align: right;">0.71 ± 0.64</td>
                    <td style="text-align: right;">0.51</td>
                </tr>
                <tr class="odd">
                    <td style="text-align: left;">Metal3D p=0.9</td>
                    <td style="text-align: right;"><strong>0.70 ± 0.64</strong></td>
                    <td style="text-align: right;">0.52</td>
                </tr>
            </tbody>
        </table>
    </div>
    <h2 class="page_break_before" id="references">References</h2>
    <!-- Explicitly insert bibliography here -->
    <div id="refs" class="references" role="doc-bibliography">
        <div id="ref-gxiv5uTU">
            <p>(1) Yu, F.; Cangelosi, V. M.; Zastrow, M. L.; Tegoni, M.; Plegaria, J. S.; Tebo, A. G.; Mocny, C. S.;
                Ruckthong, L.; Qayyum, H.; Pecoraro, V. L. Protein Design: Toward Functional Metalloenzymes. <em>Chem.
                    Rev.</em> <strong>2014</strong>, <em>114</em> (7), 3495–3578. <a
                    href="https://doi.org/10.1021/cr400458x">https://doi.org/10.1021/cr400458x</a>.</p>
        </div>
        <div id="ref-LY9UGI7">
            <p>(2) Guffy, S. L.; Der, B. S.; Kuhlman, B. Probing the Minimal Determinants of Zinc Binding with
                Computational Protein Design. <em>Protein Engineering, Design and Selection</em> <strong>2016</strong>,
                <em>29</em> (8), 327–338. <a
                    href="https://doi.org/10.1093/protein/gzw026">https://doi.org/10.1093/protein/gzw026</a>.</p>
        </div>
        <div id="ref-iqbHG3rv">
            <p>(3) Andreini, C.; Bertini, I.; Cavallaro, G.; Holliday, G. L.; Thornton, J. M. Metal Ions in Biological
                Catalysis: From Enzyme Databases to General Principles. <em>J Biol Inorg Chem</em>
                <strong>2008</strong>, <em>13</em> (8), 1205–1218. <a
                    href="https://doi.org/10.1007/s00775-008-0404-5">https://doi.org/10.1007/s00775-008-0404-5</a>.</p>
        </div>
        <div id="ref-4sXp2sDw">
            <p>(4) Koohi-Moghadam, M.; Wang, H.; Wang, Y.; Yang, X.; Li, H.; Wang, J.; Sun, H. Predicting
                Disease-Associated Mutation of Metal-Binding Sites in Proteins Using a Deep Learning Approach. <em>Nat
                    Mach Intell</em> <strong>2019</strong>, <em>1</em> (12), 561–567. <a
                    href="https://doi.org/10.1038/s42256-019-0119-z">https://doi.org/10.1038/s42256-019-0119-z</a>.</p>
        </div>
        <div id="ref-oOm5j05D">
            <p>(5) Studer, S.; Hansen, D. A.; Pianowski, Z. L.; Mittl, P. R. E.; Debon, A.; Guffy, S. L.; Der, B. S.;
                Kuhlman, B.; Hilvert, D. Evolution of a Highly Active and Enantiospecific Metalloenzyme from Short
                Peptides. <em>Science</em> <strong>2018</strong>, <em>362</em> (6420), 1285–1288. <a
                    href="https://doi.org/10.1126/science.aau3744">https://doi.org/10.1126/science.aau3744</a>.</p>
        </div>
        <div id="ref-LAjO1aDL">
            <p>(6) Key, H. M.; Dydio, P.; Clark, D. S.; Hartwig, J. F. Abiological Catalysis by Artificial Haem Proteins
                Containing Noble Metals in Place of Iron. <em>Nature</em> <strong>2016</strong>, <em>534</em> (7608),
                534–537. <a href="https://doi.org/10.1038/nature17968">https://doi.org/10.1038/nature17968</a>.</p>
        </div>
        <div id="ref-13w4SiWCx">
            <p>(7) Chalkley, M. J.; Mann, S. I.; DeGrado, W. F. De Novo Metalloprotein Design. <em>Nat Rev Chem</em>
                <strong>2021</strong>, <em>6</em> (1), 31–50. <a
                    href="https://doi.org/10.1038/s41570-021-00339-5">https://doi.org/10.1038/s41570-021-00339-5</a>.
            </p>
        </div>
        <div id="ref-khOeAdye">
            <p>(8) Der, B. S.; Edwards, D. R.; Kuhlman, B. Catalysis by a de Novo Zinc-Mediated Protein Interface:
                Implications for Natural Enzyme Evolution and Rational Enzyme Engineering. <em>Biochemistry</em>
                <strong>2012</strong>, <em>51</em> (18), 3933–3940. <a
                    href="https://doi.org/10.1021/bi201881p">https://doi.org/10.1021/bi201881p</a>.</p>
        </div>
        <div id="ref-78bpCEHk">
            <p>(9) Fujieda, N.; Schätti, J.; Stuttfeld, E.; Ohkubo, K.; Maier, T.; Fukuzumi, S.; Ward, T. R. Enzyme
                Repurposing of a Hydrolase as an Emergent Peroxidase Upon Metal Binding. <em>Chem. Sci.</em>
                <strong>2015</strong>, <em>6</em> (7), 4060–4065. <a
                    href="https://doi.org/10.1039/c5sc01065a">https://doi.org/10.1039/c5sc01065a</a>.</p>
        </div>
        <div id="ref-ROLANnJl">
            <p>(10) Brodin, J. D.; Ambroggio, X. I.; Tang, C.; Parent, K. N.; Baker, T. S.; Tezcan, F. A.
                Metal-Directed, Chemically Tunable Assembly of One-, Two- and Three-Dimensional Crystalline Protein
                Arrays. <em>Nature Chem</em> <strong>2012</strong>, <em>4</em> (5), 375–382. <a
                    href="https://doi.org/10.1038/nchem.1290">https://doi.org/10.1038/nchem.1290</a>.</p>
        </div>
        <div id="ref-Lj2sbjzc">
            <p>(11) Der, B. S.; Machius, M.; Miley, M. J.; Mills, J. L.; Szyperski, T.; Kuhlman, B. Metal-Mediated
                Affinity and Orientation Specificity in a Computationally Designed Protein Homodimer. <em>J. Am. Chem.
                    Soc.</em> <strong>2011</strong>, <em>134</em> (1), 375–385. <a
                    href="https://doi.org/10.1021/ja208015j">https://doi.org/10.1021/ja208015j</a>.</p>
        </div>
        <div id="ref-17JMFzNJo">
            <p>(12) Salgado, E. N.; Radford, R. J.; Tezcan, F. A. Metal-Directed Protein Self-Assembly. <em>Acc. Chem.
                    Res.</em> <strong>2010</strong>, <em>43</em> (5), 661–672. <a
                    href="https://doi.org/10.1021/ar900273t">https://doi.org/10.1021/ar900273t</a>.</p>
        </div>
        <div id="ref-io987fKE">
            <p>(13) Kakkis, A.; Gagnon, D.; Esselborn, J.; Britt, R. D.; Tezcan, F. A. Metal‐Templated Design of
                Chemically Switchable Protein Assemblies with High‐Affinity Coordination Sites. <em>Angew. Chem. Int.
                    Ed.</em> <strong>2020</strong>, <em>59</em> (49), 21940–21944. <a
                    href="https://doi.org/10.1002/anie.202009226">https://doi.org/10.1002/anie.202009226</a>.</p>
        </div>
        <div id="ref-JXKFLH3I">
            <p>(14) Zastrow, M. L.; Peacock, A. F. A.; Stuckey, J. A.; Pecoraro, V. L. Hydrolytic Catalysis and
                Structural Stabilization in a Designed Metalloprotein. <em>Nature Chem</em> <strong>2011</strong>,
                <em>4</em> (2), 118–123. <a
                    href="https://doi.org/10.1038/nchem.1201">https://doi.org/10.1038/nchem.1201</a>.</p>
        </div>
        <div id="ref-1FR3e6DyA">
            <p>(15) Song, L. F.; Sengupta, A.; Merz, K. M., Jr. Thermodynamics of Transition Metal Ion Binding to
                Proteins. <em>J. Am. Chem. Soc.</em> <strong>2020</strong>, <em>142</em> (13), 6365–6374. <a
                    href="https://doi.org/10.1021/jacs.0c01329">https://doi.org/10.1021/jacs.0c01329</a>.</p>
        </div>
        <div id="ref-1uPiiMnD">
            <p>(16) Alford, R. F.; Leaver-Fay, A.; Jeliazkov, J. R.; O’Meara, M. J.; DiMaio, F. P.; Park, H.;
                Shapovalov, M. V.; Renfrew, P. D.; Mulligan, V. K.; Kappel, K.; Labonte, J. W.; Pacella, M. S.; Bonneau,
                R.; Bradley, P.; Dunbrack, R. L., Jr.; Das, R.; Baker, D.; Kuhlman, B.; Kortemme, T.; Gray, J. J. The
                Rosetta All-Atom Energy Function for Macromolecular Modeling and Design. <em>J. Chem. Theory
                    Comput.</em> <strong>2017</strong>, <em>13</em> (6), 3031–3048. <a
                    href="https://doi.org/10.1021/acs.jctc.7b00125">https://doi.org/10.1021/acs.jctc.7b00125</a>.</p>
        </div>
        <div id="ref-5Vxh8hay">
            <p>(17) Brunk, E.; Rothlisberger, U. Mixed Quantum Mechanical/Molecular Mechanical Molecular Dynamics
                Simulations of Biological Systems in Ground and Electronically Excited States. <em>Chem. Rev.</em>
                <strong>2015</strong>, <em>115</em> (12), 6217–6263. <a
                    href="https://doi.org/10.1021/cr500628b">https://doi.org/10.1021/cr500628b</a>.</p>
        </div>
        <div id="ref-IXq7vmo3">
            <p>(18) Yang, Z.; Twidale, R. M.; Gervasoni, S.; Suardíaz, R.; Colenso, C. K.; Lang, E. J. M.; Spencer, J.;
                Mulholland, A. J. Multiscale Workflow for Modeling Ligand Complexes of Zinc Metalloproteins. <em>J.
                    Chem. Inf. Model.</em> <strong>2021</strong>, <em>61</em> (11), 5658–5672. <a
                    href="https://doi.org/10.1021/acs.jcim.1c01109">https://doi.org/10.1021/acs.jcim.1c01109</a>.</p>
        </div>
        <div id="ref-nlfa683y">
            <p>(19) Bozkurt, E.; Perez, M. A. S.; Hovius, R.; Browning, N. J.; Rothlisberger, U. Genetic Algorithm Based
                Design and Experimental Characterization of a Highly Thermostable Metalloprotein. <em>J. Am. Chem.
                    Soc.</em> <strong>2018</strong>, <em>140</em> (13), 4517–4521. <a
                    href="https://doi.org/10.1021/jacs.7b10660">https://doi.org/10.1021/jacs.7b10660</a>.</p>
        </div>
        <div id="ref-b5EJeHX3">
            <p>(20). <a href="https://doi.org/10.3389/fchem.2021.692200">https://doi.org/10.3389/fchem.2021.692200</a>.
            </p>
        </div>
        <div id="ref-qNdK0F1w">
            <p>(21) Passerini, A.; Andreini, C.; Menchetti, S.; Rosato, A.; Frasconi, P. Predicting Zinc Binding at the
                Proteome Level. <em>BMC Bioinformatics</em> <strong>2007</strong>, <em>8</em> (1). <a
                    href="https://doi.org/10.1186/1471-2105-8-39">https://doi.org/10.1186/1471-2105-8-39</a>.</p>
        </div>
        <div id="ref-KNmKWbrP">
            <p>(22) Hu, X.; Dong, Q.; Yang, J.; Zhang, Y. Recognizing Metal and Acid Radical Ion-Binding Sites by
                Integratingab Initiomodeling with Template-Based Transferals. <em>Bioinformatics</em>
                <strong>2016</strong>, <em>32</em> (21), 3260–3269. <a
                    href="https://doi.org/10.1093/bioinformatics/btw396">https://doi.org/10.1093/bioinformatics/btw396</a>.
            </p>
        </div>
        <div id="ref-1HMhB3vxM">
            <p>(23) Lin, Y.-F.; Cheng, C.-W.; Shih, C.-S.; Hwang, J.-K.; Yu, C.-S.; Lu, C.-H. MIB: Metal Ion-Binding
                Site Prediction and Docking Server. <em>J. Chem. Inf. Model.</em> <strong>2016</strong>, <em>56</em>
                (12), 2287–2291. <a
                    href="https://doi.org/10.1021/acs.jcim.6b00407">https://doi.org/10.1021/acs.jcim.6b00407</a>.</p>
        </div>
        <div id="ref-13fLqNwbD">
            <p>(24) Hekkelman, M. L.; de Vries, I.; Joosten, R. P.; Perrakis, A. AlphaFill: Enriching the AlphaFold
                Models with Ligands and Co-Factors, 2021. <a
                    href="https://doi.org/10.1101/2021.11.26.470110">https://doi.org/10.1101/2021.11.26.470110</a>.</p>
        </div>
        <div id="ref-W4f7UG1p">
            <p>(25) Brylinski, M.; Skolnick, J. FINDSITE-Metal: Integrating Evolutionary Information and Machine
                Learning for Structure-Based Metal-Binding Site Prediction at the Proteome Level. <em>Proteins</em>
                <strong>2010</strong>, <em>79</em> (3), 735–751. <a
                    href="https://doi.org/10.1002/prot.22913">https://doi.org/10.1002/prot.22913</a>.</p>
        </div>
        <div id="ref-iHxzzTCG">
            <p>(26) Sánchez-Aparicio, J.-E.; Tiessler-Sala, L.; Velasco-Carneros, L.; Roldán-Martín, L.; Sciortino, G.;
                Maréchal, J.-D. BioMetAll: Identifying Metal-Binding Sites in Proteins from Backbone Preorganization.
                <em>J. Chem. Inf. Model.</em> <strong>2020</strong>, <em>61</em> (1), 311–323. <a
                    href="https://doi.org/10.1021/acs.jcim.0c00827">https://doi.org/10.1021/acs.jcim.0c00827</a>.</p>
        </div>
        <div id="ref-13sNVOrHB">
            <p>(27) Haberal, İ.; Oğul, H. Prediction of Protein Metal Binding Sites Using Deep Neural Networks. <em>Mol.
                    Inf.</em> <strong>2019</strong>, <em>38</em> (7), 1800169. <a
                    href="https://doi.org/10.1002/minf.201800169">https://doi.org/10.1002/minf.201800169</a>.</p>
        </div>
        <div id="ref-yZfcMIwh">
            <p>(28) Jumper, J.; Evans, R.; Pritzel, A.; Green, T.; Figurnov, M.; Ronneberger, O.; Tunyasuvunakool, K.;
                Bates, R.; Žídek, A.; Potapenko, A.; Bridgland, A.; Meyer, C.; Kohl, S. A. A.; Ballard, A. J.; Cowie,
                A.; Romera-Paredes, B.; Nikolov, S.; Jain, R.; Adler, J.; Back, T.; Petersen, S.; Reiman, D.; Clancy,
                E.; Zielinski, M.; Steinegger, M.; Pacholska, M.; Berghammer, T.; Bodenstein, S.; Silver, D.; Vinyals,
                O.; Senior, A. W.; Kavukcuoglu, K.; Kohli, P.; Hassabis, D. Highly Accurate Protein Structure Prediction
                with AlphaFold. <em>Nature</em> <strong>2021</strong>, <em>596</em> (7873), 583–589. <a
                    href="https://doi.org/10.1038/s41586-021-03819-2">https://doi.org/10.1038/s41586-021-03819-2</a>.
            </p>
        </div>
        <div id="ref-rnu2o8C9">
            <p>(29) Baek, M.; DiMaio, F.; Anishchenko, I.; Dauparas, J.; Ovchinnikov, S.; Lee, G. R.; Wang, J.; Cong,
                Q.; Kinch, L. N.; Schaeffer, R. D.; Millán, C.; Park, H.; Adams, C.; Glassman, C. R.; DeGiovanni, A.;
                Pereira, J. H.; Rodrigues, A. V.; van Dijk, A. A.; Ebrecht, A. C.; Opperman, D. J.; Sagmeister, T.;
                Buhlheller, C.; Pavkov-Keller, T.; Rathinaswamy, M. K.; Dalwadi, U.; Yip, C. K.; Burke, J. E.; Garcia,
                K. C.; Grishin, N. V.; Adams, P. D.; Read, R. J.; Baker, D. Accurate Prediction of Protein Structures
                and Interactions Using a Three-Track Neural Network. <em>Science</em> <strong>2021</strong>,
                <em>373</em> (6557), 871–876. <a
                    href="https://doi.org/10.1126/science.abj8754">https://doi.org/10.1126/science.abj8754</a>.</p>
        </div>
        <div id="ref-ls5kyxZ3">
            <p>(30) Torng, W.; Altman, R. B. 3D Deep Convolutional Neural Networks for Amino Acid Environment Similarity
                Analysis. <em>BMC Bioinformatics</em> <strong>2017</strong>, <em>18</em> (1). <a
                    href="https://doi.org/10.1186/s12859-017-1702-0">https://doi.org/10.1186/s12859-017-1702-0</a>.</p>
        </div>
        <div id="ref-F7P74Etp">
            <p>(31) Shroff, R.; Cole, A. W.; Diaz, D. J.; Morrow, B. R.; Donnell, I.; Annapareddy, A.; Gollihar, J.;
                Ellington, A. D.; Thyer, R. Discovery of Novel Gain-of-Function Mutations Guided by Structure-Based Deep
                Learning. <em>ACS Synth. Biol.</em> <strong>2020</strong>, <em>9</em> (11), 2927–2935. <a
                    href="https://doi.org/10.1021/acssynbio.0c00345">https://doi.org/10.1021/acssynbio.0c00345</a>.</p>
        </div>
        <div id="ref-6szq3cwi">
            <p>(32) Anand, N.; Eguchi, R.; Mathews, I. I.; Perez, C. P.; Derry, A.; Altman, R. B.; Huang, P.-S. Protein
                Sequence Design with a Learned Potential. <em>Nat Commun</em> <strong>2022</strong>, <em>13</em> (1). <a
                    href="https://doi.org/10.1038/s41467-022-28313-9">https://doi.org/10.1038/s41467-022-28313-9</a>.
            </p>
        </div>
        <div id="ref-SGggOw7Z">
            <p>(33) Torng, W.; Altman, R. B. High Precision Protein Functional Site Detection Using 3D Convolutional
                Neural Networks. <em>Bioinformatics</em> <strong>2018</strong>, <em>35</em> (9), 1503–1512. <a
                    href="https://doi.org/10.1093/bioinformatics/bty813">https://doi.org/10.1093/bioinformatics/bty813</a>.
            </p>
        </div>
        <div id="ref-MKP16DSu">
            <p>(34) Feehan, R.; Franklin, M. W.; Slusky, J. S. G. Machine Learning Differentiates Enzymatic and
                Non-Enzymatic Metals in Proteins. <em>Nat Commun</em> <strong>2021</strong>, <em>12</em> (1). <a
                    href="https://doi.org/10.1038/s41467-021-24070-3">https://doi.org/10.1038/s41467-021-24070-3</a>.
            </p>
        </div>
        <div id="ref-q224Kv8w">
            <p>(35) Renaud, N.; Geng, C.; Georgievska, S.; Ambrosetti, F.; Ridder, L.; Marzella, D. F.; Réau, M. F.;
                Bonvin, A. M. J. J.; Xue, L. C. DeepRank: A Deep Learning Framework for Data Mining 3D Protein-Protein
                Interfaces. <em>Nat Commun</em> <strong>2021</strong>, <em>12</em> (1). <a
                    href="https://doi.org/10.1038/s41467-021-27396-0">https://doi.org/10.1038/s41467-021-27396-0</a>.
            </p>
        </div>
        <div id="ref-yBhqGkBa">
            <p>(36) Gainza, P.; Sverrisson, F.; Monti, F.; Rodolà, E.; Boscaini, D.; Bronstein, M. M.; Correia, B. E.
                Deciphering Interaction Fingerprints from Protein Molecular Surfaces Using Geometric Deep Learning.
                <em>Nat Methods</em> <strong>2019</strong>, <em>17</em> (2), 184–192. <a
                    href="https://doi.org/10.1038/s41592-019-0666-6">https://doi.org/10.1038/s41592-019-0666-6</a>.</p>
        </div>
        <div id="ref-Gq7n6vhH">
            <p>(37) Jiménez, J.; Doerr, S.; Martínez-Rosell, G.; Rose, A. S.; De Fabritiis, G. DeepSite: Protein-Binding
                Site Predictor Using 3D-Convolutional Neural Networks. <em>Bioinformatics</em> <strong>2017</strong>,
                <em>33</em> (19), 3036–3042. <a
                    href="https://doi.org/10.1093/bioinformatics/btx350">https://doi.org/10.1093/bioinformatics/btx350</a>.
            </p>
        </div>
        <div id="ref-mVGaXlum">
            <p>(38) Skalic, M.; Varela-Rial, A.; Jiménez, J.; Martínez-Rosell, G.; De Fabritiis, G. LigVoxel: Inpainting
                Binding Pockets Using 3D-Convolutional Neural Networks. <em>Bioinformatics</em> <strong>2018</strong>,
                <em>35</em> (2), 243–250. <a
                    href="https://doi.org/10.1093/bioinformatics/bty583">https://doi.org/10.1093/bioinformatics/bty583</a>.
            </p>
        </div>
        <div id="ref-13hjzBbje">
            <p>(39) Stärk, H.; Ganea, O.-E.; Pattanaik, L.; Barzilay, R.; Jaakkola, T. EquiBind: Geometric Deep Learning
                for Drug Binding Structure Prediction. arXiv 2022. <a
                    href="https://doi.org/10.48550/arxiv.2202.05146">https://doi.org/10.48550/arxiv.2202.05146</a>.</p>
        </div>
        <div id="ref-9UNjBvCL">
            <p>(40) Park, S.; Seok, C. GalaxyWater-CNN: Prediction of Water Positions on the Protein Structure by a
                3D-Convolutional Neural Network. <em>J. Chem. Inf. Model.</em> <strong>2022</strong>, <em>62</em> (13),
                3157–3168. <a
                    href="https://doi.org/10.1021/acs.jcim.2c00306">https://doi.org/10.1021/acs.jcim.2c00306</a>.</p>
        </div>
        <div id="ref-9K88m3YT">
            <p>(41) Li, B.; Yang, Y. T.; Capra, J. A.; Gerstein, M. B. Predicting Changes in Protein Thermodynamic
                Stability Upon Point Mutation with Deep 3D Convolutional Neural Networks. <em>PLoS Comput Biol</em>
                <strong>2020</strong>, <em>16</em> (11), e1008291. <a
                    href="https://doi.org/10.1371/journal.pcbi.1008291">https://doi.org/10.1371/journal.pcbi.1008291</a>.
            </p>
        </div>
        <div id="ref-k859wJxx">
            <p>(42) Lu, C.-H.; Lin, Y.-F.; Lin, J.-J.; Yu, C.-S. Prediction of Metal Ion–Binding Sites in Proteins Using
                the Fragment Transformation Method. <em>PLoS ONE</em> <strong>2012</strong>, <em>7</em> (6), e39252. <a
                    href="https://doi.org/10.1371/journal.pone.0039252">https://doi.org/10.1371/journal.pone.0039252</a>.
            </p>
        </div>
        <div id="ref-Qubi5dZD">
            <p>(43) Minasov, G.; Vorontsov, I. I.; Shuvalova, L.; Brunzelle, J. S.; Kiryukhina, O.; Collart, F. R.;
                Joachimiak, A.; Anderson, W. F.;. Crystal Structure of Unknown Conserved ybaA Protein from Shigella
                Flexneri, 2007. <a href="https://doi.org/10.2210/pdb2okq/pdb">https://doi.org/10.2210/pdb2okq/pdb</a>.
            </p>
        </div>
        <div id="ref-g7lHXZp5">
            <p>(44) Itoh, T.; Nakagawa, E.; Yoda, M.; Nakaichi, A.; Hibi, T.; Kimoto, H. Crystal Structure of Alginate
                Lyase from Paenibacillus Sp. Str. FPU-7, 2019. <a
                    href="https://doi.org/10.2210/pdb6kfn/pdb">https://doi.org/10.2210/pdb6kfn/pdb</a>.</p>
        </div>
        <div id="ref-w9ECLZDp">
            <p>(45) Adams, C. M.; Eckenroth, B. E.; Doublie, S. Structure of the Clostridium Perfringens CspB Protease,
                2013. <a href="https://doi.org/10.2210/pdb4i0w/pdb">https://doi.org/10.2210/pdb4i0w/pdb</a>.</p>
        </div>
        <div id="ref-wW700ShK">
            <p>(46) Davies, C. W.; Das, C. The Crystal Structure of a E280A Mutant of the Catalytic Domain of AMSH,
                2011. <a href="https://doi.org/10.2210/pdb3rzv/pdb">https://doi.org/10.2210/pdb3rzv/pdb</a>.</p>
        </div>
        <div id="ref-RFwEiPTW">
            <p>(47) Rana, M. S.; Kumar, P.; Lee, C.-J.; Verardi, R.; Rajashankar, K. R.; Banerjee, A. Fatty Acyl
                Recognition and Transfer by an Integral Membrane <em>S</em> -Acyltransferase. <em>Science</em>
                <strong>2018</strong>, <em>359</em> (6372). <a
                    href="https://doi.org/10.1126/science.aao6326">https://doi.org/10.1126/science.aao6326</a>.</p>
        </div>
        <div id="ref-jyFnBdWm">
            <p>(48) Hakansson, K.; Carlsson, M.; Svensson, L. A.; Liljas, A. STRUCTURE OF NATIVE AND APO CARBONIC
                ANHYDRASE II AND SOME OF ITS ANION-LIGAND COMPLEXES, 1993. <a
                    href="https://doi.org/10.2210/pdb2cba/pdb">https://doi.org/10.2210/pdb2cba/pdb</a>.</p>
        </div>
        <div id="ref-13XuOF3Jj">
            <p>(49) Hunt, J. B.; Neece, S. H.; Ginsburg, A. The Use of 4-(2-Pyridylazo)resorcinol in Studies of Zinc
                Release from Escherichia Coli Aspartate Transcarbamoylase. <em>Anal Biochem</em> <strong>1985</strong>,
                <em>146</em> (1), 150–157. <a
                    href="https://doi.org/10.1016/0003-2697(85)90409-9">https://doi.org/10.1016/0003-2697(85)90409-9</a>.
            </p>
        </div>
        <div id="ref-pchdLF0k">
            <p>(50) Pang, Y. P.; Xu, K.; Yazal, J. E.; Prendergas, F. G. Successful Molecular Dynamics Simulation of the
                Zinc-Bound Farnesyltransferase Using the Cationic Dummy Atom Approach. <em>Protein Sci</em>
                <strong>2000</strong>, <em>9</em> (10), 1857–1865.</p>
        </div>
        <div id="ref-D3VQQdjb">
            <p>(51) Kiefer, L. L.; Fierke, C. A. Functional Characterization of Human Carbonic Anhydrase II Variants
                with Altered Zinc Binding Sites. <em>Biochemistry</em> <strong>1994</strong>, <em>33</em> (51),
                15233–15240. <a href="https://doi.org/10.1021/bi00255a003">https://doi.org/10.1021/bi00255a003</a>.</p>
        </div>
        <div id="ref-w3tzhp92">
            <p>(52) Kiefer, L. L.; Ippolito, J. A.; Fierke, C. A.; Christianson, D. W. Redesigning the Zinc Binding Site
                of Human Carbonic Anhydrase II: Structure of a His2Asp-Zn2+ Metal Coordination Polyhedron. <em>J. Am.
                    Chem. Soc.</em> <strong>1993</strong>, <em>115</em> (26), 12581–12582. <a
                    href="https://doi.org/10.1021/ja00079a046">https://doi.org/10.1021/ja00079a046</a>.</p>
        </div>
        <div id="ref-iu33cNMg">
            <p>(53) Ippolito, J. A.; Christianson, D. W. Structure of an Engineered His3 Cys Zinc Binding Site in Human
                Carbonic Anhydrase II. <em>Biochemistry</em> <strong>1993</strong>, <em>32</em> (38), 9901–9905. <a
                    href="https://doi.org/10.1021/bi00089a005">https://doi.org/10.1021/bi00089a005</a>.</p>
        </div>
        <div id="ref-148Si7kTL">
            <p>(54) Ippolito, J. A.; Baird, T. T., Jr; McGee, S. A.; Christianson, D. W.; Fierke, C. A.
                Structure-Assisted Redesign of a Protein-Zinc-Binding Site with Femtomolar Affinity. <em>Proc. Natl.
                    Acad. Sci. U.S.A.</em> <strong>1995</strong>, <em>92</em> (11), 5017–5021. <a
                    href="https://doi.org/10.1073/pnas.92.11.5017">https://doi.org/10.1073/pnas.92.11.5017</a>.</p>
        </div>
        <div id="ref-TNbPH3Y1">
            <p>(55) Huang, C.-c.; Lesburg, C. A.; Kiefer, L. L.; Fierke, C. A.; Christianson, D. W. Reversal of the
                Hydrogen Bond to Zinc Ligand Histidine-119 Dramatically Diminishes Catalysis and Enhances Metal
                Equilibration Kinetics in Carbonic Anhydrase II. <em>Biochemistry</em> <strong>1996</strong>,
                <em>35</em> (11), 3439–3446. <a
                    href="https://doi.org/10.1021/bi9526692">https://doi.org/10.1021/bi9526692</a>.</p>
        </div>
        <div id="ref-Wt0ducot">
            <p>(56) Laitaoja, M.; Valjakka, J.; Jänis, J. Zinc Coordination Spheres in Protein Structures. <em>Inorg.
                    Chem.</em> <strong>2013</strong>, <em>52</em> (19), 10983–10991. <a
                    href="https://doi.org/10.1021/ic401072d">https://doi.org/10.1021/ic401072d</a>.</p>
        </div>
        <div id="ref-NifwNdQd">
            <p>(57) Savage, H.; Wlodawer, A. Determination of Water Structure Around Biomolecules Using X-Ray and
                Neutron Diffraction Methods. <em>Methods Enzymol</em> <strong>1986</strong>, <em>127</em>, 162–183. <a
                    href="https://doi.org/10.1016/0076-6879(86)27014-7">https://doi.org/10.1016/0076-6879(86)27014-7</a>.
            </p>
        </div>
        <div id="ref-10C4imKQR">
            <p>(58) Morozenko, A.; Stuchebrukhov, A. A. Dowser++, a New Method of Hydrating Protein Structures.
                <em>Proteins</em> <strong>2016</strong>, <em>84</em> (10), 1347–1357. <a
                    href="https://doi.org/10.1002/prot.25081">https://doi.org/10.1002/prot.25081</a>.</p>
        </div>
        <div id="ref-izkvZoLX">
            <p>(59) Sridhar, A.; Ross, G. A.; Biggin, P. C. Waterdock 2.0: Water Placement Prediction for
                Holo-Structures with a Pymol Plugin. <em>PLoS ONE</em> <strong>2017</strong>, <em>12</em> (2), e0172743.
                <a href="https://doi.org/10.1371/journal.pone.0172743">https://doi.org/10.1371/journal.pone.0172743</a>.
            </p>
        </div>
        <div id="ref-Fc42fl1x">
            <p>(60) Satorras, V. G.; Hoogeboom, E.; Welling, M. E(n) Equivariant Graph Neural Networks. arXiv 2021. <a
                    href="https://doi.org/10.48550/arxiv.2102.09844">https://doi.org/10.48550/arxiv.2102.09844</a>.</p>
        </div>
        <div id="ref-cbwvbkS6">
            <p>(61) Gligorijević, V.; Berenberg, D.; Ra, S.; Watkins, A.; Kelow, S.; Cho, K.; Bonneau, R.
                Function-Guided Protein Design by Deep Manifold Sampling, 2021. <a
                    href="https://doi.org/10.1101/2021.12.22.473759">https://doi.org/10.1101/2021.12.22.473759</a>.</p>
        </div>
        <div id="ref-Gj0pG9O5">
            <p>(62) Greener, J. G.; Moffat, L.; Jones, D. T. Design of Metalloproteins and Novel Protein Folds Using
                Variational Autoencoders. <em>Sci Rep</em> <strong>2018</strong>, <em>8</em> (1). <a
                    href="https://doi.org/10.1038/s41598-018-34533-1">https://doi.org/10.1038/s41598-018-34533-1</a>.
            </p>
        </div>
        <div id="ref-TlDhIlX">
            <p>(63) Song, H.; Wilson, D. L.; Farquhar, E. R.; Lewis, E. A.; Emerson, J. P. Revisiting Zinc Coordination
                in Human Carbonic Anhydrase II. <em>Inorg. Chem.</em> <strong>2012</strong>, <em>51</em> (20),
                11098–11105. <a href="https://doi.org/10.1021/ic301645j">https://doi.org/10.1021/ic301645j</a>.</p>
        </div>
        <div id="ref-RJ44nj3R">
            <p>(64) Handel, T. M.; Williams, S. A.; DeGrado, W. F. Metal Ion-Dependent Modulation of the Dynamics of a
                Designed Protein. <em>Science</em> <strong>1993</strong>, <em>261</em> (5123), 879–885. <a
                    href="https://doi.org/10.1126/science.8346440">https://doi.org/10.1126/science.8346440</a>.</p>
        </div>
        <div id="ref-15drt0nXZ">
            <p>(65) Arnold, F. H.; Haymore, B. L. Engineered Metal-Binding Proteins: Purification to Protein Folding.
                <em>Science</em> <strong>1991</strong>, <em>252</em> (5014), 1796–1797. <a
                    href="https://doi.org/10.1126/science.1648261">https://doi.org/10.1126/science.1648261</a>.</p>
        </div>
        <div id="ref-85PeY3aG">
            <p>(66) Krantz, B. A.; Sosnick, T. R. <em>Nat. Struct Biol.</em> <strong>2001</strong>, <em>8</em> (12),
                1042–1047. <a href="https://doi.org/10.1038/nsb723">https://doi.org/10.1038/nsb723</a>.</p>
        </div>
        <div id="ref-10bYywzs6">
            <p>(67) Tunyasuvunakool, K.; Adler, J.; Wu, Z.; Green, T.; Zielinski, M.; Žídek, A.; Bridgland, A.; Cowie,
                A.; Meyer, C.; Laydon, A.; Velankar, S.; Kleywegt, G. J.; Bateman, A.; Evans, R.; Pritzel, A.; Figurnov,
                M.; Ronneberger, O.; Bates, R.; Kohl, S. A. A.; Potapenko, A.; Ballard, A. J.; Romera-Paredes, B.;
                Nikolov, S.; Jain, R.; Clancy, E.; Reiman, D.; Petersen, S.; Senior, A. W.; Kavukcuoglu, K.; Birney, E.;
                Kohli, P.; Jumper, J.; Hassabis, D. Highly Accurate Protein Structure Prediction for the Human Proteome.
                <em>Nature</em> <strong>2021</strong>, <em>596</em> (7873), 590–596. <a
                    href="https://doi.org/10.1038/s41586-021-03828-1">https://doi.org/10.1038/s41586-021-03828-1</a>.
            </p>
        </div>
        <div id="ref-8cYXwVxI">
            <p>(68) Berman, H. M. The Protein Data Bank. <em>Nucleic Acids Research</em> <strong>2000</strong>,
                <em>28</em> (1), 235–242. <a
                    href="https://doi.org/10.1093/nar/28.1.235">https://doi.org/10.1093/nar/28.1.235</a>.</p>
        </div>
        <div id="ref-jj6qOofE">
            <p>(69) Steinegger, M.; Söding, J. MMseqs2 Enables Sensitive Protein Sequence Searching for the Analysis of
                Massive Data Sets. <em>Nat Biotechnol</em> <strong>2017</strong>, <em>35</em> (11), 1026–1028. <a
                    href="https://doi.org/10.1038/nbt.3988">https://doi.org/10.1038/nbt.3988</a>.</p>
        </div>
        <div id="ref-FB5cckAJ">
            <p>(70) Barber-Zucker, S.; Shaanan, B.; Zarivach, R. Transition Metal Binding Selectivity in Proteins and
                Its Correlation with the Phylogenomic Classification of the Cation Diffusion Facilitator Protein Family.
                <em>Sci Rep</em> <strong>2017</strong>, <em>7</em> (1). <a
                    href="https://doi.org/10.1038/s41598-017-16777-5">https://doi.org/10.1038/s41598-017-16777-5</a>.
            </p>
        </div>
        <div id="ref-CcRTAd7h">
            <p>(71) Raschka, S. BioPandas: Working with Molecular Structures in Pandas DataFrames. <em>JOSS</em>
                <strong>2017</strong>, <em>2</em> (14), 279. <a
                    href="https://doi.org/10.21105/joss.00279">https://doi.org/10.21105/joss.00279</a>.</p>
        </div>
        <div id="ref-q2gxO5bP">
            <p>(72) Doerr, S.; Harvey, M. J.; Noé, F.; De Fabritiis, G. HTMD: High-Throughput Molecular Dynamics for
                Molecular Discovery. <em>J. Chem. Theory Comput.</em> <strong>2016</strong>, <em>12</em> (4), 1845–1852.
                <a href="https://doi.org/10.1021/acs.jctc.6b00049">https://doi.org/10.1021/acs.jctc.6b00049</a>.</p>
        </div>
        <div id="ref-19MS2ZVR1">
            <p>(73) Moritz, P.; Nishihara, R.; Wang, S.; Tumanov, A.; Liaw, R.; Liang, E.; Elibol, M.; Yang, Z.; Paul,
                W.; Jordan, M. I.; Stoica, I. Ray: A Distributed Framework for Emerging AI Applications. arXiv 2017. <a
                    href="https://doi.org/10.48550/arxiv.1712.05889">https://doi.org/10.48550/arxiv.1712.05889</a>.</p>
        </div>
        <div id="ref-iOWq2JB8">
            <p>(74) Paszke, A.; Gross, S.; Massa, F.; Lerer, A.; Bradbury, J.; Chanan, G.; Killeen, T.; Lin, Z.;
                Gimelshein, N.; Antiga, L.; Desmaison, A.; Köpf, A.; Yang, E.; DeVito, Z.; Raison, M.; Tejani, A.;
                Chilamkurthy, S.; Steiner, B.; Fang, L.; Bai, J.; Chintala, S. PyTorch: An Imperative Style,
                High-Performance Deep Learning Library. arXiv 2019. <a
                    href="https://doi.org/10.48550/arxiv.1912.01703">https://doi.org/10.48550/arxiv.1912.01703</a>.</p>
        </div>
        <div id="ref-hchMCnC1">
            <p>(75) de Boer, P.-T.; Kroese, D. P.; Mannor, S.; Rubinstein, R. Y. A Tutorial on the Cross-Entropy Method.
                <em>Ann Oper Res</em> <strong>2005</strong>, <em>134</em> (1), 19–67. <a
                    href="https://doi.org/10.1007/s10479-005-5724-z">https://doi.org/10.1007/s10479-005-5724-z</a>.</p>
        </div>
        <div id="ref-8Miti2Gz">
            <p>(76) Virtanen, P.; Gommers, R.; Oliphant, T. E.; Haberland, M.; Reddy, T.; Cournapeau, D.; Burovski, E.;
                Peterson, P.; Weckesser, W.; Bright, J.; van der Walt, S. J.; Brett, M.; Wilson, J.; Millman, K. J.;
                Mayorov, N.; Nelson, A. R. J.; Jones, E.; Kern, R.; Larson, E.; Carey, C. J.; Polat, İ.; Feng, Y.;
                Moore, E. W.; VanderPlas, J.; Laxalde, D.; Perktold, J.; Cimrman, R.; Henriksen, I.; Quintero, E. A.;
                Harris, C. R.; Archibald, A. M.; Ribeiro, A. H.; Pedregosa, F.; van Mulbregt, P.; Vijaykumar, A.;
                Bardelli, A. P.; Rothberg, A.; Hilboll, A.; Kloeckner, A.; Scopatz, A.; Lee, A.; Rokem, A.; Woods, C.
                N.; Fulton, C.; Masson, C.; Häggström, C.; Fitzgerald, C.; Nicholson, D. A.; Hagen, D. R.; Pasechnik, D.
                V.; Olivetti, E.; Martin, E.; Wieser, E.; Silva, F.; Lenders, F.; Wilhelm, F.; Young, G.; Price, G. A.;
                Ingold, G.-L.; Allen, G. E.; Lee, G. R.; Audren, H.; Probst, I.; Dietrich, J. P.; Silterra, J.; Webber,
                J. T.; Slavič, J.; Nothman, J.; Buchner, J.; Kulick, J.; Schönberger, J. L.; de Miranda Cardoso, J. V.;
                Reimer, J.; Harrington, J.; Rodríguez, J. L. C.; Nunez-Iglesias, J.; Kuczynski, J.; Tritz, K.; Thoma,
                M.; Newville, M.; Kümmerer, M.; Bolingbroke, M.; Tartre, M.; Pak, M.; Smith, N. J.; Nowaczyk, N.;
                Shebanov, N.; Pavlyk, O.; Brodtkorb, P. A.; Lee, P.; McGibbon, R. T.; Feldbauer, R.; Lewis, S.; Tygier,
                S.; Sievert, S.; Vigna, S.; Peterson, S.; More, S.; Pudlik, T.; Oshima, T.; Pingel, T. J.; Robitaille,
                T. P.; Spura, T.; Jones, T. R.; Cera, T.; Leslie, T.; Zito, T.; Krauss, T.; Upadhyay, U.; Halchenko, Y.
                O.; Vázquez-Baeza, Y.;. SciPy 1.0: Fundamental Algorithms for Scientific Computing in Python. <em>Nat
                    Methods</em> <strong>2020</strong>, <em>17</em> (3), 261–272. <a
                    href="https://doi.org/10.1038/s41592-019-0686-2">https://doi.org/10.1038/s41592-019-0686-2</a>.</p>
        </div>
        <div id="ref-AujvwLp6">
            <p>(77) Pedregosa, F.; Varoquaux, G.; Gramfort, A.; Michel, V.; Thirion, B.; Grisel, O.; Blondel, M.;
                Müller, A.; Nothman, J.; Louppe, G.; Prettenhofer, P.; Weiss, R.; Dubourg, V.; Vanderplas, J.; Passos,
                A.; Cournapeau, D.; Brucher, M.; Perrot, M.; Duchesnay, É. Scikit-Learn: Machine Learning in Python.
                <em>arXiv</em> <strong>2012</strong>. <a
                    href="https://doi.org/10.48550/arxiv.1201.0490">https://doi.org/10.48550/arxiv.1201.0490</a>.</p>
        </div>
        <div id="ref-PEXAwV3k">
            <p>(78) Humphrey, W.; Dalke, A.; Schulten, K. VMD: Visual Molecular Dynamics. <em>J Mol Graph</em>
                <strong>1996</strong>, <em>14</em> (1), 33–38, 27–28. <a
                    href="https://doi.org/10.1016/0263-7855(96)00018-5">https://doi.org/10.1016/0263-7855(96)00018-5</a>.
            </p>
        </div>
        <div id="ref-BguJBlg6">
            <p>(79) Rego, N.; Koes, D. 3Dmol.js: Molecular Visualization with WebGL. <em>Bioinformatics</em>
                <strong>2014</strong>, <em>31</em> (8), 1322–1324. <a
                    href="https://doi.org/10.1093/bioinformatics/btu829">https://doi.org/10.1093/bioinformatics/btu829</a>.
            </p>
        </div>
        <div id="ref-WjHESl20">
            <p>(80) Håkansson, K.; Carlsson, M.; Svensson, L.; Liljas, A. Structure of Native and Apo Carbonic Anhydrase
                II and Structure of Some of Its Anion-Ligand Complexes. <em>Journal of Molecular Biology</em>
                <strong>1992</strong>, <em>227</em> (4), 1192–1204. <a
                    href="https://doi.org/10.1016/0022-2836(92)90531-n">https://doi.org/10.1016/0022-2836(92)90531-n</a>.
            </p>
        </div>
        <div id="ref-Mg40Hdad">
            <p>(81) Lovell, S. C.; Word, J. M.; Richardson, J. S.; Richardson, D. C. The Penultimate Rotamer Library.
                <em>Proteins</em> <strong>2000</strong>, <em>40</em> (3), 389–408.</p>
        </div>
    </div>
    <!-- default theme -->

    <style>
        /* import google fonts */
        @import url("https://fonts.googleapis.com/css?family=Inter:400,600,700");
        @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

        /* -------------------------------------------------- */
        /* global */
        /* -------------------------------------------------- */

        /* all elements */
        * {
            /* force sans-serif font unless specified otherwise */
            font-family: "Inter", "Helvetica", sans-serif;

            /* prevent text inflation on some mobile browsers */
            -webkit-text-size-adjust: none !important;
            -moz-text-size-adjust: none !important;
            -o-text-size-adjust: none !important;
            text-size-adjust: none !important;
        }

        @media only screen {

            /* "page" element */
            body {
                position: relative;
                box-sizing: border-box;
                font-size: 1.15rem;
                line-height: 1.75rem;
                max-width: 8.5in;
                margin: 20px auto;
                padding: 40px;
                border-radius: 5px;
                border: solid 1px #bdbdbd;
                box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
                background: #ffffff;
            }
        }

        /* when on screen < 8.5in wide */
        @media only screen and (max-width: 8.5in) {

            /* "page" element */
            body {
                padding: 20px;
                margin: 0;
                border-radius: 0;
                border: none;
                box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
                background: none;
            }
        }

        /* -------------------------------------------------- */
        /* headings */
        /* -------------------------------------------------- */

        /* all headings */
        h1,
        h2,
        h3,
        h4,
        h5,
        h6 {
            margin: 20px 0;
            padding: 0;
            font-weight: bold;
        }

        /* biggest heading */
        h1 {
            margin: 40px 0;
            text-align: center;
        }

        /* second biggest heading */
        h2 {
            margin-top: 30px;
            padding-bottom: 5px;
            border-bottom: solid 1px #bdbdbd;
        }

        /* heading font sizes */
        h1 {
            font-size: 2rem;
            line-height: 2.6rem;
        }

        h2 {
            font-size: 1.5em;
        }

        h3 {
            font-size: 1.35em;
        }

        h4 {
            font-size: 1.25em;
        }

        h5 {
            font-size: 1.15em;
        }

        h6 {
            font-size: 1em;
        }

        /* -------------------------------------------------- */
        /* manuscript header */
        /* -------------------------------------------------- */

        /* manuscript title */
        header>h1 {
            margin: 0;
        }

        /* manuscript title caption text (ie "automatically generated on") */
        header+p {
            text-align: center;
            margin-top: 10px;
        }

        /* -------------------------------------------------- */
        /* text elements */
        /* -------------------------------------------------- */

        /* links */
        a {
            color: #2196f3;
            overflow-wrap: break-word;
        }

        /* normal links (not empty, not button link, not syntax highlighting link) */
        a:not(:empty):not(.button):not(.sourceLine) {
            padding-left: 1px;
            padding-right: 1px;
        }

        /* superscripts and subscripts */
        sub,
        sup {
            /* prevent from affecting line height */
            line-height: 0;
        }

        /* unordered and ordered lists*/
        ul,
        ol {
            padding-left: 20px;
        }

        /* class for styling text semibold */
        .semibold {
            font-weight: 600;
        }

        /* class for styling elements horizontally left aligned */
        .left {
            display: block;
            text-align: left;
            margin-left: auto;
            margin-right: 0;
            justify-content: left;
        }

        /* class for styling elements horizontally centered */
        .center {
            display: block;
            text-align: center;
            margin-left: auto;
            margin-right: auto;
            justify-content: center;
        }

        /* class for styling elements horizontally right aligned */
        .right {
            display: block;
            text-align: right;
            margin-left: 0;
            margin-right: auto;
            justify-content: right;
        }

        /* -------------------------------------------------- */
        /* section elements */
        /* -------------------------------------------------- */

        /* horizontal divider line */
        hr {
            border: none;
            height: 1px;
            background: #bdbdbd;
        }

        /* paragraphs, horizontal dividers, figures, tables, code */
        p,
        hr,
        figure,
        table,
        pre {
            /* treat all as "paragraphs", with consistent vertical margins */
            margin-top: 20px;
            margin-bottom: 20px;
        }

        /* -------------------------------------------------- */
        /* figures */
        /* -------------------------------------------------- */

        /* figure */
        figure {
            max-width: 100%;
            margin-left: auto;
            margin-right: auto;
            border: 1px solid #c0c0c0;
            padding: 10px;
            border-radius: 5px;
        }

        /* figure caption */
        figcaption {
            padding: 0;
            padding-top: 10px;
            line-height: 1.5rem;
            font-size: 0.8rem;
        }

        /* figure image element */
        figure img {
            max-width: 100%;
            display: block;
            margin-left: auto;
            margin-right: auto;
        }

        /* figure auto-number */
        img+figcaption>span:first-of-type {
            font-weight: bold;
            margin-right: 5px;
        }

        /* -------------------------------------------------- */
        /* tables */
        /* -------------------------------------------------- */

        /* table */
        table {
            border-collapse: collapse;
            border-spacing: 0;
            width: 100%;
            margin-left: auto;
            margin-right: auto;
            border-bottom: 1px solid rgba(0, 0, 0, 0.75);
            font-size: 1rem;
        }

        /* table cells */
        th,
        td {
            /* border: solid 1px #bdbdbd; */
            padding: 10px;
            /* squash table if too wide for page by forcing line breaks */
            overflow-wrap: break-word;
            word-break: break-word;
        }

        /* header row and even rows */
        /* th,
    tr:nth-child(2n) {
        background-color: #fafafa;
    }

    /* odd rows
    tr:nth-child(2n + 1) {
        background-color: #ffffff;
    } */

        /* table caption */
        caption {
            text-align: left;
            padding: 0;
            padding-bottom: 10px;
            line-height: 1.5rem;
            font-size: 0.8rem;
        }

        /* table auto-number */
        table>caption>span:first-of-type,
        div.table_wrapper>table>caption>span:first-of-type {
            font-weight: bold;
            margin-right: 5px;
        }

        /* -------------------------------------------------- */
        /* code */
        /* -------------------------------------------------- */

        /* multi-line code block */
        pre {
            padding: 10px;
            background-color: #eeeeee;
            color: #000000;
            border-radius: 5px;
            break-inside: avoid;
            text-align: left;
        }

        /* inline code, ie code within normal text */
        :not(pre)>code {
            padding: 0 4px;
            background-color: #eeeeee;
            color: #000000;
            border-radius: 5px;
        }

        /* code text */
        /* apply all children, to reach syntax highlighting sub-elements */
        code,
        code * {
            /* force monospace font */
            font-family: "Source Code Pro", "Courier New", monospace;
        }

        /* -------------------------------------------------- */
        /* quotes */
        /* -------------------------------------------------- */

        /* quoted text */
        blockquote {
            margin: 0;
            padding: 0;
            border-left: 4px solid #bdbdbd;
            padding-left: 16px;
            break-inside: avoid;
        }

        /* -------------------------------------------------- */
        /* banners */
        /* -------------------------------------------------- */

        /* info banners */
        .banner {
            box-sizing: border-box;
            display: block;
            position: relative;
            width: 100%;
            margin-top: 20px;
            margin-bottom: 20px;
            padding: 20px;
            text-align: center;
        }

        /* paragraph in banner */
        .banner>p {
            margin: 0;
        }

        /* -------------------------------------------------- */
        /* highlight colors */
        /* -------------------------------------------------- */

        .white {
            background: #ffffff;
        }

        .lightgrey {
            background: #eeeeee;
        }

        .grey {
            background: #757575;
        }

        .darkgrey {
            background: #424242;
        }

        .black {
            background: #000000;
        }

        .lightred {
            background: #ffcdd2;
        }

        .lightyellow {
            background: #ffecb3;
        }

        .lightgreen {
            background: #dcedc8;
        }

        .lightblue {
            background: #e3f2fd;
        }

        .lightpurple {
            background: #f3e5f5;
        }

        .red {
            background: #f44336;
        }

        .orange {
            background: #ff9800;
        }

        .yellow {
            background: #ffeb3b;
        }

        .green {
            background: #4caf50;
        }

        .blue {
            background: #2196f3;
        }

        .purple {
            background: #9c27b0;
        }

        .white,
        .lightgrey,
        .lightred,
        .lightyellow,
        .lightgreen,
        .lightblue,
        .lightpurple,
        .orange,
        .yellow,
        .white a,
        .lightgrey a,
        .lightred a,
        .lightyellow a,
        .lightgreen a,
        .lightblue a,
        .lightpurple a,
        .orange a,
        .yellow a {
            color: #000000;
        }

        .grey,
        .darkgrey,
        .black,
        .red,
        .green,
        .blue,
        .purple,
        .grey a,
        .darkgrey a,
        .black a,
        .red a,
        .green a,
        .blue a,
        .purple a {
            color: #ffffff;
        }

        /* -------------------------------------------------- */
        /* buttons */
        /* -------------------------------------------------- */

        /* class for styling links like buttons */
        .button {
            display: inline-flex;
            justify-content: center;
            align-items: center;
            margin: 5px;
            padding: 10px 20px;
            font-size: 0.75em;
            font-weight: 600;
            text-transform: uppercase;
            text-decoration: none;
            letter-spacing: 1px;
            background: none;
            color: #2196f3;
            border: solid 1px #bdbdbd;
            border-radius: 5px;
        }

        /* buttons when hovered */
        .button:hover:not([disabled]),
        .icon_button:hover:not([disabled]) {
            cursor: pointer;
            background: #f5f5f5;
        }

        /* buttons when disabled */
        .button[disabled],
        .icon_button[disabled] {
            opacity: 0.35;
            pointer-events: none;
        }

        /* class for styling buttons containg only single icon */
        .icon_button {
            display: inline-flex;
            justify-content: center;
            align-items: center;
            text-decoration: none;
            margin: 0;
            padding: 0;
            background: none;
            border-radius: 5px;
            border: none;
            width: 20px;
            height: 20px;
            min-width: 20px;
            min-height: 20px;
        }

        /* icon button inner svg image */
        .icon_button>svg {
            height: 16px;
        }

        /* -------------------------------------------------- */
        /* icons */
        /* -------------------------------------------------- */

        /* class for styling icons inline with text */
        .inline_icon {
            height: 1em;
            position: relative;
            top: 0.125em;
        }

        /* -------------------------------------------------- */
        /* print control */
        /* -------------------------------------------------- */

        @media print {
            @page {
                /* suggested printing margin */
                margin: 0.5in;
            }

            /* document and "page" elements */
            html,
            body {
                margin: 0;
                padding: 0;
                width: 100%;
                height: 100%;
            }

            /* "page" element */
            body {
                font-size: 11pt !important;
                line-height: 2;
            }

            /* all headings */
            h1,
            h2,
            h3,
            h4,
            h5,
            h6 {
                margin: 15px 0;
            }

            /* figures and tables */
            figure,
            table {
                font-size: 0.85em;
            }

            /* table cells */
            th,
            td {
                padding: 5px;
            }

            /* shrink font awesome icons */
            i.fas,
            i.fab,
            i.far,
            i.fal {
                transform: scale(0.85);
            }

            /* decrease banner margins */
            .banner {
                margin-top: 15px;
                margin-bottom: 15px;
                padding: 15px;
            }

            /* class for centering an element vertically on its own page */
            .page_center {
                margin: auto;
                width: 100%;
                height: 100%;
                display: flex;
                align-items: center;
                vertical-align: middle;
                break-before: page;
                break-after: page;
            }

            /* always insert a page break before the element */
            .page_break_before {
                break-before: page;
            }

            /* always insert a page break after the element */
            .page_break_after {
                break-after: page;
            }

            /* avoid page break before the element */
            .page_break_before_avoid {
                break-before: avoid;
            }

            /* avoid page break after the element */
            .page_break_after_avoid {
                break-after: avoid;
            }

            /* avoid page break inside the element */
            .page_break_inside_avoid {
                break-inside: avoid;
            }
        }

        /* -------------------------------------------------- */
        /* override pandoc css quirks */
        /* -------------------------------------------------- */

        .sourceCode {
            /* prevent unsightly overflow in wide code blocks */
            overflow: auto !important;
        }

        div.sourceCode {
            /* prevent background fill on top-most code block  container */
            background: none !important;
        }

        .sourceCode * {
            /* force consistent line spacing */
            line-height: 1.5 !important;
        }

        div.sourceCode {
            /* style code block margins same as <pre> element */
            margin-top: 20px;
            margin-bottom: 20px;
        }

        /* -------------------------------------------------- */
        /* tablenos */
        /* -------------------------------------------------- */

        /* tablenos wrapper */
        .tablenos {
            /* show scrollbar on tables if necessary to prevent overflow */
            width: 100%;
            margin: 20px 0;
        }

        .tablenos>table {
            /* move margins from table to table_wrapper to allow margin collapsing */
            margin: 0;
        }

        @media only screen {

            /* tablenos wrapper */
            .tablenos {
                /* show scrollbar on tables if necessary to prevent overflow */
                overflow-x: auto !important;
            }

            .tablenos th,
            .tablenos td {
                overflow-wrap: unset !important;
                word-break: unset !important;
            }

            /* table in wrapper */
            .tablenos table,
            .tablenos table * {
                /* don't break table words */
                overflow-wrap: normal !important;
            }
        }

        /* -------------------------------------------------- */
        /* mathjax */
        /* -------------------------------------------------- */

        /* mathjax containers */
        .math.display>span:not(.MathJax_Preview) {
            /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
            display: flex !important;
            overflow-x: auto !important;
            overflow-y: hidden !important;
            justify-content: center;
            align-items: center;
            margin: 0 !important;
        }

        /* right click menu */
        .MathJax_Menu {
            border-radius: 5px !important;
            border: solid 1px #bdbdbd !important;
            box-shadow: none !important;
        }

        /* equation auto-number */
        span[id^="eq:"]>span.math.display+span {
            font-weight: 600;
        }

        /* equation */
        span[id^="eq:"]>span.math.display>span {
            /* nudge to make room for equation auto-number and anchor */
            margin-right: 60px !important;
        }

        /* -------------------------------------------------- */
        /* anchors plugin */
        /* -------------------------------------------------- */

        @media only screen {

            /* anchor button */
            .anchor {
                opacity: 0;
                margin-left: 5px;
            }

            /* anchor buttons within <h2>'s */
            h2 .anchor {
                margin-left: 10px;
            }

            /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
            *:hover>.anchor,
            .anchor:hover,
            .anchor:focus {
                opacity: 1;
            }

            /* anchor button when hovered */
            .anchor:hover {
                cursor: pointer;
            }
        }

        /* always show anchor button on devices with no mouse/hover ability */
        @media (hover: none) {
            .anchor {
                opacity: 1;
            }
        }

        /* always hide anchor button on print */
        @media only print {
            .anchor {
                display: none;
            }
        }

        /* -------------------------------------------------- */
        /* accordion plugin */
        /* -------------------------------------------------- */

        @media only screen {

            /* accordion arrow button */
            .accordion_arrow {
                margin-right: 10px;
            }

            /* arrow icon when <h2> data-collapsed attribute true */
            h2[data-collapsed="true"]>.accordion_arrow>svg {
                transform: rotate(-90deg);
            }

            /* all elements (except <h2>'s) when data-collapsed attribute true */
            *:not(h2)[data-collapsed="true"] {
                display: none;
            }

            /* accordion arrow button when hovered and <h2>'s when hovered */
            .accordion_arrow:hover,
            h2[data-collapsed="true"]:hover,
            h2[data-collapsed="false"]:hover {
                cursor: pointer;
            }
        }

        /* always hide accordion arrow button on print */
        @media only print {
            .accordion_arrow {
                display: none;
            }
        }

        /* -------------------------------------------------- */
        /* tooltips plugin */
        /* -------------------------------------------------- */

        @media only screen {

            /* tooltip container */
            #tooltip {
                position: absolute;
                width: 50%;
                min-width: 240px;
                max-width: 75%;
                z-index: 1;
            }

            /* tooltip content */
            #tooltip_content {
                margin-bottom: 5px;
                padding: 20px;
                border-radius: 5px;
                border: solid 1px #bdbdbd;
                box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
                background: #ffffff;
                overflow-wrap: break-word;
            }

            /* tooltip copy of paragraphs and figures */
            #tooltip_content>p,
            #tooltip_content>figure {
                margin: 0;
                max-height: 320px;
                overflow-y: auto;
            }

            /* tooltip copy of <img> */
            #tooltip_content>figure>img {
                max-height: 260px;
            }

            /* navigation bar */
            #tooltip_nav_bar {
                margin-top: 10px;
                text-align: center;
            }

            /* navigation bar previous/next buton */
            #tooltip_nav_bar>.icon_button {
                position: relative;
                top: 3px;
            }

            /* navigation bar previous button */
            #tooltip_nav_bar>.icon_button:first-of-type {
                margin-right: 5px;
            }

            /* navigation bar next button */
            #tooltip_nav_bar>.icon_button:last-of-type {
                margin-left: 5px;
            }
        }

        /* always hide tooltip on print */
        @media only print {
            #tooltip {
                display: none;
            }
        }

        /* -------------------------------------------------- */
        /* jump to first plugin */
        /* -------------------------------------------------- */

        @media only screen {

            /* jump button */
            .jump_arrow {
                position: relative;
                top: 0.125em;
                margin-right: 5px;
            }
        }

        /* always hide jump button on print */
        @media only print {
            .jump_arrow {
                display: none;
            }
        }

        /* -------------------------------------------------- */
        /* link highlight plugin */
        /* -------------------------------------------------- */

        @media only screen {

            /* anything with data-highlighted attribute true */
            [data-highlighted="true"] {
                background: #ffeb3b;
            }

            /* anything with data-selected attribute true */
            [data-selected="true"] {
                background: #ff8a65 !important;
            }

            /* animation definition for glow */
            @keyframes highlight_glow {
                0% {
                    background: none;
                }

                10% {
                    background: #bbdefb;
                }

                100% {
                    background: none;
                }
            }

            /* anything with data-glow attribute true */
            [data-glow="true"] {
                animation: highlight_glow 2s;
            }
        }

        /* -------------------------------------------------- */
        /* table of contents plugin */
        /* -------------------------------------------------- */

        @media only screen {

            /* toc panel */
            #toc_panel {
                box-sizing: border-box;
                position: fixed;
                top: 0;
                left: 0;
                background: #ffffff;
                box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
                z-index: 2;
            }

            /* toc panel when closed */
            #toc_panel[data-open="false"] {
                min-width: 60px;
                width: 60px;
                height: 60px;
                border-right: solid 1px #bdbdbd;
                border-bottom: solid 1px #bdbdbd;
            }

            /* toc panel when open */
            #toc_panel[data-open="true"] {
                min-width: 260px;
                max-width: 480px;
                /* keep panel edge consistent distance away from "page" edge */
                width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
                bottom: 0;
                border-right: solid 1px #bdbdbd;
            }

            /* toc panel header */
            #toc_header {
                box-sizing: border-box;
                display: flex;
                flex-direction: row;
                align-items: center;
                height: 60px;
                margin: 0;
                padding: 20px;
            }

            /* toc panel header when hovered */
            #toc_header:hover {
                cursor: pointer;
            }

            /* toc panel header when panel open */
            #toc_panel[data-open="true"]>#toc_header {
                border-bottom: solid 1px #bdbdbd;
            }

            /* toc open/close header button */
            #toc_button {
                margin-right: 20px;
            }

            /* hide toc list and header text when closed */
            #toc_panel[data-open="false"]>#toc_header>*:not(#toc_button),
            #toc_panel[data-open="false"]>#toc_list {
                display: none;
            }

            /* toc list of entries */
            #toc_list {
                box-sizing: border-box;
                width: 100%;
                padding: 20px;
                position: absolute;
                top: calc(60px + 1px);
                bottom: 0;
                overflow: auto;
            }

            /* toc entry, link to section in document */
            .toc_link {
                display: block;
                padding: 5px;
                position: relative;
                font-weight: 600;
                text-decoration: none;
            }

            /* toc entry when hovered or when "viewed" */
            .toc_link:hover,
            .toc_link[data-viewing="true"] {
                background: #f5f5f5;
            }

            /* toc entry, level 1 indentation */
            .toc_link[data-level="1"] {
                margin-left: 0;
            }

            /* toc entry, level 2 indentation */
            .toc_link[data-level="2"] {
                margin-left: 20px;
            }

            /* toc entry, level 3 indentation */
            .toc_link[data-level="3"] {
                margin-left: 40px;
            }

            /* toc entry, level 4 indentation */
            .toc_link[data-level="4"] {
                margin-left: 60px;
            }

            /* toc entry bullets */
            #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
                position: absolute;
                left: -15px;
                top: -1px;
                font-size: 1.5em;
            }

            /* toc entry, level 2 bullet */
            #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
                content: "\2022";
            }

            /* toc entry, level 3 bullet */
            #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
                content: "\25AB";
            }

            /* toc entry, level 4 bullet */
            #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
                content: "-";
            }
        }

        /* when on screen < 8.5in wide */
        @media only screen and (max-width: 8.5in) {

            /* push <body> ("page") element down to make room for toc icon */
            .toc_body_nudge {
                padding-top: 60px;
            }

            /* toc icon when panel closed and not hovered */
            #toc_panel[data-open="false"]:not(:hover) {
                background: rgba(255, 255, 255, 0.75);
            }
        }

        /* always hide toc panel on print */
        @media only print {
            #toc_panel {
                display: none;
            }
        }

        /* -------------------------------------------------- */
        /* lightbox plugin */
        /* -------------------------------------------------- */

        @media only screen {

            /* regular <img> in document when hovered */
            .lightbox_document_img:hover {
                cursor: pointer;
            }

            .body_no_scroll {
                overflow: hidden !important;
            }

            /* screen overlay */
            #lightbox_overlay {
                display: flex;
                flex-direction: column;
                position: fixed;
                left: 0;
                top: 0;
                right: 0;
                bottom: 0;
                background: rgba(0, 0, 0, 0.75);
                z-index: 3;
            }

            /* middle area containing lightbox image */
            #lightbox_image_container {
                flex-grow: 1;
                display: flex;
                justify-content: center;
                align-items: center;
                overflow: hidden;
                position: relative;
                padding: 20px;
            }

            /* bottom area containing caption */
            #lightbox_bottom_container {
                display: flex;
                justify-content: center;
                align-items: center;
                height: 100px;
                min-height: 100px;
                max-height: 100px;
                background: rgba(0, 0, 0, 0.5);
            }

            /* image number info text box */
            #lightbox_number_info {
                position: absolute;
                color: #ffffff;
                font-weight: 600;
                left: 2px;
                top: 0;
                z-index: 4;
            }

            /* zoom info text box */
            #lightbox_zoom_info {
                position: absolute;
                color: #ffffff;
                font-weight: 600;
                right: 2px;
                top: 0;
                z-index: 4;
            }

            /* copy of image caption */
            #lightbox_caption {
                box-sizing: border-box;
                display: inline-block;
                width: 100%;
                max-height: 100%;
                padding: 10px 0;
                text-align: center;
                overflow-y: auto;
                color: #ffffff;
            }

            /* navigation previous/next button */
            .lightbox_button {
                width: 100px;
                height: 100%;
                min-width: 100px;
                min-height: 100%;
                color: #ffffff;
            }

            /* navigation previous/next button when hovered */
            .lightbox_button:hover {
                background: none !important;
            }

            /* navigation button icon */
            .lightbox_button>svg {
                height: 25px;
            }

            /* figure auto-number */
            #lightbox_caption>span:first-of-type {
                font-weight: bold;
                margin-right: 5px;
            }

            /* lightbox image when hovered */
            #lightbox_img:hover {
                cursor: grab;
            }

            /* lightbox image when grabbed */
            #lightbox_img:active {
                cursor: grabbing;
            }
        }

        /* when on screen < 480px wide */
        @media only screen and (max-width: 480px) {

            /* make navigation buttons skinnier on small screens to make more room for caption text */
            .lightbox_button {
                width: 50px;
                min-width: 50px;
            }
        }

        /* always hide lightbox on print */
        @media only print {
            #lightbox_overlay {
                display: none;
            }
        }

        /* -------------------------------------------------- */
        /* hypothesis (annotations) plugin */
        /* -------------------------------------------------- */

        /* hypothesis activation button */
        #hypothesis_button {
            box-sizing: border-box;
            position: fixed;
            top: 0;
            right: 0;
            width: 60px;
            height: 60px;
            background: #ffffff;
            border-radius: 0;
            border-left: solid 1px #bdbdbd;
            border-bottom: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            z-index: 2;
        }

        /* hypothesis button svg */
        #hypothesis_button>svg {
            position: relative;
            top: -4px;
        }

        /* hypothesis annotation count */
        #hypothesis_count {
            position: absolute;
            left: 0;
            right: 0;
            bottom: 5px;
        }

        /* side panel */
        .annotator-frame {
            width: 280px !important;
        }

        /* match highlight color to rest of theme */
        .annotator-highlights-always-on .annotator-hl {
            background-color: #ffeb3b !important;
        }

        /* match focused color to rest of theme */
        .annotator-hl.annotator-hl-focused {
            background-color: #ff8a65 !important;
        }

        /* match bucket bar color to rest of theme */
        .annotator-bucket-bar {
            background: #f5f5f5 !important;
        }

        /* always hide button, toolbar, and tooltip on print */
        @media only print {
            #hypothesis_button {
                display: none;
            }

            .annotator-frame {
                display: none !important;
            }

            hypothesis-adder {
                display: none !important;
            }
        }


        thead {
            border-top: 1px solid rgba(0, 0, 0, 0.75);
            border-left: 0px;
            border-right: 0px;
            border-bottom: 2px solid rgba(0, 0, 0, 0.75);
        }


        .download {
            background-color: #e7e7e7;
            /* Green */
            border: none;
            color: black;
            padding: 1em 1.5em;
            text-align: center;
            text-decoration: none;
            display: inline-block;
            font-size: 0.75em;
            text-align: right;
            margin-top: 10px;
            cursor: pointer;
        }

        .download:hover {
            background-color: black;
            color: #e7e7e7;
        }

        .tabulator-header {
            border-top: 1px solid rgba(0, 0, 0, 0.75);
            border-bottom: 2px solid rgba(0, 0, 0, 0.75) !important;
        }

        .tabulator-row .tabulator-cell {
            border: 0px;
        }

        .tabulator-row {
            border: 0px;
            padding: 5px 0;
        }

        .tabulator .tabulator-header .tabulator-col {
            border: 0px;
            padding: 5px 0;
        }
    </style>
    <!-- pandoc-eqnos: equation style -->
    <style>
        .eqnos {
            display: inline-block;
            position: relative;
            width: 100%;
        }

        .eqnos br {
            display: none;
        }

        .eqnos-number {
            position: absolute;
            right: 0em;
            top: 50%;
            line-height: 0;
        }
    </style>
    <!-- anchors plugin -->

    <script>
        (function () {
            // /////////////////////////
            // DESCRIPTION
            // /////////////////////////

            // This Manubot plugin adds an anchor next to each of a certain type
            // of element that provides a human-readable url to that specific
            // item/position in the document (eg "manuscript.html#abstract"). It
            // also makes it such that scrolling out of view of a target removes
            // its identifier from the url.

            // /////////////////////////
            // OPTIONS
            // /////////////////////////

            // plugin name prefix for url parameters
            const pluginName = 'anchors';

            // default plugin options
            const options = {
                // which types of elements to add anchors next to, in
                // "document.querySelector" format
                typesQuery: 'h1, h2, h3, [id^="fig:"], [id^="tbl:"], [id^="eq:"]',
                // whether plugin is on or not
                enabled: 'true'
            };

            // change options above, or override with url parameter, eg:
            // 'manuscript.html?pluginName-enabled=false'

            // /////////////////////////
            // SCRIPT
            // /////////////////////////

            // start script
            function start() {
                // add anchor to each element of specified types
                const elements = document.querySelectorAll(options.typesQuery);
                for (const element of elements)
                    addAnchor(element);

                // attach scroll listener to window
                window.addEventListener('scroll', onScroll);
            }

            // when window is scrolled
            function onScroll() {
                // if url has hash and user has scrolled out of view of hash
                // target, remove hash from url
                const tolerance = 100;
                const target = getHashTarget();
                if (target) {
                    if (
                        target.getBoundingClientRect().top >
                        window.innerHeight + tolerance ||
                        target.getBoundingClientRect().bottom < 0 - tolerance
                    )
                        history.pushState(null, null, ' ');
                }
            }

            // add anchor to element
            function addAnchor(element) {
                let addTo; // element to add anchor button to

                // if figure or table, modify withId and addTo to get expected
                // elements
                if (element.id.indexOf('fig:') === 0) {
                    addTo = element.querySelector('figcaption');
                } else if (element.id.indexOf('tbl:') === 0) {
                    addTo = element.querySelector('caption');
                } else if (element.id.indexOf('eq:') === 0) {
                    addTo = element.querySelector('.eqnos-number');
                }

                addTo = addTo || element;
                const id = element.id || null;

                // do not add anchor if element doesn't have assigned id.
                // id is generated by pandoc and is assumed to be unique and
                // human-readable
                if (!id)
                    return;

                // create anchor button
                const anchor = document.createElement('a');
                anchor.innerHTML = document.querySelector('.icon_link').innerHTML;
                anchor.title = 'Link to this part of the document';
                anchor.classList.add('icon_button', 'anchor');
                anchor.dataset.ignore = 'true';
                anchor.href = '#' + id;
                addTo.appendChild(anchor);
            }

            // get element that is target of link or url hash
            function getHashTarget() {
                const hash = window.location.hash;
                const id = hash.slice(1);
                let target = document.querySelector('[id="' + id + '"]');
                if (!target)
                    return;

                // if figure or table, modify target to get expected element
                if (id.indexOf('fig:') === 0)
                    target = target.querySelector('figure');
                if (id.indexOf('tbl:') === 0)
                    target = target.querySelector('table');

                return target;
            }

            // load options from url parameters
            function loadOptions() {
                const url = window.location.search;
                const params = new URLSearchParams(url);
                for (const optionName of Object.keys(options)) {
                    const paramName = pluginName + '-' + optionName;
                    const param = params.get(paramName);
                    if (param !== '' && param !== null)
                        options[optionName] = param;
                }
            }
            loadOptions();

            // start script when document is finished loading
            if (options.enabled === 'true')
                window.addEventListener('load', start);
        })();
    </script>

    <!-- link icon -->

    <template class="icon_link">
        <!-- modified from: https://fontawesome.com/icons/link -->
        <svg width="16" height="16" viewBox="0 0 512 512">
            <path fill="currentColor"
                d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z">
            </path>
        </svg>
    </template>
    <!-- accordion plugin -->

    <script>
        (function () {
            // /////////////////////////
            // DESCRIPTION
            // /////////////////////////

            // This Manubot plugin allows sections of content under <h2> headings
            // to be collapsible.

            // /////////////////////////
            // OPTIONS
            // /////////////////////////

            // plugin name prefix for url parameters
            const pluginName = 'accordion';

            // default plugin options
            const options = {
                // whether to always start expanded ('false'), always start
                // collapsed ('true'), or start collapsed when screen small ('auto')
                startCollapsed: 'auto',
                // whether plugin is on or not
                enabled: 'true'
            };

            // change options above, or override with url parameter, eg:
            // 'manuscript.html?pluginName-enabled=false'

            // /////////////////////////
            // SCRIPT
            // /////////////////////////

            // start script
            function start() {
                // run through each <h2> heading
                const headings = document.querySelectorAll('h2');
                for (const heading of headings) {
                    addArrow(heading);

                    // start expanded/collapsed based on option
                    if (
                        options.startCollapsed === 'true' ||
                        (options.startCollapsed === 'auto' && isSmallScreen())
                    )
                        collapseHeading(heading);
                    else
                        expandHeading(heading);
                }

                // attach hash change listener to window
                window.addEventListener('hashchange', onHashChange);
            }

            // when hash (eg manuscript.html#introduction) changes
            function onHashChange() {
                const target = getHashTarget();
                if (target)
                    goToElement(target);
            }

            // add arrow to heading
            function addArrow(heading) {
                // add arrow button
                const arrow = document.createElement('button');
                arrow.innerHTML = document.querySelector(
                    '.icon_angle_down'
                ).innerHTML;
                arrow.classList.add('icon_button', 'accordion_arrow');
                heading.insertBefore(arrow, heading.firstChild);

                // attach click listener to heading and button
                heading.addEventListener('click', onHeadingClick);
                arrow.addEventListener('click', onArrowClick);
            }

            // determine if on mobile-like device with small screen
            function isSmallScreen() {
                return Math.min(window.innerWidth, window.innerHeight) < 480;
            }

            // scroll to and focus element
            function goToElement(element, offset) {
                // expand accordion section if collapsed
                expandElement(element);
                const y =
                    getRectInView(element).top -
                    getRectInView(document.documentElement).top -
                    (offset || 0);
                // trigger any function listening for "onscroll" event
                window.dispatchEvent(new Event('scroll'));
                window.scrollTo(0, y);
                document.activeElement.blur();
                element.focus();
            }

            // get element that is target of hash
            function getHashTarget(link) {
                const hash = link ? link.hash : window.location.hash;
                const id = hash.slice(1);
                let target = document.querySelector('[id="' + id + '"]');
                if (!target)
                    return;

                // if figure or table, modify target to get expected element
                if (id.indexOf('fig:') === 0)
                    target = target.querySelector('figure');
                if (id.indexOf('tbl:') === 0)
                    target = target.querySelector('table');

                return target;
            }

            // when <h2> heading is clicked
            function onHeadingClick(event) {
                // only collapse if <h2> itself is target of click (eg, user did
                // not click on anchor within <h2>)
                if (event.target === this)
                    toggleCollapse(this);
            }

            // when arrow button is clicked
            function onArrowClick() {
                toggleCollapse(this.parentNode);
            }

            // collapse section if expanded, expand if collapsed
            function toggleCollapse(heading) {
                if (heading.dataset.collapsed === 'false')
                    collapseHeading(heading);
                else
                    expandHeading(heading);
            }

            // elements to exclude from collapse, such as table of contents panel,
            // hypothesis panel, etc
            const exclude = '#toc_panel, div.annotator-frame, #lightbox_overlay';

            // collapse section
            function collapseHeading(heading) {
                heading.setAttribute('data-collapsed', 'true');
                const children = getChildren(heading);
                for (const child of children)
                    child.setAttribute('data-collapsed', 'true');
            }

            // expand section
            function expandHeading(heading) {
                heading.setAttribute('data-collapsed', 'false');
                const children = getChildren(heading);
                for (const child of children)
                    child.setAttribute('data-collapsed', 'false');
            }

            // get list of elements between this <h2> and next <h2> or <h1>
            // ("children" of the <h2> section)
            function getChildren(heading) {
                return nextUntil(heading, 'h2, h1', exclude);
            }

            // get position/dimensions of element or viewport
            function getRectInView(element) {
                let rect = {};
                rect.left = 0;
                rect.top = 0;
                rect.right = document.documentElement.clientWidth;
                rect.bottom = document.documentElement.clientHeight;
                let style = {};

                if (element instanceof HTMLElement) {
                    rect = element.getBoundingClientRect();
                    style = window.getComputedStyle(element);
                }

                const margin = {};
                margin.left = parseFloat(style.marginLeftWidth) || 0;
                margin.top = parseFloat(style.marginTopWidth) || 0;
                margin.right = parseFloat(style.marginRightWidth) || 0;
                margin.bottom = parseFloat(style.marginBottomWidth) || 0;

                const border = {};
                border.left = parseFloat(style.borderLeftWidth) || 0;
                border.top = parseFloat(style.borderTopWidth) || 0;
                border.right = parseFloat(style.borderRightWidth) || 0;
                border.bottom = parseFloat(style.borderBottomWidth) || 0;

                const newRect = {};
                newRect.left = rect.left + margin.left + border.left;
                newRect.top = rect.top + margin.top + border.top;
                newRect.right = rect.right + margin.right + border.right;
                newRect.bottom = rect.bottom + margin.bottom + border.bottom;
                newRect.width = newRect.right - newRect.left;
                newRect.height = newRect.bottom - newRect.top;

                return newRect;
            }

            // get list of elements after a start element up to element matching
            // query
            function nextUntil(element, query, exclude) {
                const elements = [];
                while (element = element.nextElementSibling, element) {
                    if (element.matches(query))
                        break;
                    if (!element.matches(exclude))
                        elements.push(element);
                }
                return elements;
            }

            // get closest element before specified element that matches query
            function firstBefore(element, query) {
                while (
                    element &&
                    element !== document.body &&
                    !element.matches(query)
                )
                    element = element.previousElementSibling || element.parentNode;

                return element;
            }

            // check if element is part of collapsed heading
            function isCollapsed(element) {
                while (element && element !== document.body) {
                    if (element.dataset.collapsed === 'true')
                        return true;
                    element = element.parentNode;
                }
                return false;
            }

            // expand heading containing element if necesary
            function expandElement(element) {
                if (isCollapsed(element)) {
                    const heading = firstBefore(element, 'h2');
                    if (heading)
                        heading.click();
                }
            }

            // load options from url parameters
            function loadOptions() {
                const url = window.location.search;
                const params = new URLSearchParams(url);
                for (const optionName of Object.keys(options)) {
                    const paramName = pluginName + '-' + optionName;
                    const param = params.get(paramName);
                    if (param !== '' && param !== null)
                        options[optionName] = param;
                }
            }
            loadOptions();

            // start script when document is finished loading
            if (options.enabled === 'true')
                window.addEventListener('load', start);
        })();
    </script>

    <!-- angle down icon -->

    <template class="icon_angle_down">
        <!-- modified from: https://fontawesome.com/icons/angle-down -->
        <svg width="16" height="16" viewBox="0 0 448 512">
            <path fill="currentColor"
                d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z">
            </path>
        </svg>
    </template>
    <!-- tooltips plugin -->

    <script>
        (function () {
            // /////////////////////////
            // DESCRIPTION
            // /////////////////////////

            // This Manubot plugin makes it such that when the user hovers or
            // focuses a link to a citation or figure, a tooltip appears with a
            // preview of the reference content, along with arrows to navigate
            // between instances of the same reference in the document.

            // /////////////////////////
            // OPTIONS
            // /////////////////////////

            // plugin name prefix for url parameters
            const pluginName = 'tooltips';

            // default plugin options
            const options = {
                // whether user must click off to close tooltip instead of just
                // un-hovering
                clickClose: 'false',
                // delay (in ms) between opening and closing tooltip
                delay: '100',
                // whether plugin is on or not
                enabled: 'true'
            };

            // change options above, or override with url parameter, eg:
            // 'manuscript.html?pluginName-enabled=false'

            // /////////////////////////
            // SCRIPT
            // /////////////////////////

            // start script
            function start() {
                const links = getLinks();
                for (const link of links) {
                    // attach hover and focus listeners to link
                    link.addEventListener('mouseover', onLinkHover);
                    link.addEventListener('mouseleave', onLinkUnhover);
                    link.addEventListener('focus', onLinkFocus);
                    link.addEventListener('touchend', onLinkTouch);
                }

                // attach mouse, key, and resize listeners to window
                window.addEventListener('mousedown', onClick);
                window.addEventListener('touchstart', onClick);
                window.addEventListener('keyup', onKeyUp);
                window.addEventListener('resize', onResize);
            }

            // when link is hovered
            function onLinkHover() {
                // function to open tooltip
                const delayOpenTooltip = function () {
                    openTooltip(this);
                }.bind(this);

                // run open function after delay
                this.openTooltipTimer = window.setTimeout(
                    delayOpenTooltip,
                    options.delay
                );
            }

            // when mouse leaves link
            function onLinkUnhover() {
                // cancel opening tooltip
                window.clearTimeout(this.openTooltipTimer);

                // don't close on unhover if option specifies
                if (options.clickClose === 'true')
                    return;

                // function to close tooltip
                const delayCloseTooltip = function () {
                    // if tooltip open and if mouse isn't over tooltip, close
                    const tooltip = document.getElementById('tooltip');
                    if (tooltip && !tooltip.matches(':hover'))
                        closeTooltip();
                };

                // run close function after delay
                this.closeTooltipTimer = window.setTimeout(
                    delayCloseTooltip,
                    options.delay
                );
            }

            // when link is focused (tabbed to)
            function onLinkFocus(event) {
                openTooltip(this);
            }

            // when link is touched on touch screen
            function onLinkTouch(event) {
                // attempt to force hover state on first tap always, and trigger
                // regular link click (and navigation) on second tap
                if (event.target === document.activeElement)
                    event.target.click();
                else {
                    document.activeElement.blur();
                    event.target.focus();
                }
                if (event.cancelable)
                    event.preventDefault();
                event.stopPropagation();
                return false;
            }

            // when mouse is clicked anywhere in window
            function onClick(event) {
                closeTooltip();
            }

            // when key pressed
            function onKeyUp(event) {
                if (!event || !event.key)
                    return;

                switch (event.key) {
                    // trigger click of prev button
                    case 'ArrowLeft':
                        const prevButton = document.getElementById(
                            'tooltip_prev_button'
                        );
                        if (prevButton)
                            prevButton.click();
                        break;
                    // trigger click of next button
                    case 'ArrowRight':
                        const nextButton = document.getElementById(
                            'tooltip_next_button'
                        );
                        if (nextButton)
                            nextButton.click();
                        break;
                    // close on esc
                    case 'Escape':
                        closeTooltip();
                        break;
                }
            }

            // when window is resized or zoomed
            function onResize() {
                closeTooltip();
            }

            // get all links of types we wish to handle
            function getLinks() {
                const queries = [];
                // exclude buttons, anchor links, toc links, etc
                const exclude =
                    ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
                queries.push('a[href^="#ref-"]' + exclude); // citation links
                queries.push('a[href^="#fig:"]' + exclude); // figure links
                const query = queries.join(', ');
                return document.querySelectorAll(query);
            }

            // get links with same target, get index of link in set, get total
            // same links
            function getSameLinks(link) {
                const sameLinks = [];
                const links = getLinks();
                for (const otherLink of links) {
                    if (
                        otherLink.getAttribute('href') === link.getAttribute('href')
                    )
                        sameLinks.push(otherLink);
                }

                return {
                    elements: sameLinks,
                    index: sameLinks.indexOf(link),
                    total: sameLinks.length
                };
            }

            // open tooltip
            function openTooltip(link) {
                // delete tooltip if it exists, start fresh
                closeTooltip();

                // make tooltip element
                const tooltip = makeTooltip(link);

                // if source couldn't be found and tooltip not made, exit
                if (!tooltip)
                    return;

                // make navbar elements
                const navBar = makeNavBar(link);
                if (navBar)
                    tooltip.firstElementChild.appendChild(navBar);

                // attach tooltip to page
                document.body.appendChild(tooltip);

                // position tooltip
                const position = function () {
                    positionTooltip(link);
                };
                position();

                // if tooltip contains images, position again after they've loaded
                const imgs = tooltip.querySelectorAll('img');
                for (const img of imgs)
                    img.addEventListener('load', position);
            }

            // close (delete) tooltip
            function closeTooltip() {
                const tooltip = document.getElementById('tooltip');
                if (tooltip)
                    tooltip.remove();
            }

            // make tooltip
            function makeTooltip(link) {
                // get target element that link points to
                const source = getSource(link);

                // if source can't be found, exit
                if (!source)
                    return;

                // create new tooltip
                const tooltip = document.createElement('div');
                tooltip.id = 'tooltip';
                const tooltipContent = document.createElement('div');
                tooltipContent.id = 'tooltip_content';
                tooltip.appendChild(tooltipContent);

                // make copy of source node and put in tooltip
                const sourceCopy = makeCopy(source);
                tooltipContent.appendChild(sourceCopy);

                // attach mouse event listeners
                tooltip.addEventListener('click', onTooltipClick);
                tooltip.addEventListener('mousedown', onTooltipClick);
                tooltip.addEventListener('touchstart', onTooltipClick);
                tooltip.addEventListener('mouseleave', onTooltipUnhover);

                // (for interaction with lightbox plugin)
                // transfer click on tooltip copied img to original img
                const sourceImg = source.querySelector('img');
                const sourceCopyImg = sourceCopy.querySelector('img');
                if (sourceImg && sourceCopyImg) {
                    const clickImg = function () {
                        sourceImg.click();
                        closeTooltip();
                    };
                    sourceCopyImg.addEventListener('click', clickImg);
                }

                return tooltip;
            }

            // make carbon copy of html dom element
            function makeCopy(source) {
                const sourceCopy = source.cloneNode(true);

                // delete elements marked with ignore (eg anchor and jump buttons)
                const deleteFromCopy = sourceCopy.querySelectorAll(
                    '[data-ignore="true"]'
                );
                for (const element of deleteFromCopy)
                    element.remove();

                // delete certain element attributes
                const attributes = [
                    'id',
                    'data-collapsed',
                    'data-selected',
                    'data-highlighted',
                    'data-glow'
                ];
                for (const attribute of attributes) {
                    sourceCopy.removeAttribute(attribute);
                    const elements = sourceCopy.querySelectorAll(
                        '[' + attribute + ']'
                    );
                    for (const element of elements)
                        element.removeAttribute(attribute);
                }

                return sourceCopy;
            }

            // when tooltip is clicked
            function onTooltipClick(event) {
                // when user clicks on tooltip, stop click from transferring
                // outside of tooltip (eg, click off to close tooltip, or eg click
                // off to unhighlight same refs)
                event.stopPropagation();
            }

            // when tooltip is unhovered
            function onTooltipUnhover(event) {
                if (options.clickClose === 'true')
                    return;

                // make sure new mouse/touch/focus no longer over tooltip or any
                // element within it
                const tooltip = document.getElementById('tooltip');
                if (!tooltip)
                    return;
                if (this.contains(event.relatedTarget))
                    return;

                closeTooltip();
            }

            // make nav bar to go betwen prev/next instances of same reference
            function makeNavBar(link) {
                // find other links to the same source
                const sameLinks = getSameLinks(link);

                // don't show nav bar when singular reference
                if (sameLinks.total <= 1)
                    return;

                // find prev/next links with same target
                const prevLink = getPrevLink(link, sameLinks);
                const nextLink = getNextLink(link, sameLinks);

                // create nav bar
                const navBar = document.createElement('div');
                navBar.id = 'tooltip_nav_bar';
                const text = sameLinks.index + 1 + ' of ' + sameLinks.total;

                // create nav bar prev/next buttons
                const prevButton = document.createElement('button');
                const nextButton = document.createElement('button');
                prevButton.id = 'tooltip_prev_button';
                nextButton.id = 'tooltip_next_button';
                prevButton.title =
                    'Jump to the previous occurence of this item in the document [←]';
                nextButton.title =
                    'Jump to the next occurence of this item in the document [→]';
                prevButton.classList.add('icon_button');
                nextButton.classList.add('icon_button');
                prevButton.innerHTML = document.querySelector(
                    '.icon_caret_left'
                ).innerHTML;
                nextButton.innerHTML = document.querySelector(
                    '.icon_caret_right'
                ).innerHTML;
                navBar.appendChild(prevButton);
                navBar.appendChild(document.createTextNode(text));
                navBar.appendChild(nextButton);

                // attach click listeners to buttons
                prevButton.addEventListener('click', function () {
                    onPrevNextClick(link, prevLink);
                });
                nextButton.addEventListener('click', function () {
                    onPrevNextClick(link, nextLink);
                });

                return navBar;
            }

            // get previous link with same target
            function getPrevLink(link, sameLinks) {
                if (!sameLinks)
                    sameLinks = getSameLinks(link);
                // wrap index to other side if < 1
                let index;
                if (sameLinks.index - 1 >= 0)
                    index = sameLinks.index - 1;
                else
                    index = sameLinks.total - 1;
                return sameLinks.elements[index];
            }

            // get next link with same target
            function getNextLink(link, sameLinks) {
                if (!sameLinks)
                    sameLinks = getSameLinks(link);
                // wrap index to other side if > total
                let index;
                if (sameLinks.index + 1 <= sameLinks.total - 1)
                    index = sameLinks.index + 1;
                else
                    index = 0;
                return sameLinks.elements[index];
            }

            // get element that is target of link or url hash
            function getSource(link) {
                const hash = link ? link.hash : window.location.hash;
                const id = hash.slice(1);
                let target = document.querySelector('[id="' + id + '"]');
                if (!target)
                    return;

                // if ref or figure, modify target to get expected element
                if (id.indexOf('ref-') === 0)
                    target = target.querySelector('p');
                else if (id.indexOf('fig:') === 0)
                    target = target.querySelector('figure');

                return target;
            }

            // when prev/next arrow button is clicked
            function onPrevNextClick(link, prevNextLink) {
                if (link && prevNextLink)
                    goToElement(prevNextLink, window.innerHeight * 0.5);
            }

            // scroll to and focus element
            function goToElement(element, offset) {
                // expand accordion section if collapsed
                expandElement(element);
                const y =
                    getRectInView(element).top -
                    getRectInView(document.documentElement).top -
                    (offset || 0);
                // trigger any function listening for "onscroll" event
                window.dispatchEvent(new Event('scroll'));
                window.scrollTo(0, y);
                document.activeElement.blur();
                element.focus();
            }

            // determine position to place tooltip based on link position in
            // viewport and tooltip size
            function positionTooltip(link, left, top) {
                const tooltipElement = document.getElementById('tooltip');
                if (!tooltipElement)
                    return;

                // get convenient vars for position/dimensions of
                // link/tooltip/page/view
                link = getRectInPage(link);
                const tooltip = getRectInPage(tooltipElement);
                const view = getRectInPage();

                // horizontal positioning
                if (left)
                    // use explicit value
                    left = left;
                else if (link.left + tooltip.width < view.right)
                    // fit tooltip to right of link
                    left = link.left;
                else if (link.right - tooltip.width > view.left)
                    // fit tooltip to left of link
                    left = link.right - tooltip.width;
                // center tooltip in view
                else
                    left = (view.right - view.left) / 2 - tooltip.width / 2;

                // vertical positioning
                if (top)
                    // use explicit value
                    top = top;
                else if (link.top - tooltip.height > view.top)
                    // fit tooltip above link
                    top = link.top - tooltip.height;
                else if (link.bottom + tooltip.height < view.bottom)
                    // fit tooltip below link
                    top = link.bottom;
                else {
                    // center tooltip in view
                    top = view.top + view.height / 2 - tooltip.height / 2;
                    // nudge off of link to left/right if possible
                    if (link.right + tooltip.width < view.right)
                        left = link.right;
                    else if (link.left - tooltip.width > view.left)
                        left = link.left - tooltip.width;
                }

                tooltipElement.style.left = left + 'px';
                tooltipElement.style.top = top + 'px';
            }

            // get position/dimensions of element or viewport
            function getRectInView(element) {
                let rect = {};
                rect.left = 0;
                rect.top = 0;
                rect.right = document.documentElement.clientWidth;
                rect.bottom = document.documentElement.clientHeight;
                let style = {};

                if (element instanceof HTMLElement) {
                    rect = element.getBoundingClientRect();
                    style = window.getComputedStyle(element);
                }

                const margin = {};
                margin.left = parseFloat(style.marginLeftWidth) || 0;
                margin.top = parseFloat(style.marginTopWidth) || 0;
                margin.right = parseFloat(style.marginRightWidth) || 0;
                margin.bottom = parseFloat(style.marginBottomWidth) || 0;

                const border = {};
                border.left = parseFloat(style.borderLeftWidth) || 0;
                border.top = parseFloat(style.borderTopWidth) || 0;
                border.right = parseFloat(style.borderRightWidth) || 0;
                border.bottom = parseFloat(style.borderBottomWidth) || 0;

                const newRect = {};
                newRect.left = rect.left + margin.left + border.left;
                newRect.top = rect.top + margin.top + border.top;
                newRect.right = rect.right + margin.right + border.right;
                newRect.bottom = rect.bottom + margin.bottom + border.bottom;
                newRect.width = newRect.right - newRect.left;
                newRect.height = newRect.bottom - newRect.top;

                return newRect;
            }

            // get position of element relative to page
            function getRectInPage(element) {
                const rect = getRectInView(element);
                const body = getRectInView(document.body);

                const newRect = {};
                newRect.left = rect.left - body.left;
                newRect.top = rect.top - body.top;
                newRect.right = rect.right - body.left;
                newRect.bottom = rect.bottom - body.top;
                newRect.width = rect.width;
                newRect.height = rect.height;

                return newRect;
            }

            // (for interaction with accordion plugin)
            // get closest element before specified element that matches query
            function firstBefore(element, query) {
                while (
                    element &&
                    element !== document.body &&
                    !element.matches(query)
                )
                    element = element.previousElementSibling || element.parentNode;

                return element;
            }

            // (for interaction with accordion plugin)
            // check if element is part of collapsed heading
            function isCollapsed(element) {
                while (element && element !== document.body) {
                    if (element.dataset.collapsed === 'true')
                        return true;
                    element = element.parentNode;
                }
                return false;
            }

            // (for interaction with accordion plugin)
            // expand heading containing element if necesary
            function expandElement(element) {
                if (isCollapsed(element)) {
                    const heading = firstBefore(element, 'h2');
                    if (heading)
                        heading.click();
                }
            }

            // load options from url parameters
            function loadOptions() {
                const url = window.location.search;
                const params = new URLSearchParams(url);
                for (const optionName of Object.keys(options)) {
                    const paramName = pluginName + '-' + optionName;
                    const param = params.get(paramName);
                    if (param !== '' && param !== null)
                        options[optionName] = param;
                }
            }
            loadOptions();

            // start script when document is finished loading
            if (options.enabled === 'true')
                window.addEventListener('load', start);
        })();
    </script>

    <!-- caret left icon -->

    <template class="icon_caret_left">
        <!-- modified from: https://fontawesome.com/icons/caret-left -->
        <svg width="16" height="16" viewBox="0 0 192 512">
            <path fill="currentColor"
                d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z">
            </path>
        </svg>
    </template>

    <!-- caret right icon -->

    <template class="icon_caret_right">
        <!-- modified from: https://fontawesome.com/icons/caret-right -->
        <svg width="16" height="16" viewBox="0 0 192 512">
            <path fill="currentColor"
                d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z">
            </path>
        </svg>
    </template>
    <!-- jump to first plugin -->

    <script>
        (function () {
            // /////////////////////////
            // DESCRIPTION
            // /////////////////////////

            // This Manubot plugin adds a button next to each reference entry,
            // figure, and table that jumps the page to the first occurrence of a
            // link to that item in the manuscript.

            // /////////////////////////
            // OPTIONS
            // /////////////////////////

            // plugin name prefix for url parameters
            const pluginName = 'jumpToFirst';

            // default plugin options
            const options = {
                // whether to add buttons next to reference entries
                references: 'true',
                // whether to add buttons next to figures
                figures: 'true',
                // whether to add buttons next to tables
                tables: 'true',
                // whether plugin is on or not
                enabled: 'true'
            };

            // change options above, or override with url parameter, eg:
            // 'manuscript.html?pluginName-enabled=false'

            // /////////////////////////
            // SCRIPT
            // /////////////////////////

            // start script
            function start() {
                if (options.references !== 'false')
                    makeReferenceButtons();
                if (options.figures !== 'false')
                    makeFigureButtons();
                if (options.tables !== 'false')
                    makeTableButtons();
            }

            // when jump button clicked
            function onButtonClick() {
                const first = getFirstOccurrence(this.dataset.id);
                if (!first)
                    return;

                // update url hash so navigating "back" in history will return
                // user to jump button
                window.location.hash = this.dataset.id;
                // scroll to link
                window.setTimeout(function () {
                    goToElement(first, window.innerHeight * 0.5);
                }, 0);
            }

            // get first occurence of link to item in document
            function getFirstOccurrence(id) {
                let query = 'a';
                query += '[href="#' + id + '"]';
                // exclude buttons, anchor links, toc links, etc
                query +=
                    ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
                return document.querySelector(query);
            }

            // add button next to each reference entry
            function makeReferenceButtons() {
                const references = document.querySelectorAll('div[id^="ref-"]');
                for (const reference of references) {
                    // get reference id and element to add button to
                    const id = reference.id;
                    const container = reference.firstElementChild;
                    const first = getFirstOccurrence(id);

                    // if can't find link to reference, ignore
                    if (!first)
                        continue;

                    // make jump button
                    let button = document.createElement('button');
                    button.classList.add('icon_button', 'jump_arrow');
                    button.title =
                        'Jump to the first occurence of this reference in the document';
                    button.innerHTML = document.querySelector(
                        '.icon_angle_double_up'
                    ).innerHTML;
                    button.dataset.id = id;
                    button.dataset.ignore = 'true';
                    container.innerHTML = button.outerHTML + container.innerHTML;
                    button = container.firstElementChild;
                    button.addEventListener('click', onButtonClick);
                }
            }

            // add button next to each figure
            function makeFigureButtons() {
                const figures = document.querySelectorAll('[id^="fig:"]');
                for (const figure of figures) {
                    // get figure id and element to add button to
                    const id = figure.id;
                    const container = figure.querySelector('figcaption') || figure;
                    const first = getFirstOccurrence(id);

                    // if can't find link to figure, ignore
                    if (!first)
                        continue;

                    // make jump button
                    const button = document.createElement('button');
                    button.classList.add('icon_button', 'jump_arrow');
                    button.title =
                        'Jump to the first occurence of this figure in the document';
                    button.innerHTML = document.querySelector(
                        '.icon_angle_double_up'
                    ).innerHTML;
                    button.dataset.id = id;
                    button.dataset.ignore = 'true';
                    container.insertBefore(button, container.firstElementChild);
                    button.addEventListener('click', onButtonClick);
                }
            }

            // add button next to each figure
            function makeTableButtons() {
                const tables = document.querySelectorAll('[id^="tbl:"]');
                for (const table of tables) {
                    // get ref id and element to add button to
                    const id = table.id;
                    const container = table.querySelector('caption') || table;
                    const first = getFirstOccurrence(id);

                    // if can't find link to table, ignore
                    if (!first)
                        continue;

                    // make jump button
                    const button = document.createElement('button');
                    button.classList.add('icon_button', 'jump_arrow');
                    button.title =
                        'Jump to the first occurence of this table in the document';
                    button.innerHTML = document.querySelector(
                        '.icon_angle_double_up'
                    ).innerHTML;
                    button.dataset.id = id;
                    button.dataset.ignore = 'true';
                    container.insertBefore(button, container.firstElementChild);
                    button.addEventListener('click', onButtonClick);
                }
            }

            // scroll to and focus element
            function goToElement(element, offset) {
                // expand accordion section if collapsed
                expandElement(element);
                const y =
                    getRectInView(element).top -
                    getRectInView(document.documentElement).top -
                    (offset || 0);
                // trigger any function listening for "onscroll" event
                window.dispatchEvent(new Event('scroll'));
                window.scrollTo(0, y);
                document.activeElement.blur();
                element.focus();
            }

            // get position/dimensions of element or viewport
            function getRectInView(element) {
                let rect = {};
                rect.left = 0;
                rect.top = 0;
                rect.right = document.documentElement.clientWidth;
                rect.bottom = document.documentElement.clientHeight;
                let style = {};

                if (element instanceof HTMLElement) {
                    rect = element.getBoundingClientRect();
                    style = window.getComputedStyle(element);
                }

                const margin = {};
                margin.left = parseFloat(style.marginLeftWidth) || 0;
                margin.top = parseFloat(style.marginTopWidth) || 0;
                margin.right = parseFloat(style.marginRightWidth) || 0;
                margin.bottom = parseFloat(style.marginBottomWidth) || 0;

                const border = {};
                border.left = parseFloat(style.borderLeftWidth) || 0;
                border.top = parseFloat(style.borderTopWidth) || 0;
                border.right = parseFloat(style.borderRightWidth) || 0;
                border.bottom = parseFloat(style.borderBottomWidth) || 0;

                const newRect = {};
                newRect.left = rect.left + margin.left + border.left;
                newRect.top = rect.top + margin.top + border.top;
                newRect.right = rect.right + margin.right + border.right;
                newRect.bottom = rect.bottom + margin.bottom + border.bottom;
                newRect.width = newRect.right - newRect.left;
                newRect.height = newRect.bottom - newRect.top;

                return newRect;
            }

            // get closest element before specified element that matches query
            function firstBefore(element, query) {
                while (
                    element &&
                    element !== document.body &&
                    !element.matches(query)
                )
                    element = element.previousElementSibling || element.parentNode;

                return element;
            }

            // check if element is part of collapsed heading
            function isCollapsed(element) {
                while (element && element !== document.body) {
                    if (element.dataset.collapsed === 'true')
                        return true;
                    element = element.parentNode;
                }
                return false;
            }

            // (for interaction with accordion plugin)
            // expand heading containing element if necesary
            function expandElement(element) {
                if (isCollapsed(element)) {
                    const heading = firstBefore(element, 'h2');
                    if (heading)
                        heading.click();
                }
            }

            // load options from url parameters
            function loadOptions() {
                const url = window.location.search;
                const params = new URLSearchParams(url);
                for (const optionName of Object.keys(options)) {
                    const paramName = pluginName + '-' + optionName;
                    const param = params.get(paramName);
                    if (param !== '' && param !== null)
                        options[optionName] = param;
                }
            }
            loadOptions();

            // start script when document is finished loading
            if (options.enabled === 'true')
                window.addEventListener('load', start);
        })();
    </script>

    <!-- angle double up icon -->

    <template class="icon_angle_double_up">
        <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
        <svg width="16" height="16" viewBox="0 0 320 512">
            <path fill="currentColor"
                d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z">
            </path>
        </svg>
    </template>
    <!-- link highlight plugin -->

    <script>
        (function () {
            // /////////////////////////
            // DESCRIPTION
            // /////////////////////////

            // This Manubot plugin makes it such that when a user hovers or
            // focuses a link, other links that have the same target will be
            // highlighted. It also makes it such that when clicking a link, the
            // target of the link (eg reference, figure, table) is briefly
            // highlighted.

            // /////////////////////////
            // OPTIONS
            // /////////////////////////

            // plugin name prefix for url parameters
            const pluginName = 'linkHighlight';

            // default plugin options
            const options = {
                // whether to also highlight links that go to external urls
                externalLinks: 'false',
                // whether user must click off to unhighlight instead of just
                // un-hovering
                clickUnhighlight: 'false',
                // whether to also highlight links that are unique
                highlightUnique: 'true',
                // whether plugin is on or not
                enabled: 'true'
            };

            // change options above, or override with url parameter, eg:
            // 'manuscript.html?pluginName-enabled=false'

            // /////////////////////////
            // SCRIPT
            // /////////////////////////

            // start script
            function start() {
                const links = getLinks();
                for (const link of links) {
                    // attach mouse and focus listeners to link
                    link.addEventListener('mouseenter', onLinkFocus);
                    link.addEventListener('focus', onLinkFocus);
                    link.addEventListener('mouseleave', onLinkUnhover);
                }

                // attach click and hash change listeners to window
                window.addEventListener('click', onClick);
                window.addEventListener('touchstart', onClick);
                window.addEventListener('hashchange', onHashChange);

                // run hash change on window load in case user has navigated
                // directly to hash
                onHashChange();
            }

            // when link is focused (tabbed to) or hovered
            function onLinkFocus() {
                highlight(this);
            }

            // when link is unhovered
            function onLinkUnhover() {
                if (options.clickUnhighlight !== 'true')
                    unhighlightAll();
            }

            // when the mouse is clicked anywhere in window
            function onClick(event) {
                unhighlightAll();
            }

            // when hash (eg manuscript.html#introduction) changes
            function onHashChange() {
                const target = getHashTarget();
                if (target)
                    glowElement(target);
            }

            // get element that is target of link or url hash
            function getHashTarget(link) {
                const hash = link ? link.hash : window.location.hash;
                const id = hash.slice(1);
                let target = document.querySelector('[id="' + id + '"]');
                if (!target)
                    return;

                return target;
            }

            // start glow sequence on an element
            function glowElement(element) {
                const startGlow = function () {
                    onGlowEnd();
                    element.dataset.glow = 'true';
                    element.addEventListener('animationend', onGlowEnd);
                };
                const onGlowEnd = function () {
                    element.removeAttribute('data-glow');
                    element.removeEventListener('animationend', onGlowEnd);
                };
                startGlow();
            }

            // highlight link and all others with same target
            function highlight(link) {
                // force unhighlight all to start fresh
                unhighlightAll();

                // get links with same target
                if (!link)
                    return;
                const sameLinks = getSameLinks(link);

                // if link unique and option is off, exit and don't highlight
                if (sameLinks.length <= 1 && options.highlightUnique !== 'true')
                    return;

                // highlight all same links, and "select" (special highlight) this
                // one
                for (const sameLink of sameLinks) {
                    if (sameLink === link)
                        sameLink.setAttribute('data-selected', 'true');
                    else
                        sameLink.setAttribute('data-highlighted', 'true');
                }
            }

            // unhighlight all links
            function unhighlightAll() {
                const links = getLinks();
                for (const link of links) {
                    link.setAttribute('data-selected', 'false');
                    link.setAttribute('data-highlighted', 'false');
                }
            }

            // get links with same target
            function getSameLinks(link) {
                const results = [];
                const links = getLinks();
                for (const otherLink of links) {
                    if (
                        otherLink.getAttribute('href') === link.getAttribute('href')
                    )
                        results.push(otherLink);
                }
                return results;
            }

            // get all links of types we wish to handle
            function getLinks() {
                let query = 'a';
                if (options.externalLinks !== 'true')
                    query += '[href^="#"]';
                // exclude buttons, anchor links, toc links, etc
                query +=
                    ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
                return document.querySelectorAll(query);
            }

            // load options from url parameters
            function loadOptions() {
                const url = window.location.search;
                const params = new URLSearchParams(url);
                for (const optionName of Object.keys(options)) {
                    const paramName = pluginName + '-' + optionName;
                    const param = params.get(paramName);
                    if (param !== '' && param !== null)
                        options[optionName] = param;
                }
            }
            loadOptions();

            // start script when document is finished loading
            if (options.enabled === 'true')
                window.addEventListener('load', start);
        })();
    </script>
    <!-- table of contents plugin -->

    <script>
        (function () {
            // /////////////////////////
            // DESCRIPTION
            // /////////////////////////

            // This Manubot plugin provides a "table of contents" (toc) panel on
            // the side of the document that allows the user to conveniently
            // navigate between sections of the document.

            // /////////////////////////
            // OPTIONS
            // /////////////////////////

            // plugin name prefix for url parameters
            const pluginName = 'tableOfContents';

            // default plugin options
            const options = {
                // which types of elements to add links for, in
                // "document.querySelector" format
                typesQuery: 'h1, h2, h3',
                // whether default behavior is to be closed ('false'), open
                // ('true'), or only open when screen wide enough to fit panel
                // ('auto'). note: still always starts closed when page loads.
                open: 'auto',
                // if list item is more than this many characters, text will be
                // truncated
                charLimit: '50',
                // whether or not to show bullets next to each toc item
                bullets: 'false',
                // whether plugin is on or not
                enabled: 'true'
            };

            // change options above, or override with url parameter, eg:
            // 'manuscript.html?pluginName-enabled=false'

            // /////////////////////////
            // SCRIPT
            // /////////////////////////

            // start script
            function start() {
                // make toc panel and populate with entries (links to document
                // sections)
                const panel = makePanel();
                if (!panel)
                    return;
                makeEntries(panel);
                document.body.insertBefore(panel, document.body.firstChild);

                closePanel();

                // attach click, scroll, and hash change listeners to window
                window.addEventListener('click', onClick);
                window.addEventListener('touchstart', onClick);
                window.addEventListener('scroll', onScroll);
                window.addEventListener('hashchange', onScroll);
                window.addEventListener('keyup', onKeyUp);
                onScroll();

                // add class to push document body down out of way of toc button
                document.body.classList.add('toc_body_nudge');
            }

            // determine if screen wide enough to fit toc panel
            function isSmallScreen() {
                // in default theme:
                // 816px = 8.5in = width of "page" (<body>) element
                // 260px = min width of toc panel (*2 for both sides of <body>)
                return window.innerWidth < 816 + 260 * 2;
            }

            // open/close panel based on option and screen size
            function openOrClosePanel() {
                if (
                    options.open === 'true' ||
                    (options.open === 'auto' && !isSmallScreen())
                )
                    openPanel();
                else
                    closePanel();
            }

            // when mouse is clicked anywhere in window
            function onClick() {
                const panel = document.getElementById('toc_panel');
                if (!panel)
                    return;

                if (panel.dataset.open === 'true')
                    openOrClosePanel();
            }

            // when window is scrolled or hash changed
            function onScroll() {
                highlightViewed();
            }

            // when key pressed
            function onKeyUp(event) {
                if (!event || !event.key)
                    return;

                // close on esc
                if (event.key === 'Escape')
                    closePanel();
            }

            // find entry of currently viewed document section in toc and highlight
            function highlightViewed() {
                const firstId = getFirstInView(options.typesQuery);

                // get toc entries (links), unhighlight all, then highlight viewed
                const list = document.getElementById('toc_list');
                if (!firstId || !list)
                    return;
                const links = list.querySelectorAll('a');
                for (const link of links)
                    link.dataset.viewing = 'false';
                const link = list.querySelector('a[href="#' + firstId + '"]');
                if (!link)
                    return;
                link.dataset.viewing = 'true';
            }

            // get first or previous toc listed element in top half of view
            function getFirstInView(query) {
                // get all elements matching query and with id
                const elements = document.querySelectorAll(query);
                const elementsWithIds = [];
                for (const element of elements) {
                    if (element.id)
                        elementsWithIds.push(element);
                }


                // get first or previous element in top half of view
                for (let i = 0; i < elementsWithIds.length; i++) {
                    const element = elementsWithIds[i];
                    const prevElement = elementsWithIds[Math.max(0, i - 1)];
                    if (element.getBoundingClientRect().top >= 0) {
                        if (
                            element.getBoundingClientRect().top <
                            window.innerHeight / 2
                        )
                            return element.id;
                        else
                            return prevElement.id;
                    }
                }
            }

            // make panel
            function makePanel() {
                // create panel
                const panel = document.createElement('div');
                panel.id = 'toc_panel';
                if (options.bullets === 'true')
                    panel.dataset.bullets = 'true';

                // create header
                const header = document.createElement('div');
                header.id = 'toc_header';

                // create toc button
                const button = document.createElement('button');
                button.id = 'toc_button';
                button.innerHTML = document.querySelector('.icon_th_list').innerHTML;
                button.title = 'Table of Contents';
                button.classList.add('icon_button');

                // create header text
                const text = document.createElement('h3');
                text.innerHTML = 'View Table of Contents';

                // create container for toc list
                const list = document.createElement('div');
                list.id = 'toc_list';

                // attach click listeners
                panel.addEventListener('click', onPanelClick);
                header.addEventListener('click', onHeaderClick);
                button.addEventListener('click', onButtonClick);

                // attach elements
                header.appendChild(button);
                header.appendChild(text);
                panel.appendChild(header);
                panel.appendChild(list);

                return panel;
            }

            // create toc entries (links) to each element of the specified types
            function makeEntries(panel) {
                const elements = document.querySelectorAll(options.typesQuery);
                for (const element of elements) {
                    // do not add link if element doesn't have assigned id
                    if (!element.id)
                        continue;

                    // create link/list item
                    const link = document.createElement('a');
                    link.classList.add('toc_link');
                    switch (element.tagName.toLowerCase()) {
                        case 'h1':
                            link.dataset.level = '1';
                            break;
                        case 'h2':
                            link.dataset.level = '2';
                            break;
                        case 'h3':
                            link.dataset.level = '3';
                            break;
                        case 'h4':
                            link.dataset.level = '4';
                            break;
                    }
                    link.title = element.innerText;
                    let text = element.innerText;
                    if (text.length > options.charLimit)
                        text = text.slice(0, options.charLimit) + '...';
                    link.innerHTML = text;
                    link.href = '#' + element.id;
                    link.addEventListener('click', onLinkClick);

                    // attach link
                    panel.querySelector('#toc_list').appendChild(link);
                }
            }

            // when panel is clicked
            function onPanelClick(event) {
                // stop click from propagating to window/document and closing panel
                event.stopPropagation();
            }

            // when header itself is clicked
            function onHeaderClick(event) {
                togglePanel();
            }

            // when button is clicked
            function onButtonClick(event) {
                togglePanel();
                // stop header underneath button from also being clicked
                event.stopPropagation();
            }

            // when link is clicked
            function onLinkClick() {
                openOrClosePanel();
            }

            // open panel if closed, close if opened
            function togglePanel() {
                const panel = document.getElementById('toc_panel');
                if (!panel)
                    return;

                if (panel.dataset.open === 'true')
                    closePanel();
                else
                    openPanel();
            }

            // open panel
            function openPanel() {
                const panel = document.getElementById('toc_panel');
                if (panel)
                    panel.dataset.open = 'true';
            }

            // close panel
            function closePanel() {
                const panel = document.getElementById('toc_panel');
                if (panel)
                    panel.dataset.open = 'false';
            }

            // load options from url parameters
            function loadOptions() {
                const url = window.location.search;
                const params = new URLSearchParams(url);
                for (const optionName of Object.keys(options)) {
                    const paramName = pluginName + '-' + optionName;
                    const param = params.get(paramName);
                    if (param !== '' && param !== null)
                        options[optionName] = param;
                }
            }
            loadOptions();

            // start script when document is finished loading
            if (options.enabled === 'true')
                window.addEventListener('load', start);
        })();
    </script>

    <!-- th list icon -->

    <template class="icon_th_list">
        <!-- modified from: https://fontawesome.com/icons/th-list -->
        <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
            <path fill="currentColor"
                d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                tabindex="-1"></path>
        </svg>
    </template>
    <!-- lightbox plugin -->

    <script>
        (function () {
            // /////////////////////////
            // DESCRIPTION
            // /////////////////////////

            // This Manubot plugin makes it such that when a user clicks on an
            // image, the image fills the screen and the user can pan/drag/zoom
            // the image and navigate between other images in the document.

            // /////////////////////////
            // OPTIONS
            // /////////////////////////

            // plugin name prefix for url parameters
            const pluginName = 'lightbox';

            // default plugin options
            const options = {
                // list of possible zoom/scale factors
                zoomSteps:
                    '0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1,' +
                    '1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8',
                // whether to fit image to view ('fit'), display at 100% and shrink
                // if necessary ('shrink'), or always display at 100% ('100')
                defaultZoom: 'fit',
                // whether to zoom in/out toward center of view ('true') or mouse
                // ('false')
                centerZoom: 'false',
                // whether plugin is on or not
                enabled: 'true'
            };

            // change options above, or override with url parameter, eg:
            // 'manuscript.html?pluginName-enabled=false'

            // /////////////////////////
            // SCRIPT
            // /////////////////////////

            // start script
            function start() {
                // run through each <img> element
                const imgs = document.querySelectorAll('figure > img');
                let count = 1;
                for (const img of imgs) {
                    img.classList.add('lightbox_document_img');
                    img.dataset.number = count;
                    img.dataset.total = imgs.length;
                    img.addEventListener('click', openLightbox);
                    count++;
                }

                // attach mouse and key listeners to window
                window.addEventListener('mousemove', onWindowMouseMove);
                window.addEventListener('keyup', onKeyUp);
            }

            // when mouse is moved anywhere in window
            function onWindowMouseMove(event) {
                window.mouseX = event.clientX;
                window.mouseY = event.clientY;
            }

            // when key pressed
            function onKeyUp(event) {
                if (!event || !event.key)
                    return;

                switch (event.key) {
                    // trigger click of prev button
                    case 'ArrowLeft':
                        const prevButton = document.getElementById(
                            'lightbox_prev_button'
                        );
                        if (prevButton)
                            prevButton.click();
                        break;
                    // trigger click of next button
                    case 'ArrowRight':
                        const nextButton = document.getElementById(
                            'lightbox_next_button'
                        );
                        if (nextButton)
                            nextButton.click();
                        break;
                    // close on esc
                    case 'Escape':
                        closeLightbox();
                        break;
                }
            }

            // open lightbox
            function openLightbox() {
                const lightbox = makeLightbox(this);
                if (!lightbox)
                    return;

                blurBody(lightbox);
                document.body.appendChild(lightbox);
            }

            // make lightbox
            function makeLightbox(img) {
                // delete lightbox if it exists, start fresh
                closeLightbox();

                // create screen overlay containing lightbox
                const overlay = document.createElement('div');
                overlay.id = 'lightbox_overlay';

                // create image info boxes
                const numberInfo = document.createElement('div');
                const zoomInfo = document.createElement('div');
                numberInfo.id = 'lightbox_number_info';
                zoomInfo.id = 'lightbox_zoom_info';

                // create container for image
                const imageContainer = document.createElement('div');
                imageContainer.id = 'lightbox_image_container';
                const lightboxImg = makeLightboxImg(
                    img,
                    imageContainer,
                    numberInfo,
                    zoomInfo
                );
                imageContainer.appendChild(lightboxImg);

                // create bottom container for caption and navigation buttons
                const bottomContainer = document.createElement('div');
                bottomContainer.id = 'lightbox_bottom_container';
                const caption = makeCaption(img);
                const prevButton = makePrevButton(img);
                const nextButton = makeNextButton(img);
                bottomContainer.appendChild(prevButton);
                bottomContainer.appendChild(caption);
                bottomContainer.appendChild(nextButton);

                // attach top middle and bottom to overlay
                overlay.appendChild(numberInfo);
                overlay.appendChild(zoomInfo);
                overlay.appendChild(imageContainer);
                overlay.appendChild(bottomContainer);

                return overlay;
            }

            // make <img> object that is intuitively draggable and zoomable
            function makeLightboxImg(
                sourceImg,
                container,
                numberInfoBox,
                zoomInfoBox
            ) {
                // create copy of source <img>
                const img = sourceImg.cloneNode(true);
                img.classList.remove('lightbox_document_img');
                img.removeAttribute('id');
                img.removeAttribute('width');
                img.removeAttribute('height');
                img.style.position = 'unset';
                img.style.margin = '0';
                img.style.padding = '0';
                img.style.width = '';
                img.style.height = '';
                img.style.minWidth = '';
                img.style.minHeight = '';
                img.style.maxWidth = '';
                img.style.maxHeight = '';
                img.id = 'lightbox_img';

                // build sorted list of unique zoomSteps, always including a 100%
                let zoomSteps = [];
                const optionsZooms = options.zoomSteps.split(/[^0-9.]/);
                for (const optionZoom of optionsZooms) {
                    const newZoom = parseFloat(optionZoom);
                    if (newZoom && !zoomSteps.includes(newZoom))
                        zoomSteps.push(newZoom);
                }
                if (!zoomSteps.includes(1))
                    zoomSteps.push(1);
                zoomSteps = zoomSteps.sort(function sortNumber(a, b) {
                    return a - b;
                });

                // <img> object property variables
                let zoom = 1;
                let translateX = 0;
                let translateY = 0;
                let clickMouseX = undefined;
                let clickMouseY = undefined;
                let clickTranslateX = undefined;
                let clickTranslateY = undefined;

                updateNumberInfo();

                // update image numbers displayed in info box
                function updateNumberInfo() {
                    numberInfoBox.innerHTML =
                        sourceImg.dataset.number + ' of ' + sourceImg.dataset.total;
                }

                // update zoom displayed in info box
                function updateZoomInfo() {
                    let zoomInfo = zoom * 100;
                    if (!Number.isInteger(zoomInfo))
                        zoomInfo = zoomInfo.toFixed(2);
                    zoomInfoBox.innerHTML = zoomInfo + '%';
                }

                // move to closest zoom step above current zoom
                const zoomIn = function () {
                    for (const zoomStep of zoomSteps) {
                        if (zoomStep > zoom) {
                            zoom = zoomStep;
                            break;
                        }
                    }
                    updateTransform();
                };

                // move to closest zoom step above current zoom
                const zoomOut = function () {
                    zoomSteps.reverse();
                    for (const zoomStep of zoomSteps) {
                        if (zoomStep < zoom) {
                            zoom = zoomStep;
                            break;
                        }
                    }
                    zoomSteps.reverse();

                    updateTransform();
                };

                // update display of <img> based on scale/translate properties
                const updateTransform = function () {
                    // set transform
                    img.style.transform =
                        'translate(' +
                        (translateX || 0) +
                        'px,' +
                        (translateY || 0) +
                        'px) scale(' +
                        (zoom || 1) +
                        ')';

                    // get new width/height after scale
                    const rect = img.getBoundingClientRect();
                    // limit translate
                    translateX = Math.max(translateX, -rect.width / 2);
                    translateX = Math.min(translateX, rect.width / 2);
                    translateY = Math.max(translateY, -rect.height / 2);
                    translateY = Math.min(translateY, rect.height / 2);

                    // set transform
                    img.style.transform =
                        'translate(' +
                        (translateX || 0) +
                        'px,' +
                        (translateY || 0) +
                        'px) scale(' +
                        (zoom || 1) +
                        ')';

                    updateZoomInfo();
                };

                // fit <img> to container
                const fit = function () {
                    // no x/y offset, 100% zoom by default
                    translateX = 0;
                    translateY = 0;
                    zoom = 1;

                    // widths of <img> and container
                    const imgWidth = img.naturalWidth;
                    const imgHeight = img.naturalHeight;
                    const containerWidth = parseFloat(
                        window.getComputedStyle(container).width
                    );
                    const containerHeight = parseFloat(
                        window.getComputedStyle(container).height
                    );

                    // how much zooming is needed to fit <img> to container
                    const xRatio = imgWidth / containerWidth;
                    const yRatio = imgHeight / containerHeight;
                    const maxRatio = Math.max(xRatio, yRatio);
                    const newZoom = 1 / maxRatio;

                    // fit <img> to container according to option
                    if (options.defaultZoom === 'shrink') {
                        if (maxRatio > 1)
                            zoom = newZoom;
                    } else if (options.defaultZoom === 'fit')
                        zoom = newZoom;

                    updateTransform();
                };

                // when mouse wheel is rolled anywhere in container
                const onContainerWheel = function (event) {
                    if (!event)
                        return;

                    // let ctrl + mouse wheel to zoom behave as normal
                    if (event.ctrlKey)
                        return;

                    // prevent normal scroll behavior
                    event.preventDefault();
                    event.stopPropagation();

                    // point around which to scale img
                    const viewRect = container.getBoundingClientRect();
                    const viewX = (viewRect.left + viewRect.right) / 2;
                    const viewY = (viewRect.top + viewRect.bottom) / 2;
                    const originX = options.centerZoom === 'true' ? viewX : mouseX;
                    const originY = options.centerZoom === 'true' ? viewY : mouseY;

                    // get point on image under origin
                    const oldRect = img.getBoundingClientRect();
                    const oldPercentX = (originX - oldRect.left) / oldRect.width;
                    const oldPercentY = (originY - oldRect.top) / oldRect.height;

                    // increment/decrement zoom
                    if (event.deltaY < 0)
                        zoomIn();
                    if (event.deltaY > 0)
                        zoomOut();

                    // get offset between previous image point and origin
                    const newRect = img.getBoundingClientRect();
                    const offsetX =
                        originX - (newRect.left + newRect.width * oldPercentX);
                    const offsetY =
                        originY - (newRect.top + newRect.height * oldPercentY);

                    // translate image to keep image point under origin
                    translateX += offsetX;
                    translateY += offsetY;

                    // perform translate
                    updateTransform();
                };

                // when container is clicked
                function onContainerClick(event) {
                    // if container itself is target of click, and not other
                    // element above it
                    if (event.target === this)
                        closeLightbox();
                }

                // when mouse button is pressed on image
                const onImageMouseDown = function (event) {
                    // store original mouse position relative to image
                    clickMouseX = window.mouseX;
                    clickMouseY = window.mouseY;
                    clickTranslateX = translateX;
                    clickTranslateY = translateY;
                    event.stopPropagation();
                    event.preventDefault();
                };

                // when mouse button is released anywhere in window
                const onWindowMouseUp = function (event) {
                    // reset original mouse position
                    clickMouseX = undefined;
                    clickMouseY = undefined;
                    clickTranslateX = undefined;
                    clickTranslateY = undefined;

                    // remove global listener if lightbox removed from document
                    if (!document.body.contains(container))
                        window.removeEventListener('mouseup', onWindowMouseUp);
                };

                // when mouse is moved anywhere in window
                const onWindowMouseMove = function (event) {
                    if (
                        clickMouseX === undefined ||
                        clickMouseY === undefined ||
                        clickTranslateX === undefined ||
                        clickTranslateY === undefined
                    )
                        return;

                    // offset image based on original and current mouse position
                    translateX = clickTranslateX + window.mouseX - clickMouseX;
                    translateY = clickTranslateY + window.mouseY - clickMouseY;
                    updateTransform();
                    event.preventDefault();

                    // remove global listener if lightbox removed from document
                    if (!document.body.contains(container))
                        window.removeEventListener('mousemove', onWindowMouseMove);
                };

                // when window is resized
                const onWindowResize = function (event) {
                    fit();

                    // remove global listener if lightbox removed from document
                    if (!document.body.contains(container))
                        window.removeEventListener('resize', onWindowResize);
                };

                // attach the necessary event listeners
                img.addEventListener('dblclick', fit);
                img.addEventListener('mousedown', onImageMouseDown);
                container.addEventListener('wheel', onContainerWheel);
                container.addEventListener('mousedown', onContainerClick);
                container.addEventListener('touchstart', onContainerClick);
                window.addEventListener('mouseup', onWindowMouseUp);
                window.addEventListener('mousemove', onWindowMouseMove);
                window.addEventListener('resize', onWindowResize);

                // run fit() after lightbox atttached to document and <img> Loaded
                // so needed container and img dimensions available
                img.addEventListener('load', fit);

                return img;
            }

            // make caption
            function makeCaption(img) {
                const caption = document.createElement('div');
                caption.id = 'lightbox_caption';
                const captionSource = img.nextElementSibling;
                if (captionSource.tagName.toLowerCase() === 'figcaption') {
                    const captionCopy = makeCopy(captionSource);
                    caption.innerHTML = captionCopy.innerHTML;
                }

                caption.addEventListener('touchstart', function (event) {
                    event.stopPropagation();
                });

                return caption;
            }

            // make carbon copy of html dom element
            function makeCopy(source) {
                const sourceCopy = source.cloneNode(true);

                // delete elements marked with ignore (eg anchor and jump buttons)
                const deleteFromCopy = sourceCopy.querySelectorAll(
                    '[data-ignore="true"]'
                );
                for (const element of deleteFromCopy)
                    element.remove();

                // delete certain element attributes
                const attributes = [
                    'id',
                    'data-collapsed',
                    'data-selected',
                    'data-highlighted',
                    'data-glow'
                ];
                for (const attribute of attributes) {
                    sourceCopy.removeAttribute(attribute);
                    const elements = sourceCopy.querySelectorAll(
                        '[' + attribute + ']'
                    );
                    for (const element of elements)
                        element.removeAttribute(attribute);
                }

                return sourceCopy;
            }

            // make button to jump to previous image in document
            function makePrevButton(img) {
                const prevButton = document.createElement('button');
                prevButton.id = 'lightbox_prev_button';
                prevButton.title = 'Jump to the previous image in the document [←]';
                prevButton.classList.add('icon_button', 'lightbox_button');
                prevButton.innerHTML = document.querySelector(
                    '.icon_caret_left'
                ).innerHTML;

                // attach click listeners to button
                prevButton.addEventListener('click', function () {
                    getPrevImg(img).click();
                });

                return prevButton;
            }

            // make button to jump to next image in document
            function makeNextButton(img) {
                const nextButton = document.createElement('button');
                nextButton.id = 'lightbox_next_button';
                nextButton.title = 'Jump to the next image in the document [→]';
                nextButton.classList.add('icon_button', 'lightbox_button');
                nextButton.innerHTML = document.querySelector(
                    '.icon_caret_right'
                ).innerHTML;

                // attach click listeners to button
                nextButton.addEventListener('click', function () {
                    getNextImg(img).click();
                });

                return nextButton;
            }

            // get previous image in document
            function getPrevImg(img) {
                const imgs = document.querySelectorAll('.lightbox_document_img');

                // find index of provided img
                let index;
                for (index = 0; index < imgs.length; index++) {
                    if (imgs[index] === img)
                        break;
                }


                // wrap index to other side if < 1
                if (index - 1 >= 0)
                    index--;
                else
                    index = imgs.length - 1;
                return imgs[index];
            }

            // get next image in document
            function getNextImg(img) {
                const imgs = document.querySelectorAll('.lightbox_document_img');

                // find index of provided img
                let index;
                for (index = 0; index < imgs.length; index++) {
                    if (imgs[index] === img)
                        break;
                }


                // wrap index to other side if > total
                if (index + 1 <= imgs.length - 1)
                    index++;
                else
                    index = 0;
                return imgs[index];
            }

            // close lightbox
            function closeLightbox() {
                focusBody();

                const lightbox = document.getElementById('lightbox_overlay');
                if (lightbox)
                    lightbox.remove();
            }

            // make all elements behind lightbox non-focusable
            function blurBody(overlay) {
                const all = document.querySelectorAll('*');
                for (const element of all)
                    element.tabIndex = -1;
                document.body.classList.add('body_no_scroll');
            }

            // make all elements focusable again
            function focusBody() {
                const all = document.querySelectorAll('*');
                for (const element of all)
                    element.removeAttribute('tabIndex');
                document.body.classList.remove('body_no_scroll');
            }

            // load options from url parameters
            function loadOptions() {
                const url = window.location.search;
                const params = new URLSearchParams(url);
                for (const optionName of Object.keys(options)) {
                    const paramName = pluginName + '-' + optionName;
                    const param = params.get(paramName);
                    if (param !== '' && param !== null)
                        options[optionName] = param;
                }
            }
            loadOptions();

            // start script when document is finished loading
            if (options.enabled === 'true')
                window.addEventListener('load', start);
        })();
    </script>

    <!-- caret left icon -->

    <template class="icon_caret_left">
        <!-- modified from: https://fontawesome.com/icons/caret-left -->
        <svg width="16" height="16" viewBox="0 0 192 512">
            <path fill="currentColor"
                d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z">
            </path>
        </svg>
    </template>

    <!-- caret right icon -->

    <template class="icon_caret_right">
        <!-- modified from: https://fontawesome.com/icons/caret-right -->
        <svg width="16" height="16" viewBox="0 0 192 512">
            <path fill="currentColor"
                d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z">
            </path>
        </svg>
    </template>
    <!-- attributes plugin -->

    <script>
        (function () {
            // /////////////////////////
            // DESCRIPTION
            // /////////////////////////

            // This Manubot plugin allows arbitrary HTML attributes to be attached
            // to (almost) any element. Place an HTML comment inside or next to the
            // desired element in the format <!-- $attribute="value" -->

            // /////////////////////////
            // OPTIONS
            // /////////////////////////

            // plugin name prefix for url parameters
            const pluginName = 'attributes';

            // default plugin options
            const options = {
                // whether plugin is on or not
                enabled: 'true'
            };

            // change options above, or override with url parameter, eg:
            // 'manuscript.html?pluginName-enabled=false'

            // /////////////////////////
            // SCRIPT
            // /////////////////////////

            // start script
            function start() {
                // get list of comments in document
                const comments = findComments();

                for (const comment of comments)
                    if (comment.parentElement)
                        addAttributes(
                            comment.parentElement,
                            comment.nodeValue.trim()
                        );
            }

            // add html attributes to specified element based on string of 
            // html attributes and values
            function addAttributes(element, text) {
                // regex's for finding attribute/value pairs in the format of
                // attribute="value" or attribute='value
                const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
                const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

                // loop through attribute/value pairs
                let match;
                while (match = text.match(regex2) || text.match(regex1)) {
                    // get attribute and value from regex capture groups
                    let attribute = match[1];
                    let value = match[2];

                    // remove from string
                    text = text.substring(match.index + match[0].length);

                    if (!attribute || !value)
                        break;

                    // set attribute of parent element
                    try {
                        element.setAttribute(attribute, value);
                    } catch (error) {
                        console.log(error);
                    }

                    // special case for colspan
                    if (attribute === 'colspan')
                        removeTableCells(element, value);
                }
            }

            // get list of comment elements in document
            function findComments() {
                const comments = [];

                // iterate over comment nodes in document
                function acceptNode(node) {
                    return NodeFilter.FILTER_ACCEPT;
                }
                const iterator = document.createNodeIterator(
                    document.body,
                    NodeFilter.SHOW_COMMENT,
                    acceptNode
                );
                let node;
                while (node = iterator.nextNode())
                    comments.push(node);

                return comments;
            }

            // remove certain number of cells after specified cell
            function removeTableCells(cell, number) {
                number = parseInt(number);
                if (!number)
                    return;

                // remove elements
                for (; number > 1; number--) {
                    if (cell.nextElementSibling)
                        cell.nextElementSibling.remove();
                }
            }

            // load options from url parameters
            function loadOptions() {
                const url = window.location.search;
                const params = new URLSearchParams(url);
                for (const optionName of Object.keys(options)) {
                    const paramName = pluginName + '-' + optionName;
                    const param = params.get(paramName);
                    if (param !== '' && param !== null)
                        options[optionName] = param;
                }
            }
            loadOptions();

            // start script when document is finished loading
            if (options.enabled === 'true')
                window.addEventListener('load', start);
        })();
    </script>
    <!-- math plugin configuration -->

    <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        "CommonHTML": { linebreaks: { automatic: true } },
        "HTML-CSS": { linebreaks: { automatic: true } },
        "SVG": { linebreaks: { automatic: true } },
        "fast-preview": { disabled: true }
    });
</script>

    <!-- math plugin -->

    <script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML'>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'MathJax' allows the proper rendering of
    // math/equations written in LaTeX.

    // https://www.mathjax.org/
    </script>
    <!-- annotations plugin -->

    <script>
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This third-party plugin 'Hypothesis' allows public annotation of the
        // manuscript.

        // https://web.hypothes.is/

        // plugin configuration
        window.hypothesisConfig = function () {
            return {
                branding: {
                    accentColor: '#2196f3',
                    appBackgroundColor: '#f8f8f8',
                    ctaBackgroundColor: '#f8f8f8',
                    ctaTextColor: '#000000',
                    selectionFontFamily: 'Open Sans, Helvetica, sans serif',
                    annotationFontFamily: 'Open Sans, Helvetica, sans serif'
                }
            };
        };

        // hypothesis client script
        const embed = 'https://hypothes.is/embed.js';
        // hypothesis annotation count query url
        const query = 'https://api.hypothes.is/api/search?limit=0&url='


        // start script
        function start() {
            const button = makeButton();
            document.body.insertBefore(button, document.body.firstChild);
            insertCount(button);
        }

        // make button
        function makeButton() {
            // create button
            const button = document.createElement('button');
            button.id = 'hypothesis_button';
            button.innerHTML = document.querySelector('.icon_hypothesis').innerHTML;
            button.title = 'Hypothesis annotations';
            button.classList.add('icon_button');

            function onClick(event) {
                onButtonClick(event, button);
            }

            // attach click listeners
            button.addEventListener('click', onClick);

            return button;
        }

        // insert annotations count
        async function insertCount(button) {
            // get annotation count from Hypothesis based on url
            let count = '-';
            try {
                const canonical = document.querySelector('link[rel="canonical"]');
                const location = window.location;
                const url = encodeURIComponent((canonical || location).href);
                const response = await fetch(query + url);
                const json = await response.json();
                count = json.total || '-';
            } catch (error) {
                console.log(error);
            }

            // put count into button
            const counter = document.createElement('span');
            counter.id = 'hypothesis_count';
            counter.innerHTML = count;
            button.title = 'View ' + count + ' Hypothesis annotations';
            button.append(counter);
        }

        // when button is clicked
        function onButtonClick(event, button) {
            const script = document.createElement('script');
            script.src = embed;
            document.body.append(script);
            button.remove();
        }

        window.addEventListener('load', start);
    </script>

    <!-- hypothesis icon -->

    <template class="icon_hypothesis">
        <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
        <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
            <path fill="currentColor"
                d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
                tabindex="-1"></path>
        </svg>
    </template>
    <!-- analytics plugin -->

    <!-- copy and paste code from Google Analytics or similar service here -->
    <script>
        var tables = document.getElementsByClassName("interactive");
        console.log(tables);
        for (var i = 0; i < tables.length; i++) {
            console.log(tables.item(i).childNodes[0])
            var table = new Tabulator(tables.item(i).firstElementChild, {});
        }


        document.getElementById("download-csv").addEventListener("click", function () {
            table.download("csv", "data.csv");
        });



        //trigger download of data.xlsx file
        document.getElementById("download-xlsx").addEventListener("click", function () {
            table.download("xlsx", "data.xlsx", { sheetName: "My Data" });
        });

    </script>
</body>

</html>